# Arthropod metabarcoding (roller and spray eDNA aggregation methods)
Second field effort testing roller and spray aggregation methods with n = 20 samples each.

# Load libraries. Read, clean and format data
```{r}
library(dplyr)
library(readxl)
library(ggplot2)
library(stringr)
# library(vegan)
library(BiodiversityR)
library(wesanderson)
library(patchwork)
library(tidyr)
library(iNEXT)
library(forcats)
select <- dplyr::select
# path for taxonomy and ecology lookup tables
tax_data_path <- "data/arthropod.families_20220816.xlsx"
# note: the 20220816 version is up-to-date for 2methods
#   but hadn't received new families from 4methods yet
'%notin%' <- Negate('%in%')

# read in Zeale & Coleop MOTU data with taxonomy
  # excludes MOTUs that were filtered out during negative subtraction etc.
zs3 <- read.csv("data/pinebugs.samples.onlyarth.negsub.Z.csv") %>%
  filter(sampsum != 0)

cs3 <- read.csv("data/pinebugs.samples.onlyarth.negsub.C.csv") %>%
  filter(sampsum != 0)

# read in family trait info (life mode)
    # note: class and order added with taxize package using script:
    # "scripts/adding_taxonomy_to_families.R"
    # terrestrial/aquatic info from Horren et al. database & others
lifemode <- read_xlsx(tax_data_path, 
                      sheet = "arthropod.families.lifemode") %>%
  select(-person) %>%
  mutate(life.cat = case_when(larva_aquatic >= 1 & 
                                 adult_aquatic >= 1 ~ 
                                 "Fully aquatic",
                               larva_terrestrial >= 1 &
                                 adult_terrestrial >= 1 ~ 
                                 "Fully terrestrial",
                               larva_aquatic >= 1 &
                                 adult_terrestrial >= 1 ~
                                 "All larvae aquatic",
                               TRUE ~ "Some larvae aquatic"))

# list families without trait info
missing.familes <- cs3 %>% 
  select(family) %>%
  bind_rows(select(zs3, family)) %>%
  filter(!is.na(family)) %>%
  left_join(select(lifemode, family, larva_terrestrial)) %>%
  select(family, larva_terrestrial) %>%
  distinct() %>%
  filter(is.na(larva_terrestrial)) %>%
  select(family)

# read in family trait info (diet/feeding)
    # note: class and order added with taxize package using script:
    # "scripts/adding_taxonomy_to_families.R"
    # diet/feeding info from Horren et al. database & others
diet <- read_xlsx(tax_data_path, 
                      sheet = "arthropod.families.diet") %>%
  select(-person) %>%
  # create a condensed diet/feeding classification
  mutate(diet.cat = case_when(final_diet == "mixed" ~ 
                                "multiple categories",
                              final_diet %in% 
                                c("zoophagous", "phytophagous") ~
                                paste(final_diet, feed_mode, 
                                      sep = "-"),
                              final_diet %in% c("saprophagous", 
                                                "detritophagous") ~ 
                                "detritophagous/scavenger",
                              TRUE ~ final_diet),
         diet.cat = case_when(notes == ".9 det/sap/cop" ~ 
                                "detritophagous/scavenger",
                              family %in% c("Calliphoridae",
                                                  "Polleniidae",
                                                  "Rhiniidae") ~
                                "detritophagous/scavenger",
                              diet.cat == "zoophagous-necro" ~
                                "detritophagous/scavenger",
                           diet.cat %in% c("phytophagous-mixed",
                                        "phytophagous-not specified") ~
                             "phytophagous-multiple/other",
                           diet.cat == "phytophagous-flower" ~
                             "phytophagous-multiple/other",
                           diet.cat %in% c("phytophagous-miners",
                                           "phytophagous-gall") ~
                             "phytophagous-galls/miners",
                           diet.cat %in% c("zoophagous-parasite",
                                           "zoophagous-parasitoid",
                                           "zoophagous-micropredator") ~
                             "zoophagous-parasite/parasitoid",
                              TRUE ~ diet.cat),
         diet.cat2 = case_when(diet.cat == 
                                "detritophagous/scavenger" ~ 
                                "Scavenger",
                              diet.cat == 
                                "phytophagous-galls/miners" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-multiple/other" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-phyllophagous" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-sap sucking" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-saproxylic" ~
                                "Phytophagous",
                              diet.cat == 
                                "zoophagous-parasite/parasitoid" ~
                                "Parasite/parasitoid",
                              diet.cat == 
                                "zoophagous-predator" ~
                                "Predator",
                              diet.cat == 
                                "zoophagous-mixed" ~
                                "Predator",
                              diet.cat == 
                                "multiple categories" ~
                                "Mulitple categories",
                              diet.cat == 
                                "mycetophagous" ~
                                "Mycetophagous",
                              TRUE ~ diet.cat))

table(diet$diet.cat)
table(diet$diet.cat2)
# final category notes: 
# detritophagous/scavenger = detrit., copro., sapro., and zoo.-necro.

# read in family trait info (body size)
    # note: class and order added with taxize package using script:
    # "scripts/adding_taxonomy_to_families.R"
    # diet/feeding info from Horren et al. database & others
size <- read_xlsx(tax_data_path, 
                      sheet = "arthropod.families.size") %>%
  select(-person) %>%
  select(family:final_source) %>%
  mutate(Body_Size = case_when(final_mid > 20 ~ ">20 mm",
                               final_mid > 10 & 
                                 final_mid <= 20 ~ "10-20 mm",
                               final_mid > 5 &
                                 final_mid <= 10 ~ "5-10 mm",
                               final_mid <= 5 ~ "<5 mm",
                               TRUE ~ "Unknown"),
         Body_Size_order = case_when(Body_Size == "<5 mm" ~ 5,
                           Body_Size == "5-10 mm" ~ 4,
                           Body_Size == "10-20 mm" ~ 3,
                           Body_Size == ">20 mm" ~ 2,
                           Body_Size == "Unknown" ~ 1)
  )

art_full <- cs3 %>%
  mutate(primer = "coleop") %>%
  bind_rows(mutate(zs3, primer = "zeale")) %>%
  select(primer, id, phylum, class, order, family, 
         genus, species, contains("CS")[c(2:10,1)], contains("DS")[c(2:7,1)],
         contains("CR")[c(2:10,1)], contains("DR")[c(2:10,1)], sampsum) %>%
  rename(sample.sum = sampsum) %>%
  mutate(sample.3CS = case_when(primer == "zeale" ~ 0,
                                  TRUE ~ sample.3CS))

# rename columns
new_col_names <- art_full %>% 
  select(starts_with("sample")) %>%
  colnames() %>%
  substr(., 1, 11) %>%
  gsub(., pattern = ":", replacement = "") %>%
  gsub(., pattern = "B", replacement = "") %>%
  gsub(., pattern = "Z", replacement = "")

colnames(art_full) # check which columns are being replaced

colnames(art_full)[9:46] <- c(new_col_names)

colnames(art_full) # check that it worked

# look at read depth
sort(colSums(select(filter(art_full, primer == "coleop"), starts_with("sample")), na.rm = T))
sort(colSums(select(filter(art_full, primer == "zeale"), starts_with("sample")), na.rm = T))

mean(colSums(select(filter(art_full, primer == "coleop"), starts_with("sample"), -sample.sum), na.rm = T))
mean(sort(colSums(select(filter(art_full, primer == "zeale"), starts_with("sample"), -sample.sum), na.rm = T))[2:37])

csd <- sd(colSums(select(filter(art_full, primer == "coleop"), starts_with("sample"), -sample.sum), na.rm = T))
zsd <- sd(sort(colSums(select(filter(art_full, primer == "zeale"), starts_with("sample"), -sample.sum), na.rm = T))[2:37])

cn <- sum(colSums(select(filter(art_full, primer == "coleop"), starts_with("sample"), -sample.sum), na.rm = T)>0)
zn <- sum(colSums(select(filter(art_full, primer == "zeale"), starts_with("sample"), -sample.sum), na.rm = T)>0)

(cse <- csd/sqrt(cn))
(zse <- zsd/sqrt(zn))

# make family-lineage lookup table
lineage_lookup <- art_full %>%
  dplyr::select(phylum, class, order, family, genus, species) %>%
  distinct()

# format MOTU database for use in sp accum 
art_motu_reads <- art_full %>%
  select(primer, id, starts_with("sample")) %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(id, ends_with("coleop"), ends_with("zeale"))

# subset databases that summarize by order, family, genera, species-level IDs
art_order_reads <- art_full %>%
  filter(order != "NA" & is.na(order)==F) %>%
  group_by(primer, order) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(order, ends_with("coleop"), ends_with("zeale"))

art_order_motus <- art_full %>%
  filter(order != "NA" & is.na(order)==F) %>%
  group_by(primer, order) %>%
  summarise(across(starts_with("sample"), function(x) sum(x>0)),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(order, ends_with("coleop"), ends_with("zeale"))

art_fam_reads <- art_full %>%
  filter(family != "NA" & is.na(family)==F) %>%
  group_by(primer, family) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(family, ends_with("coleop"), ends_with("zeale"))

art_fam_motus <- art_full %>%
  filter(family != "NA" & is.na(family)==F) %>%
  group_by(primer, family) %>%
  summarise(across(starts_with("sample"), function(x) sum(x>0)),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(family, ends_with("coleop"), ends_with("zeale"))

art_gen_reads <- art_full %>%
  filter(genus != "NA" & is.na(genus)==F) %>%
  group_by(primer, genus) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(genus, ends_with("coleop"), ends_with("zeale"))

art_gen_motus <- art_full %>%
  filter(genus != "NA" & is.na(genus)==F) %>%
  group_by(primer, genus) %>%
  summarise(across(starts_with("sample"), function(x) sum(x>0)),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(genus, ends_with("coleop"), ends_with("zeale"))

art_sp_reads <- art_full %>%
  filter(species != "NA" & is.na(species)==F) %>% # this part fixes 2 typos in sci
  group_by(primer, species) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(species, ends_with("coleop"), ends_with("zeale"))

#### Format data for accumulation curves

# format data matrices

# spray samples - zeale - motu level
sprayZ_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "Z")) %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no sample for the Zeale primer
  t()
sprayZ_motu_mat[is.na(sprayZ_motu_mat)] <- 0 
sprayZ_motu_mat[sprayZ_motu_mat>0] <- 1 
colnames(sprayZ_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "Z"))$id

# roller samples - zeale - motu level
rollZ_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "Z")) %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_motu_mat[is.na(rollZ_motu_mat)] <- 0 
rollZ_motu_mat[rollZ_motu_mat>0] <- 1 
colnames(rollZ_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "Z"))$id

# both - zeale - sp level
zeromat.motu.Z <- matrix(0,1,645)
sprayZ_motu_mat2 <- rbind(sprayZ_motu_mat[1:2,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[3:9,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[10,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[11:15,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[16,])

bothZ_motu_mat <- sprayZ_motu_mat2 + rollZ_motu_mat
bothZ_motu_mat[bothZ_motu_mat>0] <- 1

# spray samples - coleop - motu level
sprayC_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "C")) %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_motu_mat[is.na(sprayC_motu_mat)] <- 0 
sprayC_motu_mat[sprayC_motu_mat>0] <- 1 
colnames(sprayC_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "C"))$id

# roller samples - coleop - motu level
rollC_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "C")) %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_motu_mat[is.na(rollC_motu_mat)] <- 0 
rollC_motu_mat[rollC_motu_mat>0] <- 1 
colnames(rollC_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "C"))$id

# both - coleop - motu level
zeromat.motu.C <- matrix(0,1,1025)
sprayC_motu_mat2 <- rbind(sprayC_motu_mat[1:10,],
                         zeromat.motu.C,
                         sprayC_motu_mat[11,],
                         zeromat.motu.C,
                         sprayC_motu_mat[12:16,],
                         zeromat.motu.C,
                         sprayC_motu_mat[17,])

bothC_motu_mat <- sprayC_motu_mat2 + rollC_motu_mat
bothC_motu_mat[bothC_motu_mat>0] <- 1

# spray samples - zeale - sp level
sprayZ_sp_mat <- art_sp_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no reads for the Zeale primer
  t()
sprayZ_sp_mat[is.na(sprayZ_sp_mat)] <- 0 
sprayZ_sp_mat[sprayZ_sp_mat>0] <- 1 
colnames(sprayZ_sp_mat) <- art_sp_reads$species

# roller samples - zeale - sp level
rollZ_sp_mat <- art_sp_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_sp_mat[is.na(rollZ_sp_mat)] <- 0 
rollZ_sp_mat[rollZ_sp_mat>0] <- 1 
colnames(rollZ_sp_mat) <- art_sp_reads$species

# both - zeale - sp level
zeromat.sp <- matrix(0,1,270)
sprayZ_sp_mat2 <- rbind(sprayZ_sp_mat[1:2,],
                         zeromat.sp,
                         sprayZ_sp_mat[3:9,],
                         zeromat.sp,
                         sprayZ_sp_mat[10,],
                         zeromat.sp,
                         sprayZ_sp_mat[11:15,],
                         zeromat.sp,
                         sprayZ_sp_mat[16,])

bothZ_sp_mat <- sprayZ_sp_mat2 + rollZ_sp_mat
bothZ_sp_mat[bothZ_sp_mat>0] <- 1

# spray samples - coleop - sp level
sprayC_sp_mat <- art_sp_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_sp_mat[is.na(sprayC_sp_mat)] <- 0 
sprayC_sp_mat[sprayC_sp_mat>0] <- 1 
colnames(sprayC_sp_mat) <- art_sp_reads$species

# roller samples - coleop - sp level
rollC_sp_mat <- art_sp_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_sp_mat[is.na(rollC_sp_mat)] <- 0 
rollC_sp_mat[rollC_sp_mat>0] <- 1 
colnames(rollC_sp_mat) <- art_sp_reads$species

# both - coleop - sp level
sprayC_sp_mat2 <- rbind(sprayC_sp_mat[1:10,],
                         zeromat.sp,
                         sprayC_sp_mat[11,],
                         zeromat.sp,
                         sprayC_sp_mat[12:16,],
                         zeromat.sp,
                         sprayC_sp_mat[17,])

bothC_sp_mat <- sprayC_sp_mat2 + rollC_sp_mat
bothC_sp_mat[bothC_sp_mat>0] <- 1

# pool all primers and sampling methods - species level
bothCZ_sp_mat <- bothC_sp_mat + bothZ_sp_mat
bothCZ_sp_mat[bothCZ_sp_mat>0] <- 1

# pooled roller samples - species level
roll_sp_mat <- rollC_sp_mat + rollZ_sp_mat
roll_sp_mat[roll_sp_mat>0] <- 1

# pooled spray samples - species level
spray_sp_mat <- sprayC_sp_mat2 + sprayZ_sp_mat2
spray_sp_mat[spray_sp_mat>0] <- 1

# spray samples - zeale - genus level
sprayZ_gen_mat <- art_gen_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no reads for the Zeale primer
  t()
sprayZ_gen_mat[is.na(sprayZ_gen_mat)] <- 0 
sprayZ_gen_mat[sprayZ_gen_mat>0] <- 1 
colnames(sprayZ_gen_mat) <- art_gen_reads$genus

# roller samples - zeale - genus level
rollZ_gen_mat <- art_gen_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_gen_mat[is.na(rollZ_gen_mat)] <- 0 
rollZ_gen_mat[rollZ_gen_mat>0] <- 1 
colnames(rollZ_gen_mat) <- art_gen_reads$genus

# both - zeale - genus level
zeromat.gen <- matrix(0,1,354)
sprayZ_gen_mat2 <- rbind(sprayZ_gen_mat[1:2,],
                         zeromat.gen,
                         sprayZ_gen_mat[3:9,],
                         zeromat.gen,
                         sprayZ_gen_mat[10,],
                         zeromat.gen,
                         sprayZ_gen_mat[11:15,],
                         zeromat.gen,
                         sprayZ_gen_mat[16,])

bothZ_gen_mat <- sprayZ_gen_mat2 + rollZ_gen_mat
bothZ_gen_mat[bothZ_gen_mat>0] <- 1

# spray samples - coleop - genus level
sprayC_gen_mat <- art_gen_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_gen_mat[is.na(sprayC_gen_mat)] <- 0 
sprayC_gen_mat[sprayC_gen_mat>0] <- 1 
colnames(sprayC_gen_mat) <- art_gen_reads$genus

# roller samples - coleop - genus level
rollC_gen_mat <- art_gen_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_gen_mat[is.na(rollC_gen_mat)] <- 0 
rollC_gen_mat[rollC_gen_mat>0] <- 1 
colnames(rollC_gen_mat) <- art_gen_reads$genus

# both - coleop - genus level
sprayC_gen_mat2 <- rbind(sprayC_gen_mat[1:10,],
                         zeromat.gen,
                         sprayC_gen_mat[11,],
                         zeromat.gen,
                         sprayC_gen_mat[12:16,],
                         zeromat.gen,
                         sprayC_gen_mat[17,])

bothC_gen_mat <- sprayC_gen_mat2 + rollC_gen_mat
bothC_gen_mat[bothC_gen_mat>0] <- 1

# pool all primers and sampling methods - genus level
bothCZ_gen_mat <- bothC_gen_mat + bothZ_gen_mat
bothCZ_gen_mat[bothCZ_gen_mat>0] <- 1

# pooled roller samples - genus level
roll_gen_mat <- rollC_gen_mat + rollZ_gen_mat
roll_gen_mat[roll_gen_mat>0] <- 1

# pooled spray samples - genus level
spray_gen_mat <- sprayC_gen_mat2 + sprayZ_gen_mat2
spray_gen_mat[spray_gen_mat>0] <- 1

# spray samples - zeale - family level
sprayZ_fam_mat <- art_fam_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no reads for the Zeale primer
  t()
sprayZ_fam_mat[is.na(sprayZ_fam_mat)] <- 0 
sprayZ_fam_mat[sprayZ_fam_mat>0] <- 1 
colnames(sprayZ_fam_mat) <- art_fam_reads$family

# roller samples - zeale primer - family level
rollZ_fam_mat <- art_fam_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_fam_mat[is.na(rollZ_fam_mat)] <- 0 
rollZ_fam_mat[rollZ_fam_mat>0] <- 1 
colnames(rollZ_fam_mat) <- art_fam_reads$family

# both - zeale - family level
zeromat <- matrix(0,1,179)
sprayZ_fam_mat2 <- rbind(sprayZ_fam_mat[1:2,],
                         zeromat,
                         sprayZ_fam_mat[3:9,],
                         zeromat,
                         sprayZ_fam_mat[10,],
                         zeromat,
                         sprayZ_fam_mat[11:15,],
                         zeromat,
                         sprayZ_fam_mat[16,])

bothZ_fam_mat <- sprayZ_fam_mat2 + rollZ_fam_mat
bothZ_fam_mat[bothZ_fam_mat>0] <- 1

# spray samples - coleop - family level
sprayC_fam_mat <- art_fam_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_fam_mat[is.na(sprayC_fam_mat)] <- 0 
sprayC_fam_mat[sprayC_fam_mat>0] <- 1 
colnames(sprayC_fam_mat) <- art_fam_reads$family

# roller samples - coleop primer - family level
rollC_fam_mat <- art_fam_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_fam_mat[is.na(rollC_fam_mat)] <- 0 
rollC_fam_mat[rollC_fam_mat>0] <- 1 
colnames(rollC_fam_mat) <- art_fam_reads$family

# both - coleop - family level
sprayC_fam_mat2 <- rbind(sprayC_fam_mat[1:10,],
                         zeromat,
                         sprayC_fam_mat[11,],
                         zeromat,
                         sprayC_fam_mat[12:16,],
                         zeromat,
                         sprayC_fam_mat[17,])

bothC_fam_mat <- sprayC_fam_mat2 + rollC_fam_mat
bothC_fam_mat[bothC_fam_mat>0] <- 1

# pool all primers and sampling methods - family level
bothCZ_fam_mat <- bothC_fam_mat + bothZ_fam_mat
bothCZ_fam_mat[bothCZ_fam_mat>0] <- 1

# pooled roller samples - family level
roll_fam_mat <- rollC_fam_mat + rollZ_fam_mat
roll_fam_mat[roll_fam_mat>0] <- 1

# pooled spray samples - family level
spray_fam_mat <- sprayC_fam_mat2 + sprayZ_fam_mat2
spray_fam_mat[spray_fam_mat>0] <- 1

# counting families
genct <- art_gen_reads %>% 
  select(genus, sample.sum_coleop, sample.sum_zeale) %>%
  filter(!(is.na(sample.sum_coleop) & is.na(sample.sum_zeale))) %>%
  filter(!((sample.sum_coleop == 0) & (sample.sum_zeale == 0)))

# read counts by unique taxon
readsC = cs3 %>% group_by(phylum, class, order, family, genus, species) %>% summarize(count = sum(count))
readsZ = zs3 %>% group_by(phylum, class, order, family, genus, species) %>% summarize(count = sum(count))

```
# Fig. 1 (old) - Plot relative abundance of orders by method (for each primer set)
Number of motus ID'd to each taxonomic level
```{r}
#################
#### Plotting abundance of each order by method (bar charts)
#### (no. MOTUs)
#################

### Coleop ###

ordersC <- art_full %>%
  filter(!is.na(order), primer == "coleop", sample.sum > 0) %>%
  mutate(idsp = case_when(!is.na(species) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & is.na(species) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0))%>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "16S")

ordersZ <- art_full %>%
  filter(!is.na(order), primer == "zeale", sample.sum > 0) %>%
  mutate(idsp = case_when(!is.na(species) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & is.na(species) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0)) %>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "COI") 

orders_df <- ordersC %>%
  bind_rows(ordersZ) %>%
  mutate(engname = case_when(order == "Mesostigmata" ~ "Mites",
                          order == "Sarcoptiformes" ~ "Mites",
                          order == "Trombidiformes" ~ "Mites",
                          class == "Collembola" ~ "Springtails",
                          class == "Chilopoda" ~ "Centipedes",
                          order == "Araneae" ~ "Spiders",
                          order == "Opiliones" ~ "Harvestmen",
                          order == "Blattodea" ~ "Cockroaches",
                          order == "Coleoptera" ~ "Beetles",
                          order == "Dermaptera" ~ "Earwigs",
                          order == "Diptera" ~ "Flies",
                          order == "Ephemeroptera" ~ "Mayflies",
                          order == "Hemiptera" ~ "True bugs",
                          order == "Hymenoptera" ~ "Bees, wasps, & ants",
                          order == "Lepidoptera" ~ "Butterflies & moths",
                          order == "Mantodea" ~ "Mantids",
                          order == "Megaloptera" ~ "Dobsonflies",
                          order == "Cyclopoida" ~ "Copepods",
                          order == "Neuroptera" ~ "Lacewings",
                          order == "Odonata" ~ "Dragonflies & damselflies",
                          order == "Orthoptera" ~ "Crickets & grasshoppers",
                          order == "Phasmatodea" ~ "Walking sticks",
                          order == "Phthiraptera" ~ "Thrips",
                          order == "Psocoptera" ~ "Bark lice",
                          order == "Siphonaptera" ~ "Fleas",
                          order == "Trichoptera" ~ "Caddisflies"
                          )) %>%
  group_by(class, primer, engname) %>%
  summarize(Order = sum(ord, na.rm = T),
            Family = sum(fam, na.rm = T),
            Genus = sum(gen, na.rm = T),
            Species = sum(sp, na.rm = T),
            .groups = "drop") %>%
  pivot_longer(Order:Species, names_to = "idlevel", values_to = "motus") %>%
  group_by(class, engname) %>%
  mutate(summotu = sum(motus)) %>%
  ungroup() %>%
  arrange(primer, class, engname, idlevel) %>%
  mutate(taxorder = case_when(idlevel == "Species" ~ 1,
                              idlevel == "Genus" ~ 2,
                              idlevel == "Family" ~ 3,
                              idlevel == "Order" ~ 4)) %>%
  mutate(engname = fct_reorder(.f = engname, .x = summotu, .desc = F),
         idlevel = fct_reorder(.f = idlevel, .x = taxorder, .desc = T))

# make plot for less-diverse taxa
library(wesanderson)
(
ldiv_plot <- 
orders_df %>%
    mutate(group = case_when(summotu >= 31 ~ "More",
                             TRUE ~ "Less")) %>%
    filter(group == "Less") %>%
  ggplot() +
  geom_col(aes(y = engname, x = motus,
               fill = idlevel)) +
  facet_wrap(~ primer, ncol = 2) +
  # scale_fill_manual(values = wes_palettes$Zissou1) +
  # scale_color_manual(values = wes_palettes$Zissou1) +
  scale_fill_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  scale_color_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 11), 
        legend.position = c(.89,.22),
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "black")
        ) +
  theme(legend.key.size = unit(.45, 'cm'), #change legend key size
        legend.key.height = unit(.45, 'cm'), #change legend key height
        legend.key.width = unit(.45, 'cm'), #change legend key width
        legend.title = element_text(size=12), #change legend title font size
        legend.text = element_text(size=12)) +
  labs(x = "Number of MOTUs",
       y = "",
       fill = "Lowest \ntaxonomic \nID") #+
  #  guides(fill = "none")
)

# make plot for more-diverse taxa
(
hdiv_plot <- 
orders_df %>%
    mutate(group = case_when(summotu >= 31 ~ "More",
                             TRUE ~ "Less")) %>%
    filter(group == "More") %>%
  ggplot() +
  geom_col(aes(y = engname, x = motus,
               fill = idlevel)) +
  facet_wrap(~ primer, ncol = 2) +
  # scale_fill_manual(values = wes_palettes$Darjeeling1) +
  # scale_color_manual(values = wes_palettes$Darjeeling1) +
  scale_fill_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  scale_color_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 11)#, legend.position = c(.89,.24)
        ) +
  labs(x = "",
       y = "",
       fill = "Lowest \ntaxonomic \nID") +
    guides(fill = "none")
)

library(patchwork)
hdiv_plot / ldiv_plot

ggsave("figures/rev_Fig1_ordermotus_swarm1_1.png", height = 9, width = 7.5, dpi = 400)

```
# Fig. 4 (alternate) - this time including leaves vs. bark
# Plot relative abundance of orders by method (for each primer set)
% of samples detected
```{r}
#################
#### Plotting abundance of each order by method (bar charts)
#### (no. MOTUs)
#################

### Coleop ###

# format data for spray samples (coleop)
ordersCspray <- art_full %>%
  filter(!is.na(order), primer == "coleop") %>%
  select(class, order, family, genus, species, contains(c("CS", "DS"))) %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("sample")), na.rm = T)) %>%
  ungroup() %>%
  mutate(idsp = case_when(!is.na(species) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & is.na(species) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0))%>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "Understory (16S)")

# format data for roller samples (coleop)
ordersCroll <- art_full %>%
  filter(!is.na(order), primer == "coleop") %>%
  select(class, order, family, genus, species, contains(c("CR", "DR"))) %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("sample")), na.rm = T)) %>%
  ungroup() %>%
  mutate(idsp = case_when(!is.na(species) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & is.na(species) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0))%>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "Bark (16S)")

### Zeale ###

# format data for spray samples (zeale)
ordersZspray <- art_full %>%
  filter(!is.na(order), primer == "zeale") %>%
  select(class, order, family, genus, species, contains(c("CS", "DS"))) %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("sample")), na.rm = T)) %>%
  ungroup() %>%
  mutate(idsp = case_when(!is.na(species) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & is.na(species) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0))%>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "Understory (COI)")

# format data for roll samples (zeale)
ordersZroll <- art_full %>%
  filter(!is.na(order), primer == "zeale") %>%
  select(class, order, family, genus, species, contains(c("CR", "DR"))) %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("sample")), na.rm = T)) %>%
  ungroup() %>%
  mutate(idsp = case_when(!is.na(species) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & is.na(species) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0))%>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "Bark (COI)")

# combine MOTU count datasets into one for plotting

orders_df2 <- ordersCspray %>%
  bind_rows(ordersZspray) %>%
  bind_rows(ordersCroll) %>%
  bind_rows(ordersZroll) %>%
  mutate(engname = case_when(order == "Mesostigmata" ~ "Mites",
                          order == "Sarcoptiformes" ~ "Mites",
                          order == "Trombidiformes" ~ "Mites",
                          class == "Collembola" ~ "Springtails",
                          class == "Chilopoda" ~ "Centipedes",
                          order == "Araneae" ~ "Spiders",
                          order == "Opiliones" ~ "Harvestmen",
                          order == "Blattodea" ~ "Cockroaches",
                          order == "Coleoptera" ~ "Beetles",
                          order == "Dermaptera" ~ "Earwigs",
                          order == "Diptera" ~ "Flies",
                          order == "Ephemeroptera" ~ "Mayflies",
                          order == "Hemiptera" ~ "True bugs",
                          order == "Hymenoptera" ~ "Bees, wasps, & ants",
                          order == "Lepidoptera" ~ "Butterflies & moths",
                          order == "Mantodea" ~ "Mantids",
                          order == "Megaloptera" ~ "Dobsonflies",
                          order == "Cyclopoida" ~ "Copepods",
                          order == "Neuroptera" ~ "Lacewings",
                          order == "Odonata" ~ "Dragonflies & damselflies",
                          order == "Orthoptera" ~ "Crickets & grasshoppers",
                          order == "Phasmatodea" ~ "Walking sticks",
                          order == "Phthiraptera" ~ "Thrips",
                          order == "Psocoptera" ~ "Bark lice",
                          order == "Siphonaptera" ~ "Fleas",
                          order == "Trichoptera" ~ "Caddisflies"
                          )) %>%
  group_by(class, primer, engname) %>%
  summarize(Order = sum(ord, na.rm = T),
            Family = sum(fam, na.rm = T),
            Genus = sum(gen, na.rm = T),
            Species = sum(sp, na.rm = T),
            .groups = "drop") %>%
  pivot_longer(Order:Species, names_to = "idlevel", values_to = "motus") %>%
  group_by(class, engname) %>%
  mutate(summotu = sum(motus)) %>%
  ungroup() %>%
  arrange(primer, class, engname, idlevel) %>%
  mutate(taxorder = case_when(idlevel == "Species" ~ 1,
                              idlevel == "Genus" ~ 2,
                              idlevel == "Family" ~ 3,
                              idlevel == "Order" ~ 4),
         primorder = case_when(primer == "Understory (COI)" ~ 1,
                              primer == "Bark (COI)" ~ 2,
                              primer == "Understory (16S)" ~ 3,
                              primer == "Bark (16S)" ~ 4)) %>%
  mutate(engname = fct_reorder(.f = engname, .x = summotu, .desc = F),
         idlevel = fct_reorder(.f = idlevel, .x = taxorder, .desc = T),
         primer = fct_reorder(.f = primer, .x = primorder, .desc = T))

# make plot for less-diverse taxa
library(forcats)
# library(wesanderson)
(
ldiv_plot <- 
orders_df2 %>%
    mutate(group = case_when(summotu >= 50 ~ "More",
                             TRUE ~ "Less")) %>%
    filter(group == "Less") %>%
  ggplot() +
  geom_col(aes(y = engname, x = motus,
               fill = idlevel)) +
  facet_wrap(~ primer, ncol = 4) +
  # scale_fill_manual(values = wes_palettes$Zissou1) +
  # scale_color_manual(values = wes_palettes$Zissou1) +
  scale_fill_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  scale_color_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 11), 
        legend.position = c(.89,.22),
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "black")
        ) +
  theme(legend.key.size = unit(.45, 'cm'), #change legend key size
        legend.key.height = unit(.45, 'cm'), #change legend key height
        legend.key.width = unit(.45, 'cm'), #change legend key width
        legend.title = element_text(size=12), #change legend title font size
        legend.text = element_text(size=12)) +
  labs(x = "Number of MOTUs",
       y = "",
       fill = "Lowest \ntaxonomic \nID") #+
  #  guides(fill = "none")
)

# make plot for more-diverse taxa
(
hdiv_plot <- 
orders_df2 %>%
    mutate(group = case_when(summotu >= 50 ~ "More",
                             TRUE ~ "Less")) %>%
    filter(group == "More") %>%
  ggplot() +
  geom_col(aes(y = engname, x = motus,
               fill = idlevel)) +
  facet_wrap(~ primer, ncol = 4) +
  # scale_fill_manual(values = wes_palettes$Darjeeling1) +
  # scale_color_manual(values = wes_palettes$Darjeeling1) +
  scale_fill_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  scale_color_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 11)#, legend.position = c(.89,.24)
        ) +
  labs(x = "",
       y = "",
       fill = "Lowest \ntaxonomic \nID") +
    guides(fill = "none")
)

library(patchwork)
hdiv_plot / ldiv_plot

# ggsave("figures/rev_Fig1alt_ordermotussub_swarm1_1.png",
#        height = 9, width = 7.5, dpi = 400)

```
# Table 1. composition of MOTUs
```{r}
# note: ordersCspray and similar dfs were created in the preceding chunk
orders_df3 <- ordersCspray %>%
  bind_rows(ordersZspray) %>%
  bind_rows(ordersCroll) %>%
  bind_rows(ordersZroll) %>%
  mutate(engname = case_when(order == "Mesostigmata" ~ "Mites",
                          order == "Sarcoptiformes" ~ "Mites",
                          order == "Trombidiformes" ~ "Mites",
                          class == "Collembola" ~ "Springtails",
                          class == "Chilopoda" ~ "Centipedes",
                          order == "Araneae" ~ "Spiders",
                          order == "Opiliones" ~ "Harvestmen",
                          order == "Blattodea" ~ "Cockroaches",
                          order == "Coleoptera" ~ "Beetles",
                          order == "Dermaptera" ~ "Earwigs",
                          order == "Diptera" ~ "Flies",
                          order == "Ephemeroptera" ~ "Mayflies",
                          order == "Hemiptera" ~ "True bugs",
                          order == "Hymenoptera" ~ "Bees, wasps, & ants",
                          order == "Lepidoptera" ~ "Butterflies & moths",
                          order == "Mantodea" ~ "Mantids",
                          order == "Megaloptera" ~ "Dobsonflies",
                          order == "Cyclopoida" ~ "Copepods",
                          order == "Neuroptera" ~ "Lacewings",
                          order == "Odonata" ~ "Dragonflies & damselflies",
                          order == "Orthoptera" ~ "Crickets & grasshoppers",
                          order == "Phasmatodea" ~ "Walking sticks",
                          order == "Phthiraptera" ~ "Thrips",
                          order == "Psocoptera" ~ "Bark lice",
                          order == "Siphonaptera" ~ "Fleas",
                          order == "Trichoptera" ~ "Caddisflies"
                          )) %>%
  group_by(class, primer, engname) %>%
  summarize(Order = sum(ord, na.rm = T),
            Family = sum(fam, na.rm = T),
            Genus = sum(gen, na.rm = T),
            Species = sum(sp, na.rm = T),
            .groups = "drop") %>%
  rowwise() %>%
  mutate(motus = sum(c_across(Order:Species), na.rm = T)) %>%
  select(-Order, -Family, -Genus, -Species) %>%
  pivot_wider(id_cols = c(class, engname), names_from = primer, values_from = motus) %>%
  select(class, engname, ends_with("Bark (COI)"), ends_with("Understory (COI)"),
         ends_with("Bark (16S)"), ends_with("Understory (16S)")) %>%
  rowwise() %>%
  mutate(csum = sum(c_across(3:6))) %>%
  ungroup() %>%
  arrange(class, desc(csum)) %>%
  select(-csum)

library(knitr)
library(kableExtra)
library(magrittr)

kable(orders_df3, "html") %>%
  kable_styling(bootstrap_options = c("striped", "hover")) %>%
  cat(., file = "figures/table1_motu_tally.html")


```
# Fig. 2 (new) - accumulation curves using iNEXT
```{r}

# format data for iNEXT
accum.data.motu <- list('Understory (COI)' = t(sprayZ_motu_mat),
                 'Bark (COI)' = t(rollZ_motu_mat),
                 'Both substrates (COI)' = t(bothZ_motu_mat),
                 'Understory (16S)' = t(sprayC_motu_mat),
                 'Bark (16S)' = t(rollC_motu_mat),
                 'Both substrates (16S)' = t(bothC_motu_mat))

accum.data.motu.primer <- list('Primer COI' = t(bothZ_motu_mat),
                 'Primer 16S' = t(bothC_motu_mat))

accum.data.motu.sub <- list('Understory (COI)' = t(sprayZ_motu_mat),
                 'Bark (COI)' = t(rollZ_motu_mat),
                 'Understory (16S)' = t(sprayC_motu_mat),
                 'Bark (16S)' = t(rollC_motu_mat))

accum.data.sp.primer <- list(
                 'Primer COI' = t(bothZ_sp_mat),
                 'Primer 16S' = t(bothC_sp_mat),
                 'All pooled' = t(bothCZ_sp_mat))

accum.data.sp.sub <- list(
                 'Bark' = t(roll_sp_mat),
                 'Understory' = t(spray_sp_mat),
                 'All pooled' = t(bothCZ_sp_mat))

accum.data.gen.primer <- list(
                 'Primer COI' = t(bothZ_gen_mat),
                 'Primer 16S' = t(bothC_gen_mat),
                 'All pooled' = t(bothCZ_gen_mat))

accum.data.gen.sub <- list(
                 'Bark' = t(roll_gen_mat),
                 'Understory' = t(spray_gen_mat),
                 'All pooled' = t(bothCZ_gen_mat))

accum.data.fam.primer <- list(
                 'Primer COI' = t(bothZ_fam_mat),
                 'Primer 16S' = t(bothC_fam_mat),
                 'All pooled' = t(bothCZ_fam_mat))

accum.data.fam.sub <- list(
                 'Bark' = t(roll_fam_mat),
                 'Understory' = t(spray_fam_mat),
                 'All pooled' = t(bothCZ_fam_mat))

# run accumulation curve analyses in iNEXT
accum.motu.primer <- iNEXT(accum.data.motu.primer, q=0, datatype="incidence_raw", endpoint = 100)
# accum.motu.primer$iNextEst$coverage_based

accum.motu.sub <- iNEXT(accum.data.motu.sub, q=0, datatype="incidence_raw", endpoint = 100)
# accum.motu$iNextEst$coverage_based

accum.sp.primer <- iNEXT(accum.data.sp.primer, q=0, datatype="incidence_raw", endpoint = 100)
# accum.sp.primer$iNextEst$coverage_based

accum.sp.sub <- iNEXT(accum.data.sp.sub, q=0, datatype="incidence_raw", endpoint = 100)
# accum.sp.sub$iNextEst$coverage_based

accum.gen.primer <- iNEXT(accum.data.gen.primer, q=0, datatype="incidence_raw", endpoint = 100)
# accum.gen.primer$iNextEst$coverage_based

accum.gen.sub <- iNEXT(accum.data.gen.sub, q=0, datatype="incidence_raw", endpoint = 100)
# accum.gen.sub$iNextEst$coverage_based

accum.fam.primer <- iNEXT(accum.data.fam.primer, q=0, datatype="incidence_raw", endpoint = 100)
# accum.fam.primer$iNextEst$coverage_based

accum.fam.sub <- iNEXT(accum.data.fam.sub, q=0, datatype="incidence_raw", endpoint = 100)
# accum.fam.sub$iNextEst$coverage_based

# create plots for motu, sp, genus, family

(motuplot.primer <-
ggiNEXT(accum.motu.primer, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  scale_y_continuous(limits = c(0,2000)) +
  labs(y = "No. MOTUs",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("darkgreen", "darkorange3")) +
    scale_fill_manual(values = c("darkgreen", "darkorange3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        legend.key.height = unit(.07, 'cm'), #change legend key height
        legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(motuplot.sub <-
ggiNEXT(accum.motu.sub, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  scale_y_continuous(limits = c(0,2000)) +
  labs(y = "",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("darkblue", "steelblue", "darkred", "indianred3")) +
    scale_fill_manual(values = c("darkblue", "steelblue", "darkred", "indianred3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        legend.key.height = unit(.07, 'cm'), #change legend key height
        legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(spplot.primer <-
ggiNEXT(accum.sp.primer, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  labs(y = "No. species identified",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("black", "darkgreen", "darkorange3")) +
    scale_fill_manual(values = c("black", "darkgreen", "darkorange3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        legend.key.height = unit(.07, 'cm'), #change legend key height
        legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(spplot.sub <-
ggiNEXT(accum.sp.sub, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  labs(y = "",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("black", "steelblue", "indianred3")) +
    scale_fill_manual(values = c("black", "steelblue", "indianred3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        # legend.key.height = unit(.07, 'cm'), #change legend key height
        # legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(genplot.primer <-
ggiNEXT(accum.gen.primer, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  labs(y = "No. genera identified",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("black", "darkgreen", "darkorange3")) +
    scale_fill_manual(values = c("black", "darkgreen", "darkorange3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        legend.key.height = unit(.07, 'cm'), #change legend key height
        legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(genplot.sub <-
ggiNEXT(accum.gen.sub, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  labs(y = "",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("black", "steelblue", "indianred3")) +
    scale_fill_manual(values = c("black", "steelblue", "indianred3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        # legend.key.height = unit(.07, 'cm'), #change legend key height
        # legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(famplot.primer <-
ggiNEXT(accum.fam.primer, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  labs(y = "No. families identified",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("black", "darkgreen", "darkorange3")) +
    scale_fill_manual(values = c("black", "darkgreen", "darkorange3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        legend.key.height = unit(.07, 'cm'), #change legend key height
        legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

(famplot.sub <-
ggiNEXT(accum.fam.sub, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,80)) +
  labs(y = "",
       x = "No. samples") +
    theme_bw() +
    scale_color_manual(values = c("black", "steelblue", "indianred3")) +
    scale_fill_manual(values = c("black", "steelblue", "indianred3")) +
  theme(text = element_text(size = 14),
        legend.position = c(0.65, 0.1),
        legend.background = element_rect(fill = "transparent")) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.07, 'cm'), #change legend key size
        # legend.key.height = unit(.07, 'cm'), #change legend key height
        # legend.key.width = unit(.07, 'cm'), #change legend key width
        legend.title = element_text(size=7), #change legend title font size
        legend.text = element_text(size=7)) #change legend text font size
)

library(patchwork)
motuplot.primer + motuplot.sub + plot_spacer() + spplot.primer + spplot.sub + 
  genplot.primer + genplot.sub + plot_spacer() + famplot.primer + famplot.sub + 
  plot_layout(ncol = 5, widths = c(0.5, 0.5, 0.15, 0.5, 0.5, 0.5, 0.5, 0.15, 0.5, 0.5))
ggsave("figures/rev_Fig2new_accumcurves.swarm1_15.png", dpi = 400, width = 11, height = 10)


```
# Fig. 2 (old) - construct accumulation curves using iNEXT
```{r}

# format data for iNEXT
accum.data.motu <- list('Understory (COI)' = t(sprayZ_motu_mat),
                 'Bark (COI)' = t(rollZ_motu_mat),
                 'Both substrates (COI)' = t(bothZ_motu_mat),
                 'Understory (16S)' = t(sprayC_motu_mat),
                 'Bark (16S)' = t(rollC_motu_mat),
                 'Both substrates (16S)' = t(bothC_motu_mat))

accum.data.sp <- list('Understory (COI)' = t(sprayZ_sp_mat),
                 'Bark (COI)' = t(rollZ_sp_mat),
                 'Both substrates (COI)' = t(bothZ_sp_mat),
                 'Understory (16S)' = t(sprayC_sp_mat),
                 'Bark (16S)' = t(rollC_sp_mat),
                 'Both substrates (16S)' = t(bothC_sp_mat))

accum.data.sp.pooled <- list(
                 'Both substrates (COI)' = t(bothZ_sp_mat),
                 'Both substrates (16S)' = t(bothC_sp_mat),
                 'Both substrates (both primers)' = t(bothCZ_sp_mat))

accum.data.gen <- list('Understory (COI)' = t(sprayZ_gen_mat),
                 'Bark (COI)' = t(rollZ_gen_mat),
                 'Both substrates (COI)' = t(bothZ_gen_mat),
                 'Understory (16S)' = t(sprayC_gen_mat),
                 'Bark (16S)' = t(rollC_gen_mat),
                 'Both substrates (16S)' = t(bothC_gen_mat))

accum.data.fam <- list('Understory (COI)' = t(sprayZ_fam_mat),
                 'Bark (COI)' = t(rollZ_fam_mat),
                 'Both substrates (COI)' = t(bothZ_fam_mat),
                 'Understory (16S)' = t(sprayC_fam_mat),
                 'Bark (16S)' = t(rollC_fam_mat),
                 'Both substrates (16S)' = t(bothC_fam_mat))

accum.data.motu.primers <- list('Both substrates (COI)' = t(bothZ_motu_mat),
                 'Both substrates (16S)' = t(bothC_motu_mat))

# run accumulation curve analyses in iNEXT
accum.motu <- iNEXT(accum.data.motu, q=0, datatype="incidence_raw", endpoint = 60)
# accum.motu$iNextEst$coverage_based

accum.sp <- iNEXT(accum.data.sp, q=0, datatype="incidence_raw", endpoint = 60)
# accum.sp$iNextEst$coverage_based

accum.gen <- iNEXT(accum.data.gen, q=0, datatype="incidence_raw", endpoint = 60)
# accum.gen$iNextEst$coverage_based

accum.fam <- iNEXT(accum.data.fam, q=0, datatype="incidence_raw", endpoint = 60)
# accum.fam$iNextEst$coverage_based

accum.sp.pooled <- iNEXT(accum.data.sp.pooled, q=0, datatype="incidence_raw", endpoint = 60)
# accum.sp.pooled$iNextEst$coverage_based

# create plots for motu, sp, genus, family
(motuplot <-
ggiNEXT(accum.motu, type=1, se=TRUE, 
        facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. MOTUs") +
    theme_bw() +
  theme(text = element_text(size = 14)) +
  guides(color = "none", shape = "none", fill = "none", linetype = "none")
)

(spplot <-
ggiNEXT(accum.sp, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. species identified") +
    theme_bw() +
  theme(text = element_text(size = 14)) +
  guides(color = "none", shape = "none", fill = "none", linetype = "none")
)

(genplot <-
ggiNEXT(accum.gen, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. genera identified") +
    theme_bw() +
  theme(text = element_text(size = 14)) +
  guides(color = "none", shape = "none", fill = "none", linetype = "none")
)

(famplot <- 
ggiNEXT(accum.fam, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. families identified") +
    theme_bw() +
  theme(text = element_text(size = 15),
        legend.position = c(.78, .19)) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.12, 'cm'), #change legend key size
        legend.key.height = unit(.12, 'cm'), #change legend key height
        legend.key.width = unit(.12, 'cm'), #change legend key width
        legend.title = element_text(size=12), #change legend title font size
        legend.text = element_text(size=12)) #change legend text font size
)

library(patchwork)
(motuplot + spplot) / (genplot + famplot)
# ggsave("figures/rev1_accumcurves.png", width = 12, height = 5, dpi = 400) # 3-panel
ggsave("figures/rev_Fig1_accumcurves.swarm1_2.png", width = 12, height = 10, dpi = 400) # 4-panel

```
# Fig. 3 - Richness, diversity, evenness - by substrate, tree type, primer
# rarefied to 20000 reads (drops 6 samples)
Plots & models of rarefied species richness, shannon & simpson indexes, relative read frequency by primer
```{r}

# MOTU rarefied species richness by sample and primer set
names(art_full) # to get column numbers of matrix
data_cols <- 9:45
coleop_rare_mat <- art_full %>%
  filter(primer == "coleop") %>%
  select(all_of(data_cols)) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("V")), na.rm = T)) %>%
  as.data.frame()
rownames(coleop_rare_mat) <- coleop_rare_mat$rowname

zeale_rare_mat <- art_full %>%
  filter(primer == "zeale") %>%
  select(all_of(data_cols)) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("V")), na.rm = T)) %>%
  as.data.frame()
rownames(zeale_rare_mat) <- zeale_rare_mat$rowname

# get list of samples with low read depth (< 20000)
lowreadsampsC20 <- coleop_rare_mat %>%
  filter(sample.sum < 20000); (lowreadsampsC20 <- lowreadsampsC20$rowname)

lowreadsampsZ20 <- zeale_rare_mat %>%
  filter(sample.sum < 20000); (lowreadsampsZ20 <- lowreadsampsZ20$rowname)

# create versions of the dataset with only samples reaching threshold read depths

# 20000 threshold removes 6 samples ...
coleop_rare_mat20 <- coleop_rare_mat %>%
  filter(rowname %notin% unique(c(lowreadsampsC20, lowreadsampsZ20))) %>%
  select(-rowname)

zeale_rare_mat20 <- zeale_rare_mat %>%
  filter(rowname %notin% unique(c(lowreadsampsC20, lowreadsampsZ20))) %>%
  select(-rowname)

# get lowest number of reads
minreads20 <- min(c(coleop_rare_mat20$sample.sum, zeale_rare_mat20$sample.sum))

rarefy_coleop20 <-rarefy(select(coleop_rare_mat20,-sample.sum), 
                         minreads20, se = T)
rarefy_zeale20 <- rarefy(select(zeale_rare_mat20,-sample.sum), 
                         minreads20, se = T)

rarefy_coleop_df20 <- rarefy_coleop20 %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_coleop20),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "16S")

rarefy_zeale_df20 <- rarefy_zeale20 %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_zeale20),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "COI")

rarefy_df20 <- rarefy_coleop_df20 %>%
  bind_rows(rarefy_zeale_df20) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# now create a rarified version of the read matrixes for diversity metrics

coleop_rare_mat_rare20 <- rrarefy(select(coleop_rare_mat20,-sample.sum), 
                                  minreads20)

zeale_rare_mat_rare20 <- rrarefy(select(zeale_rare_mat20,-sample.sum), 
                                 minreads20)

# Calculate diversity metrics
coleop_rare_shannon20 <- diversity(coleop_rare_mat_rare20, 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare20),.) %>%
  rename(sample = 1, shannon = 2) %>%
  mutate(primer = "16S")

zeale_rare_shannon20 <- diversity(zeale_rare_mat_rare20, 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare20),.) %>%
  rename(sample = 1, shannon = 2) %>%
  mutate(primer = "COI")

shannon_df20 <- coleop_rare_shannon20 %>%
  bind_rows(zeale_rare_shannon20) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# Calculate Simpsons index on rarefied data
coleop_rare_simpson20 <- diversity(coleop_rare_mat_rare20, 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare20),.) %>%
  rename(sample = 1, simpson = 2) %>%
  mutate(primer = "16S")

zeale_rare_simpson20 <- diversity(zeale_rare_mat_rare20, 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(zeale_rare_mat_rare20),.) %>%
  rename(sample = 1, simpson = 2) %>%
  mutate(primer = "COI")

simpson_df20 <- coleop_rare_simpson20 %>%
  bind_rows(zeale_rare_simpson20) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# set font size for all plot annotations below
annfont <- 2
ytextsize = 10
xtextsize = 11

# Richness model
  
library(brms)
# (Smod <- brm(log(S) ~ tree + method + primer, data = rarefy_df20, cores = 4,
#              iter = 4000, warmup = 2000))
# saveRDS(Smod, "output/Smod_swarm1.rds")
(Smod <- readRDS("output/Smod_swarm1.rds"))
# *** = 99% don't overlap; ** = 95% don't overlap; * = 90% don't overlap
#                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
# Intercept            3.67      0.12     3.44     3.90 1.00    10329     5941
# treeDeciduous        0.20      0.12    -0.05     0.43 1.00    10371     6484
# methodUnderstory     0.60      0.12     0.36     0.85 1.00     9017     5652***
# primerCOI           -0.71      0.12    -0.95    -0.48 1.00     9787     6310***


library(tidybayes)
quantile(spread_draws(Smod, b_treeDeciduous)$b_treeDeciduous, c(0.05, 0.95)) 
# -0.01400608  0.38811674 
quantile(spread_draws(Smod, b_methodUnderstory)$b_methodUnderstory, c(0.005, 0.995)) 
# 0.2655983 0.9279090  
quantile(spread_draws(Smod, b_primerCOI)$b_primerCOI, c(0.005, 0.995)) 
# -1.0281707 -0.4169117 

Smod_df <- conditional_effects(Smod, prob = 0.95) %>%
  do.call(rbind, .) %>%
  mutate(var = c("Tree type", "Tree type", "Substrate", "Substrate", "Primer set", "Primer set"),
         effect1__ = case_when(effect1__ == "Coniferous" ~ "Conifer",
                              TRUE ~ effect1__)) %>%
  rename(q97.5 = upper__,
         q2.5 = lower__) %>%
  bind_cols(select(do.call(rbind, conditional_effects(Smod, prob = 0.80)), q10 = lower__, q90 = upper__))

Smod_ann <- data.frame(var = c("Tree type", "Substrate", "Primer set"),
                       x = 1.5,
                       y = 2.25,
                       label = c("", "***", "***"))

(Splot <- 
Smod_df %>%
  ggplot() +
  geom_point(aes(x=effect1__, y = exp(estimate__)),
             size = 3) +
  geom_errorbar(aes(x = effect1__, ymin = exp(q2.5), ymax = exp(q97.5)), width = 0) +
  geom_errorbar(aes(x = effect1__, ymin = exp(q10), ymax = exp(q90)), width = 0, size = 1) +
  geom_text(aes(x = x, y = y, label = label), data = Smod_ann, size = 6) +
  facet_wrap(~var, scales = "free_x") +
  labs(y = "Rarefied MOTU richness", x = "") +
  theme_bw()
)

# Evenness model

  library(brms)
 # (Simpmod <- brm(simpson ~ tree + method + primer, data = simpson_df20, cores = 4,
 #                 iter = 4000, warmup = 2000))
 # saveRDS(Simpmod, "output/Simpmod_swarm1.rds")
(Simpmod <- readRDS("output/Simpmod_swarm1.rds"))
#                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
# Intercept            0.50      0.07     0.36     0.63 1.00    10184     5924
# treeDeciduous        0.09      0.07    -0.05     0.22 1.00    11007     5805
# methodUnderstory     0.13      0.07    -0.01     0.28 1.00    10281     5852*
# primerCOI           -0.13      0.07    -0.26     0.01 1.00    10235     6063*

library(tidybayes)
get_variables(Simpmod)
quantile(spread_draws(Simpmod, b_treeDeciduous)$b_treeDeciduous, c(0.05, 0.95)) 
# -0.02774543  0.20388438
quantile(spread_draws(Simpmod, b_methodUnderstory)$b_methodUnderstory, c(0.05, 0.95))
# 0.01001708 0.25452282   
quantile(spread_draws(Simpmod, b_primerCOI)$b_primerCOI, c(0.05, 0.95))
# -0.240993801 -0.009948897 

Simpmod_df <- conditional_effects(Simpmod, prob = 0.95) %>%
  do.call(rbind, .) %>%
  mutate(var = c("Tree type", "Tree type", "Substrate", "Substrate", "Primer set", "Primer set"),
         effect1__ = case_when(effect1__ == "Coniferous" ~ "Conifer",
                              TRUE ~ effect1__)) %>%
  rename(q97.5 = upper__,
         q2.5 = lower__) %>%
  bind_cols(select(do.call(rbind, conditional_effects(Simpmod, prob = 0.80)), q10 = lower__, q90 = upper__))

Simpmod_ann <- data.frame(var = c("Tree type", "Substrate", "Primer set"),
                       x = 1.5,
                       y = .15,
                       label = c("", "*", "*"))

(Simpplot <-
Simpmod_df %>%
  ggplot() +
  geom_point(aes(x=effect1__, y = estimate__),
             size = 3) +
  geom_errorbar(aes(x = effect1__, ymin = q2.5, ymax = q97.5), width = 0) +
  geom_errorbar(aes(x = effect1__, ymin = q10, ymax = q90), width = 0, size = 1) +
  geom_text(aes(x = x, y = y, label = label), data = Simpmod_ann, size = 6) +
  facet_wrap(~var, scales = "free_x") +
  labs(y = "Rarefied MOTU evenness (Simpson)", x = "") +
  theme_bw()
)
  
# Diversity model
 # library(brms)
 # (Shanmod <- brm(shannon ~ tree + method + primer, data = shannon_df20, cores = 4,
 #                 iter = 4000, warmup = 2000))
# saveRDS(Shanmod, "output/Shanmod_swarm1.rds")
Shanmod <- readRDS("output/Shanmod_swarm1.rds")
#                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
# Intercept            1.17      0.20     0.77     1.57 1.00     9655     5989
# treeDeciduous        0.30      0.22    -0.11     0.71 1.00     9306     6024
# methodUnderstory     0.70      0.22     0.27     1.14 1.00    10610     6467***
# primerCOI           -0.35      0.21    -0.78     0.07 1.00     9983     5895*

library(tidybayes)
get_variables(Shanmod)
quantile(spread_draws(Shanmod, b_treeDeciduous)$b_treeDeciduous, c(0.05, 0.95)) 
# -0.04712158  0.65368848
quantile(spread_draws(Shanmod, b_methodUnderstory)$b_methodUnderstory, c(0.995, 0.005))
# 1.2695077 0.1182835
quantile(spread_draws(Shanmod, b_primerCOI)$b_primerCOI, c(0.05, 0.95))
# -0.7017889342 -0.0009793297 

Shanmod_df <- conditional_effects(Shanmod, prob = 0.95) %>%
  do.call(rbind, .) %>%
  mutate(var = c("Tree type", "Tree type", "Substrate", "Substrate", "Primer set", "Primer set"),
         effect1__ = case_when(effect1__ == "Coniferous" ~ "Conifer",
                              TRUE ~ effect1__)) %>%
  rename(q97.5 = upper__,
         q2.5 = lower__) %>%
  bind_cols(select(do.call(rbind, conditional_effects(Shanmod, prob = 0.80)), q10 = lower__, q90 = upper__))

Shanmod_ann <- data.frame(var = c("Tree type", "Substrate", "Primer set"),
                       x = 1.5,
                       y = .1,
                       label = c("", "***", "*"))

(Shanplot <-
Shanmod_df %>%
  ggplot() +
  geom_point(aes(x=effect1__, y = estimate__),
             size = 3) +
  geom_errorbar(aes(x = effect1__, ymin = q2.5, ymax = q97.5), width = 0) +
  geom_errorbar(aes(x = effect1__, ymin = q10, ymax = q90), width = 0, size = 1) +
  geom_text(aes(x = x, y = y, label = label), data = Shanmod_ann, size = 6) +
  facet_wrap(~var, scales = "free_x") +
  labs(y = "Rarefied MOTU diversity (Shannon)", x = "") +
  theme_bw()
)

library(patchwork)

# plot of marginal effects
gg = gridExtra::arrangeGrob(Shanplot, Splot, Simpplot)

plot(gg)

# ggsave(plot = gg, filename = "figures/rev_Fig3_rarediversity_swarm1_7.png", 
#        height = 9, width = 4, dpi = 400)
```
# Fig. 4A - NMDS COI vs 16S, Spray vs Roller, conifer vs. deciduous 
```{r}
# format data matrices
mds.data.sp <- rbind(sprayZ_sp_mat,
                  rollZ_sp_mat,
                  sprayC_sp_mat,
                  rollC_sp_mat)

mds.data.gen <- rbind(sprayZ_gen_mat,
                  rollZ_gen_mat,
                  sprayC_gen_mat,
                  rollC_gen_mat)

mds.data.fam <- rbind(sprayZ_fam_mat,
                  rollZ_fam_mat,
                  sprayC_fam_mat,
                  rollC_fam_mat)

# check for blank rows
apply(mds.data.sp, 1, max)
apply(mds.data.gen, 1, max)
apply(mds.data.fam, 1, max)

# remove blank rows (no blanks)
  # these result from samples where no MOTUs were ID'd to this taxonomic level
mds.data.sp.nb <- mds.data.sp[-which(rownames(mds.data.sp) %in% 
                                     c("sample.9DR_zeale")),]

mds.data.gen.nb <- mds.data.gen

mds.data.fam.nb <- mds.data.fam
  
# format covariate data - species level
env.data <- data.frame(
  tree = str_sub(gsub(gsub(rownames(mds.data.sp.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                      -2, -2),
  method = str_sub(gsub(gsub(rownames(mds.data.sp.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                        -1, -1)) %>%
  mutate(tree = case_when(tree == "C" ~ "Conifer",
                          tree == "D" ~ "Decid."),
         method = case_when(method == "R" ~ "Bark",
                            method == "S" ~ "Leaves"),
  primer = case_when(grepl(rownames(mds.data.sp.nb), pattern = "zeal") ~ "COI",
                     TRUE ~ "16S"))

# package vegan NMDS
set.seed(110)
# arthNMDS.sp = metaMDS(mds.data.sp.nb,
#                      k=2, distance = 'jaccard', trymax = 10000)
# saveRDS(arthNMDS.sp, "output/arthNMDS.sp.swarm1.rds")
arthNMDS.sp <- readRDS("output/arthNMDS.sp.rds")
arthNMDS.sp # stress: 0.0964

plot1 <- ordiplot(arthNMDS.sp, choices = c(1, 2))

sites.long1 <-
  BiodiversityR::sites.long(plot1, env.data = env.data) %>%
  mutate(axis1 = as.numeric(axis1),
         axis2 = as.numeric(axis2),
         Method = paste0(method, " - ", tree, " (", primer, ")"))
head(sites.long1)
str(sites.long1)

(
  plot.sp <-
    ggplot(sites.long1) +
    geom_vline(
      xintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    geom_hline(
      yintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    scale_x_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_y_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_color_manual(values = c("steelblue", "steelblue", "steelblue", "steelblue", 
                                  "firebrick","firebrick", "firebrick", "firebrick")) +
    scale_fill_manual(values = c("white", "steelblue", "white", "steelblue", 
                                  "white", "firebrick", "white", "firebrick")) +
    scale_shape_manual(values = c(21, 21, 24, 24, 21, 21, 24, 24)) +
    geom_point(
      aes(
        x = axis1,
        y = axis2,
        shape = Method,
        color = Method,
        fill = Method,
      ),
      size = 3,
      alpha = 0.85
    ) +
    annotate("text", x = -1.75, y = -0.67, label = "Stress = 0.096", hjust = 0, size = 3) +
    # coord_fixed(ratio = 1) +
    labs(
      y = "NMDS2",
      x = "NMDS1",
      color = "Substrate",
      shape = "Tree type"
    ) +
    theme_bw() +
    theme(text = element_text(size = 12)) +
    ggtitle("Species") +
   guides(shape = "none", color = "none", fill = "none")
)


# Test for differences among Method and Tree type
# PERMANOVA
set.seed(12555)
perm_arth <- adonis2(mds.data.sp.nb ~ primer + method + tree, data = env.data, 
       permutations = 9999, method = "jaccard",
       strata = NULL, contr.unordered = "contr.sum",
       contr.ordered = "contr.poly")

perm_arth
#          Df SumOfSqs      R2      F Pr(>F)    
# primer    1   3.6424 0.11623 9.5804 0.0001 ***
# method    1   1.3288 0.04240 3.4952 0.0001 ***
# tree      1   0.5141 0.01641 1.3523 0.0626 .  
# Residual 68  25.8529 0.82496                  
# Total    71  31.3382 1.00000             

### genus level ###

# format covariate data - species level

# format covariate data - species level
env.data.gen <- data.frame(
  tree = str_sub(gsub(gsub(rownames(mds.data.gen.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                      -2, -2),
  method = str_sub(gsub(gsub(rownames(mds.data.gen.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                        -1, -1)) %>%
  mutate(tree = case_when(tree == "C" ~ "Conifer",
                          tree == "D" ~ "Decid."),
         method = case_when(method == "R" ~ "Bark",
                            method == "S" ~ "Leaves"),
  primer = case_when(grepl(rownames(mds.data.gen.nb), pattern = "zeal") ~ "COI",
                     TRUE ~ "16S"))

# package vegan NMDS
set.seed(110)
# arthNMDS.gen = metaMDS(mds.data.gen.nb,
#                      k=2, distance = 'jaccard', trymax = 10000)
# saveRDS(arthNMDS.gen, "output/arthNMDS.gen.swarm1.rds")
arthNMDS.gen <- readRDS("output/arthNMDS.gen.swarm1.rds")
arthNMDS.gen # stress = 0.1615

plot1.gen <- ordiplot(arthNMDS.gen, choices = c(1, 2))

sites.long1.gen <-
  BiodiversityR::sites.long(plot1.gen, env.data = env.data.gen) %>%
  mutate(axis1 = as.numeric(axis1),
         axis2 = as.numeric(axis2),
         Method = paste0(method, " - ", tree, " (", primer, ")"))
head(sites.long1.gen)
str(sites.long1.gen)

(
  plot.gen <-
    ggplot(sites.long1.gen) +
    geom_vline(
      xintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    geom_hline(
      yintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    scale_x_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_y_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_color_manual(values = c("steelblue", "steelblue", "steelblue", "steelblue", 
                                  "firebrick","firebrick", "firebrick", "firebrick")) +
    scale_fill_manual(values = c("white", "steelblue", "white", "steelblue", 
                                  "white", "firebrick", "white", "firebrick")) +
    scale_shape_manual(values = c(21, 21, 24, 24, 21, 21, 24, 24)) +
    geom_point(
      aes(
        x = axis1,
        y = axis2,
        shape = Method,
        color = Method,
        fill = Method
      ),
      size = 3,
      alpha = 0.85
    ) +
    annotate("text", x = -2.3, y = -0.7, label = "Stress = 0.161", hjust = 0, size = 3) +
    # coord_fixed(ratio = 1) +
    labs(
      y = "NMDS2",
      x = "NMDS1",
      color = "Substrate",
      shape = "Tree type"
    ) +
    theme_bw() +
    theme(text = element_text(size = 12)) +
    ggtitle("Genus") +
    guides(shape = "none", color = "none", fill = "none")
)

# Test for differences among Method and Tree type (genus level)
# PERMANOVA
set.seed(12555)
perm_arth.gen <- adonis2(mds.data.gen.nb ~ primer + method + tree, data = env.data.gen, 
       permutations = 9999, method = "jaccard",
       strata = NULL, contr.unordered = "contr.sum",
       contr.ordered = "contr.poly")

perm_arth.gen
#          Df SumOfSqs      R2      F Pr(>F)    
# primer    1   2.7396 0.08925 7.2833 0.0001 ***
# method    1   1.5012 0.04890 3.9911 0.0001 ***
# tree      1   0.5020 0.01635 1.3346 0.0457 *  
# Residual 69  25.9541 0.84550                  
# Total    72  30.6970 1.00000   

### family level ###

# format covariate data - family level
env.data.fam <- data.frame(
  tree = str_sub(gsub(gsub(rownames(mds.data.fam.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                      -2, -2),
  method = str_sub(gsub(gsub(rownames(mds.data.fam.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                        -1, -1)) %>%
  mutate(tree = case_when(tree == "C" ~ "Conifer",
                          tree == "D" ~ "Decid."),
         method = case_when(method == "R" ~ "Bark",
                            method == "S" ~ "Leaves"),
  primer = case_when(grepl(rownames(mds.data.fam.nb), pattern = "zeal") ~ "COI",
                     TRUE ~ "16S"))

# package vegan NMDS
set.seed(110)
# arthNMDS.fam = metaMDS(mds.data.fam.nb,
#                      k=2, distance = 'jaccard', trymax = 10000)
# saveRDS(arthNMDS.fam, "output/arthNMDS.fam.swarm1.rds")
arthNMDS.fam <- readRDS("output/arthNMDS.fam.rds")
arthNMDS.fam # stress = 0.202

plot1.fam <- ordiplot(arthNMDS.fam, choices = c(1, 2))

sites.long1.fam <-
  BiodiversityR::sites.long(plot1.fam, env.data = env.data.fam) %>%
  mutate(axis1 = as.numeric(axis1),
         axis2 = as.numeric(axis2),
         Method = paste0(method, " - ", tree, " (", primer, ")"))
head(sites.long1.fam)
str(sites.long1.fam)

(
  plot.fam <-
    ggplot(sites.long1.fam) +
    geom_vline(
      xintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    geom_hline(
      yintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    scale_x_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_y_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_color_manual(values = c("steelblue", "steelblue", "steelblue", "steelblue", 
                                  "firebrick","firebrick", "firebrick", "firebrick")) +
    scale_fill_manual(values = c("white", "steelblue", "white", "steelblue", 
                                  "white", "firebrick", "white", "firebrick")) +
    scale_shape_manual(values = c(21, 21, 24, 24, 21, 21, 24, 24)) +
    geom_point(
      aes(
        x = axis1,
        y = axis2,
        shape = Method,
        color = Method,
        fill = Method
      ),
      size = 3,
      alpha = 0.85
    ) +
    annotate("text", x = -1.7, y = -0.7, label = "Stress = 0.202", hjust = 0, size = 3) +
    # coord_fixed(ratio = 1) +
    labs(
      y = "NMDS2",
      x = "NMDS1",
      color = "Substrate",
      shape = "Tree type"
    ) +
    theme_bw() +
    theme(text = element_text(size = 12)) +
    ggtitle("Family") +
    guides(shape = "none", color = "none", fill = "none")
)

# Test for differences among Method and Tree type
# PERMANOVA
set.seed(12555)
perm_arth.fam <- adonis2(mds.data.fam.nb ~ primer + method + tree, data = env.data.fam, 
       permutations = 9999, method = "jaccard",
       strata = NULL, contr.unordered = "contr.sum",
       contr.ordered = "contr.poly")

perm_arth.fam
#          Df SumOfSqs      R2       F Pr(>F)    
# primer    1   3.2447 0.12936 11.3494 0.0001 ***
# method    1   1.6348 0.06518  5.7183 0.0001 ***
# tree      1   0.4766 0.01900  1.6670 0.0197 *  
# Residual 69  19.7265 0.78646                   
# Total    72  25.0826 1.00000  

library(patchwork)
plot.sp / plot.gen / plot.fam #+ 
  # plot_layout(widths = c(3, 3, 3),
              # heights = c(3,3,3))

# ggsave("figures/rev2_nmds_all_swarm1_1.png", height = 9, width = 3, dpi = 400)
```
# Fig. 4B - Count up taxa by method and make Venn diagram
```{r}

### family

# subdivide 'art' into columns by primer / substrate / tree type for counting
Z_spray <- art_fam_reads %>%
  select(family, contains(c("CS_zeale", "DS_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() 

Z_roll <- art_fam_reads %>%
  select(family, contains(c("CR_zeale", "DR_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() 

C_spray <- art_fam_reads %>%
  select(family, contains(c("CS_coleop", "DS_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() 

C_roll <- art_fam_reads %>%
  select(family, contains(c("CR_coleop", "DR_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() 

# get unique families within each method (change names to get different combos)
# zeale has 115 families (52 of which are not found in coleop)
# coleop has 127 families (64 of which are not found in zeale)
(Z_uni <- Z_spray %>%
    bind_rows(Z_roll) %>%
    distinct() %>%
  filter(family %notin% unique(c(
                           C_spray$family,
                           C_roll$family)))) 

(C_uni <- C_spray %>%
    bind_rows(C_roll) %>%
    distinct() %>%
  filter(family %notin% unique(c(
                           Z_spray$family,
                           Z_roll$family)))) 

# Venn diagram of the 8 primer/substrates
# code from: https://www.datanovia.com/en/blog/venn-diagram-with-r-or-rstudio-a-million-ways/
# devtools::install_github("yanlinlin82/ggvenn")

venn_methods_fam <- list(
  'US\n(COI)' = Z_spray$family,
  'US (16S)' = C_spray$family,
  'Bark (COI)' = Z_roll$family,
  'Bark\n(16S)' = C_roll$family
)

library(ggVennDiagram)
(vennfam <- ggVennDiagram(venn_methods_fam, label_alpha = 0, edge_lty = 1,
              label_color = "black", edge_size = 0.5, set_size = 3, label_size = 3) +
  scale_fill_gradient(low="pink",high = "firebrick") +
  scale_color_manual(values = c(rep("darkgray", 4))) +
  labs(fill = "")
)

# library(ggvenn)
# (vennfam <- ggvenn(
#   venn_methods_fam, 
#   fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF", "#0073C2FF"),
#   # blue, yellow, gray, red
#   stroke_size = 0.6, set_name_size = 3,
#   text_size = 2.25
#   ) + scale_x_continuous(expand = expansion(mult = .2)) #+
#   #ggtitle("Family")
# )

# pairwise overlap (manually counted from Venn diagram)
# leaves-bark: 96 overlapping of 179 total (54% overlap)
unique(c(Z_spray$family, C_spray$family))[unique(c(Z_spray$family, C_spray$family)) %in% unique(c(Z_roll$family, C_roll$family))]
# COI-16S: 63 overlapping of 179 total (35% overlap)
unique(c(Z_spray$family, Z_roll$family))[unique(c(Z_spray$family, Z_roll$family)) %in% unique(c(C_spray$family, C_roll$family))]
# conif-decid: xx overlapping of 179 total (xx% overlap)

### genus

# subdivide 'art' into columns by primer / substrate / tree type for counting
Z_spray.gen <- art_gen_reads %>%
  select(genus, contains(c("CS_zeale", "DS_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 147 for zeale/spray

Z_roll.gen <- art_gen_reads %>%
  select(genus, contains(c("CR_zeale", "DR_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 98 for zeale/roll

C_spray.gen <- art_gen_reads %>%
  select(genus, contains(c("CS_coleop", "DS_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 126 for coleop/spray

C_roll.gen <- art_gen_reads %>%
  select(genus, contains(c("CR_coleop", "DR_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 81 for coleop/roll

# get unique genera within each method (change names to get different combos)
# zeale has 225 genera (177 of which are not found in coleop)
# coleop has 177 genera (129 of which are not found in zeale)
(C_uni.gen <- C_spray.gen %>%
    bind_rows(C_roll.gen) %>%
    distinct() %>%
  filter(genus %notin% unique(c(
                           Z_spray.gen$genus, 
                           Z_roll.gen$genus)))) 

(Z_uni.gen <- Z_spray.gen %>%
    bind_rows(Z_roll.gen) %>%
    distinct() %>%
  filter(genus %notin% unique(c(
                           C_spray.gen$genus, 
                           C_roll.gen$genus)))) 

# Venn diagram of the 8 primer/substrates
# code from: https://www.datanovia.com/en/blog/venn-diagram-with-r-or-rstudio-a-million-ways/
# devtools::install_github("yanlinlin82/ggvenn")

venn_methods_gen <- list(
  'US\n(COI)' = Z_spray.gen$genus,
  'US (16S)' = C_spray.gen$genus,
  'Bark (COI)' = Z_roll.gen$genus,
  'Bark\n(16S)' = C_roll.gen$genus
)

library(ggVennDiagram)
(venngen <- ggVennDiagram(venn_methods_gen, label_alpha = 0, edge_lty = 1,
              label_color = "black", edge_size = 0.5, set_size = 3, label_size = 3) +
  scale_fill_gradient(low="pink",high = "firebrick") +
  scale_color_manual(values = c(rep("darkgray", 4))) +
  labs(fill = "")
)

# library(ggvenn)
# (venngen <- ggvenn(
#   venn_methods_gen, 
#   fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF", "#0073C2FF"),
#   # blue, yellow, gray, red
#   stroke_size = 0.6, set_name_size = 3,
#   text_size = 2.25
#   ) + scale_x_continuous(expand = expansion(mult = .2))#+
#   #ggtitle("Genus")
# )

# pairwise overlap (manually counted from Venn diagram)
# leaves-bark: 111 overlapping of 354 total (31% overlap)
unique(c(Z_spray.gen$genus, C_spray.gen$genus))[unique(c(Z_spray.gen$genus, C_spray.gen$genus)) %in% unique(c(Z_roll.gen$genus, C_roll.gen$genus))]
# COI-16S: 48 overlapping of 354 total (14% overlap)
unique(c(Z_spray.gen$genus, Z_roll.gen$genus))[unique(c(Z_spray.gen$genus, Z_roll.gen$genus)) %in% unique(c(C_roll.gen$genus, C_spray.gen$genus))]
# conif-decid: XX overlapping of XX total (xx% overlap)

### species

# subdivide 'art' into columns by primer / substrate / tree type for counting
Z_spray.sp <- art_sp_reads %>%
  select(species, contains(c("CS_zeale", "DS_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(species) %>%
  distinct() # 143 for zeale/spray

Z_roll.sp <- art_sp_reads %>%
  select(species, contains(c("CR_zeale", "DR_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(species) %>%
  distinct() # 81 for zeale/roll

C_spray.sp <- art_sp_reads %>%
  select(species, contains(c("CS_coleop", "DS_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(species) %>%
  distinct() # 92 for coleop/spray

C_roll.sp <- art_sp_reads %>%
  select(species, contains(c("CR_coleop", "DR_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(species) %>%
  distinct() # 47 for coleop/roll

# get unique genera within each method (change names to get different combos)
# zeale has 186 sp (166 of which are not found in coleop)
# coleop has 104 sp (84 of which are not found in zeale)
# so they only share 20 species in common, zeale has 166 unique sp, coleop has 85 unique sp
(C_uni.sp <- C_spray.sp %>%
    bind_rows(C_roll.sp) %>%
    distinct() %>%
  filter(species %notin% unique(c(
                           Z_spray.sp$species, 
                           Z_roll.sp$species)))) 

(Z_uni.sp <- Z_spray.sp %>%
    bind_rows(Z_roll.sp) %>%
    distinct() %>%
  filter(species %notin% unique(c(
                           C_spray.sp$species, 
                           C_roll.sp$species)))) 

# Venn diagram of the 8 primer/substrates
# code from: https://www.datanovia.com/en/blog/venn-diagram-with-r-or-rstudio-a-million-ways/
# devtools::install_github("yanlinlin82/ggvenn")

venn_methods_sp <- list(
  'US\n(COI)' = Z_spray.sp$species,
  'US (16S)' = C_spray.sp$species,
  'Bark (COI)' = Z_roll.sp$species,
  'Bark\n(16S)' = C_roll.sp$species
)

library(ggVennDiagram)
(vennsp <- ggVennDiagram(venn_methods_sp, label_alpha = 0, edge_lty = 1,
              label_color = "black", edge_size = 0.5, set_size = 3, label_size = 3) +
  scale_fill_gradient(low="pink",high = "firebrick") +
  scale_color_manual(values = c(rep("darkgray", 4))) +
  labs(fill = "")
)

# library(ggvenn)
# (vennsp <- ggvenn(
#   venn_methods_sp, 
#   fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF", "#0073C2FF"),
#   # blue, yellow, gray, red
#   stroke_size = 0.6, set_name_size = 3,
#   text_size = 2.25
#   ) + scale_x_continuous(expand = expansion(mult = .2))#+
#   #ggtitle("Species")
# )

# pairwise overlap (manually counted from Venn diagram)
# leaves-bark: 67 overlapping of 271 total (25% overlap)
unique(c(Z_spray.sp$species, C_spray.sp$species))[unique(c(Z_spray.sp$species, C_spray.sp$species)) %in% unique(c(Z_roll.sp$species, C_roll.sp$species))]
# COI-16S: 20 overlapping of 271 total (7% overlap)
unique(c(Z_spray.sp$species, Z_roll.sp$species))[unique(c(Z_spray.sp$species, Z_roll.sp$species)) %in% unique(c(C_spray.sp$species, C_roll.sp$species))]
# conif-decid: XX overlapping of XX total (xx% overlap)

library(patchwork)

plot.sp + plot_spacer() + vennsp + plot.gen + plot_spacer() + venngen + plot.fam + plot_spacer() + vennfam + 
  plot_layout(ncol = 3, widths = c(1,.1,1,1,.1,1,1,.1,1))

ggsave(filename = "figures/rev_Fig4_nmdsvenn_all_swarm1_10.png",
       height = 10, width = 7, dpi = 400)

# gg = gridExtra::arrangeGrob(plot.sp, vennsp, plot.gen, 
#                             venngen, plot.fam, vennfam)
# 
# plot(gg)

# ggsave(plot = gg, filename = "figures/rev_Fig4_nmdsvenn_all_swarm1_3.png", 
#        height = 10, width = 7, dpi = 400)

# rm(art_spray, art_water, art_roll, art_soil,
#    soil_unique, spray_unique, roll_unique, water_unique,
#    venn_methods)
```
# Fig 5 (part 1) - multinomial models for feeding guild data
https://discourse.mc-stan.org/t/example-with-family-multinomial/8707
https://discourse.mc-stan.org/t/interpret-parameters-from-multinomial-categorical-logit-regression/26374
```{r}
fam_pa <- art_fam_motus[,-1]
# fam_pa[fam_pa > 0] <- 1 # previously used sum of families (pres-absence); now using sum of MOTUs
fam_pa[is.na(fam_pa)] <- 0
fam_pa <- cbind(art_fam_motus[,1], fam_pa)

feed_freq <- fam_pa %>%
  left_join(select(diet, family, diet.cat2), by = join_by(family)) %>%
  select(-family, -starts_with("sample.sum"), -contains("3CS_zeale"))

feed_freq_sum1 <- aggregate(feed_freq[,1:73], 
                           by = list(feed_freq$diet.cat2), sum)

feed_freq_sumC1 <- feed_freq_sum1 %>%
  select(contains("coleop")) %>%
  t()
colnames(feed_freq_sumC1) <- feed_freq_sum1$Group.1

feed_freq_sumZ1 <- feed_freq_sum1 %>%
  select(contains("zeale")) %>%
  t()
colnames(feed_freq_sumZ1) <- feed_freq_sum1$Group.1

feed_freq_sumC <- feed_freq_sumC1 %>%
  as.data.frame() %>%
  mutate(sample = rownames(feed_freq_sumC1),
         tree = stringr::str_sub(sample, -9,-9),
         method = stringr::str_sub(sample, -8, -8),
         primer = "16S") %>%
  rename(mult = 1, myce = 2, para = 3, phyt = 4, pred = 5, scav = 6) %>%
  select(-sample)

feed_freq_sumZ <- feed_freq_sumZ1 %>%
  as.data.frame() %>%
  mutate(sample = rownames(feed_freq_sumZ1),
         tree = stringr::str_sub(sample, -8,-8),
         method = stringr::str_sub(sample, -7, -7),
         primer = "COI") %>%
  rename(mult = 1, myce = 2, para = 3, phyt = 4, pred = 5, scav = 6) %>%
  select(-sample)

feed_freq_sum <- feed_freq_sumC %>%
  bind_rows(feed_freq_sumZ) %>%
  rowwise() %>%
  mutate(scav = sum(c_across(c(scav, myce)))) %>%
  ungroup() %>%
  select(-myce)

rm(feed_freq, feed_freq_sum1, feed_freq_sumC1, feed_freq_sumZ1, feed_freq_sumC, feed_freq_sumZ)

library(brms)

dat <- feed_freq_sum
dat$a1 <- 1
dat$size <- with(dat, mult + para + phyt + pred + scav)
dat$y <- with(dat, cbind(mult, para, phyt, pred, scav))

# # model with primer, method, and tree type
# set.seed(102)
# fit.prim.method.tree <- brm(bf(y | trials(size) ~ primer + method + tree),
#             data = dat, family = multinomial(refcat = "mult"),
#             save_pars = save_pars(all = TRUE), cores = 4, 
#             warmup = 2000, iter = 4000)
# 
# # model with only primer and method
# set.seed(102)
# fit.prim.method <- brm(bf(y | trials(size) ~ primer + method),
#             data = dat, family = multinomial(refcat = "mult"),
#             save_pars = save_pars(all = TRUE), cores = 4, 
#             warmup = 2000, iter = 4000)
# 
# # model with only primer and tree
# set.seed(102)
# fit.prim.tree <- brm(bf(y | trials(size) ~ primer + tree),
#             data = dat, family = multinomial(refcat = "mult"),
#             save_pars = save_pars(all = TRUE), cores = 4, 
#             warmup = 2000, iter = 4000)
# 
# # model with only primer
# set.seed(102)
# fit.prim <- brm(bf(y | trials(size) ~ primer),
#             data = dat, family = multinomial(refcat = "mult"),
#             save_pars = save_pars(all = TRUE), cores = 4,
#             warmup = 2000, iter = 4000)

# # adding LOOIC criterion to 3 models
# fit.prim.method.tree <- add_criterion(fit.prim.method.tree, "loo",
#                                  moment_match = T,
#                           reloo = TRUE)
# fit.prim.method <- add_criterion(fit.prim.method, "loo",
#                                  moment_match = T,
#                           reloo = TRUE)
# fit.prim.tree <- add_criterion(fit.prim.tree, "loo",
#                                  moment_match = T,
#                           reloo = TRUE)
# fit.prim <- add_criterion(fit.prim, "loo",
#                                  moment_match = T,
#                           reloo = TRUE)

# save model output so we don't have to refit each time
# saveRDS(fit.prim.method.tree, "output/feed_multinomial_prim.method.tree_nb_swarm1.rds")
# saveRDS(fit.prim.method, "output/feed_multinomial_prim.method_nb_swarm1.rds")
# saveRDS(fit.prim.tree, "output/feed_multinomial_prim.tree_nb_swarm1.rds")
# saveRDS(fit.prim, "output/feed_multinomial_prim_nb_swarm1.rds")

# load the models so we don't have to refit them each time
fit.prim.method.tree <- readRDS("output/feed_multinomial_prim.method.tree_nb_swarm1.rds")
fit.prim.method <- readRDS("output/feed_multinomial_prim.method.tree_nb_swarm1.rds")
fit.prim.tree <- readRDS("output/feed_multinomial_prim.method.tree_nb_swarm1.rds")
fit.prim <- readRDS("output/feed_multinomial_prim.method.tree_nb_swarm1.rds")

# evaluating models based on LOOIC
brms::loo_compare(fit.prim.method.tree, fit.prim.method, fit.prim.tree, fit.prim,
            criterion = "loo") %>%
  print(simplify = F)

# method is lowest, nearest model is delta elpd = -4 (se = 2.4)
```
# Fig 5 (part 2) - multinomial models for lifemode data
https://discourse.mc-stan.org/t/example-with-family-multinomial/8707
https://discourse.mc-stan.org/t/interpret-parameters-from-multinomial-categorical-logit-regression/26374
```{r}
fam_pa <- art_fam_motus[,-1]
# fam_pa[fam_pa > 0] <- 1 # previously used sum of families (pres-absence); now using sum of MOTUs
fam_pa[is.na(fam_pa)] <- 0
fam_pa <- cbind(art_fam_motus[,1], fam_pa)

life_freq <- fam_pa  %>%
  left_join(select(lifemode, family, life.cat), by = join_by(family)) %>%
  select(-family, -starts_with("sample.sum"), -contains("3CS_zeale"))

life_freq_sum1 <- aggregate(life_freq[,1:73], 
                           by = list(life_freq$life.cat), sum)

life_freq_sumC1 <- life_freq_sum1 %>%
  select(contains("coleop")) %>%
  t()
colnames(life_freq_sumC1) <- life_freq_sum1$Group.1

life_freq_sumZ1 <- life_freq_sum1 %>%
  select(contains("zeale")) %>%
  t()
colnames(life_freq_sumZ1) <- life_freq_sum1$Group.1

life_freq_sumC <- life_freq_sumC1 %>%
  as.data.frame() %>%
  mutate(sample = rownames(life_freq_sumC1),
         tree = stringr::str_sub(sample, -9,-9),
         method = stringr::str_sub(sample, -8, -8),
         primer = "16S") %>%
  rename(aqlar = 1, fullaq = 2, fullter = 3, 
         someaq = 4) %>%
  select(-sample)

life_freq_sumZ <- life_freq_sumZ1 %>%
  as.data.frame() %>%
  mutate(sample = rownames(life_freq_sumZ1),
         tree = stringr::str_sub(sample, -8,-8),
         method = stringr::str_sub(sample, -7, -7),
         primer = "COI") %>%
  rename(aqlar = 1, fullaq = 2, fullter = 3, 
         someaq = 4) %>%
  select(-sample)

# format final data file
life_freq_sum <- life_freq_sumC %>%
  bind_rows(life_freq_sumZ) %>%
  mutate(aqlar = fullaq + aqlar) %>%
  select(-fullaq)

# remove intermediate steps
rm(life_freq, life_freq_sum1, life_freq_sumC1, life_freq_sumZ1, life_freq_sumC, life_freq_sumZ)

library(brms)

dat <- life_freq_sum
dat$size <- with(dat, aqlar + fullter + someaq) # removed fullaq
dat$y <- with(dat, cbind(aqlar, fullter, someaq)) # removed fullaq

# model with primer, method, and tree type
set.seed(102)
life.prim.method.tree <- brms::brm(bf(y | trials(size) ~ primer + method + tree), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# model with primer, method
set.seed(102)
life.prim.method <- brms::brm(bf(y | trials(size) ~ primer + method), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# model with primer, tree type
set.seed(102)
life.prim.tree <- brms::brm(bf(y | trials(size) ~ primer + tree), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# model with just primer
set.seed(102)
life.prim <- brms::brm(bf(y | trials(size) ~ primer), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# adding LOOIC criterion to 3 models
life.prim.method.tree <- brms::add_criterion(life.prim.method.tree, "loo",
                                 moment_match = T,
                          reloo = TRUE)
life.prim.method <- brms::add_criterion(life.prim.method, "loo",
                                 moment_match = T,
                          reloo = TRUE)
life.prim.tree <- brms::add_criterion(life.prim.tree, "loo",
                                 moment_match = T,
                          reloo = TRUE)
life.prim <- brms::add_criterion(life.prim, "loo",
                                 moment_match = T,
                          reloo = TRUE)

# save model output so we don't have to refit each time
saveRDS(life.prim.method.tree, "output/life_multinomial_prim.method.tree_nb_swarm1.rds")
saveRDS(life.prim.method, "output/life_multinomial_prim.method_nb_swarm1.rds")
saveRDS(life.prim.tree, "output/life_multinomial_prim.tree_nb_swarm1.rds")
saveRDS(life.prim, "output/life_multinomial_prim_nb_swarm1.rds")

# load the models so we don't have to refit them each time
life.prim.method.tree <- readRDS("output/life_multinomial_prim.method.tree_nb_swarm1.rds")
life.prim.method <- readRDS("output/life_multinomial_prim.method_nb_swarm1.rds")
life.prim.tree <- readRDS("output/life_multinomial_prim.tree_nb_swarm1.rds")
life.prim <- readRDS("output/life_multinomial_prim_nb_swarm1.rds")

# evaluating models based on LOOIC
brms::loo_compare(life.prim.method.tree, life.prim.method, life.prim.tree, life.prim,
            criterion = "loo") %>%
  print(simplify = F)

# method is lowest, delta elpd to next lowest model = -1 (se 2.4)
```
# Fig 5 (part 3) - multinomial models for size data
https://discourse.mc-stan.org/t/example-with-family-multinomial/8707
https://discourse.mc-stan.org/t/interpret-parameters-from-multinomial-categorical-logit-regression/26374
```{r}
fam_pa <- art_fam_motus[,-1]
# fam_pa[fam_pa > 0] <- 1 # previously used sum of families (pres-absence); now using sum of MOTUs
fam_pa[is.na(fam_pa)] <- 0
fam_pa <- cbind(art_fam_motus[,1], fam_pa)

size_freq <- fam_pa  %>%
  left_join(select(size, family, size.cat = Body_Size), by = join_by(family)) %>%
  select(-family, -starts_with("sample.sum"), -contains("3CS_zeale"))

size_freq_sum1 <- aggregate(size_freq[,1:73], 
                           by = list(size_freq$size.cat), sum)

size_freq_sumC1 <- size_freq_sum1 %>%
  select(contains("coleop")) %>%
  t()
colnames(size_freq_sumC1) <- size_freq_sum1$Group.1

size_freq_sumZ1 <- size_freq_sum1 %>%
  select(contains("zeale")) %>%
  t()
colnames(size_freq_sumZ1) <- size_freq_sum1$Group.1

size_freq_sumC <- size_freq_sumC1 %>%
  as.data.frame() %>%
  mutate(sample = rownames(size_freq_sumC1),
         tree = stringr::str_sub(sample, -9,-9),
         method = stringr::str_sub(sample, -8, -8),
         primer = "16S") %>%
  rename(vsmal = 1, vbig = 2, big = 3, 
         smal = 4) %>%
  select(-sample)

size_freq_sumZ <- size_freq_sumZ1 %>%
  as.data.frame() %>%
  mutate(sample = rownames(size_freq_sumZ1),
         tree = stringr::str_sub(sample, -8,-8),
         method = stringr::str_sub(sample, -7, -7),
         primer = "COI") %>%
  rename(vsmal = 1, vbig = 2, big = 3, 
         smal = 4) %>%
  select(-sample)

# format final data file
size_freq_sum <- size_freq_sumC %>%
  bind_rows(size_freq_sumZ)

# remove intermediate steps
rm(size_freq, size_freq_sum1, size_freq_sumC1, size_freq_sumZ1, size_freq_sumC, size_freq_sumZ)

library(brms)

dat <- size_freq_sum
dat$size <- with(dat, vsmal + vbig + big + smal)
dat$y <- with(dat, cbind(vsmal, vbig, big, smal))

# model with primer, method and tree type
set.seed(102)
size.prim.method.tree <- brms::brm(bf(y | trials(size) ~ primer + method + tree), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# model with primer, method
set.seed(102)
size.prim.method <- brms::brm(bf(y | trials(size) ~ primer + method), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# model with primer, tree type
set.seed(102)
size.prim.tree <- brms::brm(bf(y | trials(size) ~ primer + tree), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# model with just primer
set.seed(102)
size.prim <- brms::brm(bf(y | trials(size) ~ primer), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 4, cores = 4)

# adding LOOIC criterion to 3 models
size.prim.method.tree <- brms::add_criterion(size.prim.method.tree, "loo",
                                 moment_match = T,
                          reloo = TRUE)
size.prim.method <- brms::add_criterion(size.prim.method, "loo",
                                 moment_match = T,
                          reloo = TRUE)
size.prim.tree <- brms::add_criterion(size.prim.tree, "loo",
                                 moment_match = T,
                          reloo = TRUE)
size.prim <- brms::add_criterion(size.prim, "loo",
                                 moment_match = T,
                          reloo = TRUE)

# save model output so we don't have to refit each time
saveRDS(size.prim.method.tree, "output/size_multinomial_prim.method.tree_nb_swarm1.rds")
saveRDS(size.prim.method, "output/size_multinomial_prim.method_nb_swarm1.rds")
saveRDS(size.prim.tree, "output/size_multinomial_prim.tree_nb_swarm1.rds")
saveRDS(size.prim, "output/size_multinomial_prim_nb_swarm1.rds")

# load the models so we don't have to refit them each time
size.prim.method.tree <- readRDS("output/size_multinomial_prim.method.tree_nb_swarm1.rds")
size.prim.method <- readRDS("output/size_multinomial_prim.method_nb_swarm1.rds")
size.prim.tree <- readRDS("output/size_multinomial_prim.tree_nb_swarm1.rds")
size.prim <- readRDS("output/size_multinomial_prim_nb_swarm1.rds")

# evaluating models based on LOOIC
brms::loo_compare(size.prim.method.tree, size.prim.method, size.prim.tree, size.prim,
            criterion = "loo") %>%
  print(simplify = F)

# method is lowest; delta elpd to nearest = -2.6 (2.2 se)
```
# Fig 5 (final) - Multinomial model plots (feeding, life mode, body size)
```{r}
### Feeding guild

# plot settings
dod <- .5
psiz <- 2.5
thin <- 0.3
thik <- 0.75

# load the models so we don't have to refit them each time
fit.prim.method <- readRDS("output/feed_multinomial_prim.method_nb_swarm1.rds")

# plot top model predictions
fit.newdata = expand.grid(method = c("S", "R"),
                          primer = c("COI", "16S"),
                                 size = 1)

fitsies <- fitted(fit.prim.method, 
                  newdata = fit.newdata,probs = c(0.025, 0.1,
                                                  0.9, 0.975))

dim(fitsies )
fitsies # dims are expand grid rows (4), quantiles etc. (6), levels of Y (5)
names(fitsies[1,1,])

fitsies.list <- list()
for(i in 1:dim(fitsies)[3]){
  fitsies.list[[i]] <- fitsies[,,i]
}

fits_df <- do.call(rbind, fitsies.list) %>%
  as.data.frame() %>%
  mutate(method = rep(fit.newdata$method, 5),
         primer = rep(fit.newdata$primer, 5),
         feed = c(rep("Mulitple categories", 4),
                  rep("Parasite/Parasitoid", 4),
                  rep("Phytophagous", 4),
                  rep("Predator", 4),
                  rep("Scavenger", 4))) %>%
  group_by(feed) %>%
  mutate(mean_p = mean(Estimate)) %>%
  ungroup() %>%
  mutate(feed = fct_reorder(feed, mean_p),
         method = case_when(method == "S" ~ "Understory",
                            method == "R" ~ "Bark"))
         
(feed_plot2 <-
fits_df %>%
    mutate(methodprimer = paste0(method, " (", primer, ")")) %>%
  ggplot() +
  geom_errorbar(aes(y = feed, xmin = 100*Q2.5, 
                    xmax = 100*Q97.5, color = methodprimer,
                    group = methodprimer),
                position = position_dodge(width = dod),
                width = 0, size = thin) +  
  geom_errorbar(aes(y = feed, xmin = 100*Q10, 
                    xmax = 100*Q90, color = methodprimer,
                    group = methodprimer),
                size = thik,
                position = position_dodge(width = dod),
                width = 0) +
  geom_point(aes(y = feed, x = 100*Estimate, color = methodprimer,
                 fill = methodprimer,
                 shape = methodprimer),
             position = position_dodge(width = dod),
             size = psiz) +
  scale_color_manual(values = c("steelblue", "steelblue", "firebrick", "firebrick")) +
  scale_fill_manual(values = c("steelblue", "white", "firebrick", "white")) +
  scale_shape_manual(values = c(21,21,21,21)) +
  labs(color = "",
       y = "",
       x = "",
       title = "Feeding guild",
       fill = "Sample type",
       shape = "Sample type") +
  theme_bw() +
  theme(text = element_text(size = 13),
        legend.position = c(.8,.2)) +
    guides(color = "none", shape = "none", fill = "none")
 )

### Body size

# load the models so we don't have to refit them each time
size.prim.method.tree <- readRDS("output/size_multinomial_prim.method_nb_swarm1.rds")

# plot top model predictions
fit.newdata = expand.grid(method = c("S", "R"),
                          primer = c("COI", "16S"),
                                 size = 1)

fitsies <- fitted(size.prim.method, 
                  newdata = fit.newdata,probs = c(0.025, 0.1,
                                                  0.9, 0.975))

dim(fitsies)
fitsies # dims are expand grid rows (4), quantiles etc. (6), levels of Y (4)
names(fitsies[1,1,])

fitsies.list <- list()
for(i in 1:dim(fitsies)[3]){
  fitsies.list[[i]] <- fitsies[,,i]
}

size_fits_df <- do.call(rbind, fitsies.list) %>%
  as.data.frame() %>%
  mutate(method = rep(fit.newdata$method, 4),
         primer = rep(fit.newdata$primer, 4),
         size = c(rep("< 5 mm", 4),
                  rep("> 20 mm", 4),
                  rep("10-20 mm", 4),
                  rep("5-10 mm", 4))) %>%
  group_by(size) %>%
  mutate(mean_p = mean(Estimate)) %>%
  ungroup() %>%
  mutate(size = fct_reorder(size, mean_p),
         method = case_when(method == "S" ~ "Understory",
                            method == "R" ~ "Bark"))
         
(size_plot2 <- 
size_fits_df %>%
    mutate(methodtreeprimer = paste0(method, " (", primer, ")")) %>%
  ggplot() +
  geom_errorbar(aes(y = size, xmin = 100*Q2.5, 
                    xmax = 100*Q97.5, color = methodtreeprimer,
                    group = methodtreeprimer),
                position = position_dodge(width = dod),
                width = 0, size = thin) +  
  geom_errorbar(aes(y = size, xmin = 100*Q10, 
                    xmax = 100*Q90, color = methodtreeprimer,
                    group = methodtreeprimer),
                size = thik,
                position = position_dodge(width = dod),
                width = 0) +
  geom_point(aes(y = size, x = 100*Estimate, color = methodtreeprimer,
                 fill = methodtreeprimer,
                 shape = methodtreeprimer),
             position = position_dodge(width = dod),
             size = psiz) +
  scale_color_manual(values = c("steelblue", "steelblue", "firebrick", "firebrick")) +
  scale_fill_manual(values = c("steelblue", "white", "firebrick", "white")) +
  scale_shape_manual(values = c(21,21,21,21)) +
  labs(color = "",
       y = "",
       x = "",
       title = "Body size",
       fill = "Sample type",
       shape = "Sample type") +
  theme_bw() +
  theme(text = element_text(size = 13),
        legend.position = c(.8,.2)) +
    guides(color = "none", shape = "none", fill = "none")
 )

### Life mode

# load the models so we don't have to refit them each time
life.prim.method <- readRDS("output/life_multinomial_prim.method_nb_swarm1.rds")

# plot top model predictions
fit.newdata = expand.grid(method = c("S", "R"),
                          primer = c("COI", "16S"),
                                 size = 1)

fitsies <- fitted(life.prim.method, 
                  newdata = fit.newdata,probs = c(0.025, 0.1,
                                                  0.9, 0.975))

dim(fitsies)
fitsies # dims are expand grid rows (4), quantiles etc. (6), levels of Y (3)
names(fitsies[1,1,])

fitsies.list <- list()
for(i in 1:dim(fitsies)[3]){
  fitsies.list[[i]] <- fitsies[,,i]
}

life_fits_df <- do.call(rbind, fitsies.list) %>%
  as.data.frame() %>%
  mutate(method = rep(fit.newdata$method, 3),
         primer = rep(fit.newdata$primer, 3),
         life = c(rep("All larvae aquatic", 4),
                  rep("Fully terrestrial", 4),
                  rep("Some larvae aquatic", 4))) %>%
  group_by(life) %>%
  mutate(mean_p = mean(Estimate)) %>%
  ungroup() %>%
  mutate(life = fct_reorder(life, mean_p),
         method = case_when(method == "S" ~ "Understory",
                            method == "R" ~ "Bark"))
         
(life_plot2 <-
life_fits_df %>%
    mutate(methodtreeprimer = paste0(method, " (", primer, ")")) %>%
  ggplot() +
  geom_errorbar(aes(y = life, xmin = 100*Q2.5, 
                    xmax = 100*Q97.5, color = methodtreeprimer,
                    group = methodtreeprimer),
                position = position_dodge(width = dod),
                width = 0, size = thin) +  
  geom_errorbar(aes(y = life, xmin = 100*Q10, 
                    xmax = 100*Q90, color = methodtreeprimer,
                    group = methodtreeprimer),
                size = thik,
                position = position_dodge(width = dod),
                width = 0) +
  geom_point(aes(y = life, x = 100*Estimate, color = methodtreeprimer,
                 fill = methodtreeprimer,
                 shape = methodtreeprimer),
             position = position_dodge(width = dod),
             size = psiz) +
  scale_color_manual(values = c("steelblue", "steelblue", "firebrick", "firebrick")) +
  scale_fill_manual(values = c("steelblue", "white", "firebrick", "white")) +
  scale_shape_manual(values = c(21,21,21,21)) +
  labs(color = "",
       y = "",
       x = "% of MOTUs",
       title = "Life mode",
       fill = "Sample type",
       shape = "Sample type",
       color = "Sample type") +
  theme_bw() +
  theme(text = element_text(size = 13),
        legend.position = c(.75,.15)) +
  guides(color = "none") +
  theme(legend.key.size = unit(.10, 'cm'), #change legend key size
        legend.key.height = unit(.10, 'cm'), #change legend key height
        legend.key.width = unit(.10, 'cm'), #change legend key width
        legend.title = element_text(size=9), #change legend title font size
        legend.text = element_text(size=9), #change legend text font size
        legend.background = element_rect(fill = "transparent"))  
 )

### Combine all 3 plots with patchwork
feed_plot2 / size_plot2 / life_plot2

ggsave("figures/rev_Fig5_guilds_multinom_swarm1_4.jpg",
       width = 5, height = 12, dpi = 400)
```
# Fig. S1 (new) - Richness, diversity, evenness - by substrate, tree type, primer
# rarefied to 40000 reads (drops 11 samples)
Plots & models of rarefied species richness, shannon & simpson indexes, relative read frequency by primer
```{r}

# MOTU rarefied species richness by sample and primer set
names(art_full) # to get column numbers of matrix
data_cols <- 9:45
coleop_rare_mat <- art_full %>%
  filter(primer == "coleop") %>%
  select(all_of(data_cols)) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("V")), na.rm = T)) %>%
  as.data.frame()
rownames(coleop_rare_mat) <- coleop_rare_mat$rowname

zeale_rare_mat <- art_full %>%
  filter(primer == "zeale") %>%
  select(all_of(data_cols)) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("V")), na.rm = T)) %>%
  as.data.frame()
rownames(zeale_rare_mat) <- zeale_rare_mat$rowname

# get list of samples with low read depth (< 20000)
lowreadsampsC40 <- coleop_rare_mat %>%
  filter(sample.sum < 50000); (lowreadsampsC40 <- lowreadsampsC40$rowname)

lowreadsampsZ40 <- zeale_rare_mat %>%
  filter(sample.sum < 50000); (lowreadsampsZ40 <- lowreadsampsZ40$rowname)

# create versions of the dataset with only samples reaching threshold read depths

# 40000 threshold removes 11 samples ...
coleop_rare_mat40 <- coleop_rare_mat %>%
  filter(rowname %notin% unique(c(lowreadsampsC40, lowreadsampsZ40))) %>%
  select(-rowname)

zeale_rare_mat40 <- zeale_rare_mat %>%
  filter(rowname %notin% unique(c(lowreadsampsC40, lowreadsampsZ40))) %>%
  select(-rowname)

# get lowest number of reads
minreads40 <- min(c(coleop_rare_mat40$sample.sum, zeale_rare_mat40$sample.sum))

rarefy_coleop40 <-rarefy(select(coleop_rare_mat40,-sample.sum), 
                         minreads40, se = T)
rarefy_zeale40 <- rarefy(select(zeale_rare_mat40,-sample.sum), 
                         minreads40, se = T)

rarefy_coleop_df40 <- rarefy_coleop40 %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_coleop40),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "16S")

rarefy_zeale_df40 <- rarefy_zeale40 %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_zeale40),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "COI")

rarefy_df40 <- rarefy_coleop_df40 %>%
  bind_rows(rarefy_zeale_df40) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# now create a rarified version of the read matrixes for diversity metrics

coleop_rare_mat_rare40 <- rrarefy(select(coleop_rare_mat40,-sample.sum), 
                                  minreads40)

zeale_rare_mat_rare40 <- rrarefy(select(zeale_rare_mat40,-sample.sum), 
                                 minreads40)

# Calculate diversity metrics
coleop_rare_shannon40 <- diversity(coleop_rare_mat_rare40, 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare40),.) %>%
  rename(sample = 1, shannon = 2) %>%
  mutate(primer = "16S")

zeale_rare_shannon40 <- diversity(zeale_rare_mat_rare40, 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare40),.) %>%
  rename(sample = 1, shannon = 2) %>%
  mutate(primer = "COI")

shannon_df40 <- coleop_rare_shannon40 %>%
  bind_rows(zeale_rare_shannon40) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# Calculate Simpsons index on rarefied data
coleop_rare_simpson40 <- diversity(coleop_rare_mat_rare40, 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare40),.) %>%
  rename(sample = 1, simpson = 2) %>%
  mutate(primer = "16S")

zeale_rare_simpson40 <- diversity(zeale_rare_mat_rare40, 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(zeale_rare_mat_rare40),.) %>%
  rename(sample = 1, simpson = 2) %>%
  mutate(primer = "COI")

simpson_df40 <- coleop_rare_simpson40 %>%
  bind_rows(zeale_rare_simpson40) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# set font size for all plot annotations below
annfont <- 2
ytextsize = 10
xtextsize = 11

# Richness model
  
library(brms)
(Smod40 <- brm(log(S) ~ tree + method + primer, data = rarefy_df40, cores = 4,
               iter = 4000, warmup = 2000))
# saveRDS(Smod40, "output/Smod40_swarm1.rds")
(Smod40 <- readRDS("output/Smod_swarm1.rds"))
# *** = 99% don't overlap; ** = 95% don't overlap; * = 90% don't overlap
#                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
# Intercept            3.65      0.13     3.40     3.90 1.00     9947     5963
# treeDeciduous        0.28      0.13     0.02     0.55 1.00    10201     5463**
# methodUnderstory     0.58      0.13     0.32     0.84 1.00    10677     6433***
# primerCOI           -0.71      0.13    -0.97    -0.46 1.00     9737     5621***

library(tidybayes)
quantile(spread_draws(Smod40, b_treeDeciduous)$b_treeDeciduous, c(0.005, 0.995)) 
# -0.05915236  0.63231089 
quantile(spread_draws(Smod40, b_methodUnderstory)$b_methodUnderstory, c(0.005, 0.995)) 
# 0.2271297 0.9197950  
quantile(spread_draws(Smod40, b_primerCOI)$b_primerCOI, c(0.005, 0.995)) 
# -1.0721585 -0.3672875 

Smod40_df <- conditional_effects(Smod40, prob = 0.95) %>%
  do.call(rbind, .) %>%
  mutate(var = c("Tree type", "Tree type", "Substrate", "Substrate", "Primer set", "Primer set"),
         effect1__ = case_when(effect1__ == "Coniferous" ~ "Conifer",
                              TRUE ~ effect1__)) %>%
  rename(q97.5 = upper__,
         q2.5 = lower__) %>%
  bind_cols(select(do.call(rbind, conditional_effects(Smod40, prob = 0.80)), q10 = lower__, q90 = upper__))

Smod_ann <- data.frame(var = c("Tree type", "Substrate", "Primer set"),
                       x = 1.5,
                       y = 2.25,
                       label = c("**", "***", "***"))

(Splot40 <- 
Smod40_df %>%
  ggplot() +
  geom_point(aes(x=effect1__, y = exp(estimate__)),
             size = 3) +
  geom_errorbar(aes(x = effect1__, ymin = exp(q2.5), ymax = exp(q97.5)), width = 0) +
  geom_errorbar(aes(x = effect1__, ymin = exp(q10), ymax = exp(q90)), width = 0, size = 1) +
  geom_text(aes(x = x, y = y, label = label), data = Smod_ann, size = 6) +
  facet_wrap(~var, scales = "free_x") +
  labs(y = "Rarefied MOTU richness", x = "") +
  theme_bw()
)

# Evenness model

  library(brms)
 (Simpmod40 <- brm(simpson ~ tree + method + primer, data = simpson_df40, cores = 4,
                 iter = 4000, warmup = 2000))
 # saveRDS(Simpmod40, "output/Simpmod40_swarm1.rds")
(Simpmod40 <- readRDS("output/Simpmod40_swarm1.rds"))
#                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
# Intercept            0.43      0.08     0.27     0.58 1.00    10640     6056
# treeDeciduous        0.13      0.08    -0.02     0.29 1.00     9993     6089
# methodUnderstory     0.16      0.08    -0.01     0.32 1.00    10377     6219
# primerCOI           -0.09      0.08    -0.24     0.06 1.00    10391     5909

library(tidybayes)
get_variables(Simpmod40)
quantile(spread_draws(Simpmod40, b_treeDeciduous)$b_treeDeciduous, c(0.05, 0.95)) 
# -0.02626559  0.20343576 
quantile(spread_draws(Simpmod40, b_methodUnderstory)$b_methodUnderstory, c(0.05, 0.95))
# 0.01157486 0.24795090  
quantile(spread_draws(Simpmod40, b_primerCOI)$b_primerCOI, c(0.05, 0.95))
# -0.248345478 -0.005403437 

Simpmod40_df <- conditional_effects(Simpmod40, prob = 0.95) %>%
  do.call(rbind, .) %>%
  mutate(var = c("Tree type", "Tree type", "Substrate", "Substrate", "Primer set", "Primer set"),
         effect1__ = case_when(effect1__ == "Coniferous" ~ "Conifer",
                              TRUE ~ effect1__)) %>%
  rename(q97.5 = upper__,
         q2.5 = lower__) %>%
  bind_cols(select(do.call(rbind, conditional_effects(Simpmod40, prob = 0.80)), q10 = lower__, q90 = upper__))

Simpmod_ann <- data.frame(var = c("Tree type", "Substrate", "Primer set"),
                       x = 1.5,
                       y = .15,
                       label = c("*", "*", ""))

(Simpplot40 <-
Simpmod40_df %>%
  ggplot() +
  geom_point(aes(x=effect1__, y = estimate__),
             size = 3) +
  geom_errorbar(aes(x = effect1__, ymin = q2.5, ymax = q97.5), width = 0) +
  geom_errorbar(aes(x = effect1__, ymin = q10, ymax = q90), width = 0, size = 1) +
  geom_text(aes(x = x, y = y, label = label), data = Simpmod_ann, size = 6) +
  facet_wrap(~var, scales = "free_x") +
  labs(y = "Rarefied MOTU evenness (Simpson)", x = "") +
  theme_bw()
)
  
# Diversity model
 # library(brms)
 # (Shanmod40 <- brm(shannon ~ tree + method + primer, data = shannon_df40, cores = 4,
 #                   iter = 4000, warmup = 2000))
# saveRDS(Shanmod40, "output/Shanmod40_swarm1.rds")
Shanmod40 <- readRDS("output/Shanmod40_swarm1.rds")
#                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
# Intercept            0.93      0.22     0.50     1.37 1.00     9816     6207
# treeDeciduous        0.46      0.23     0.01     0.91 1.00    10341     6130**
# methodUnderstory     0.81      0.24     0.36     1.29 1.00     9930     5668***
# primerCOI           -0.25      0.23    -0.70     0.19 1.00    10341     5407

library(tidybayes)
get_variables(Shanmod40)
quantile(spread_draws(Shanmod40, b_treeDeciduous)$b_treeDeciduous, c(0.005, 0.995)) 
# -0.1657612  1.0925440 
quantile(spread_draws(Shanmod40, b_methodUnderstory)$b_methodUnderstory, c(0.995, 0.005))
# 1.4567435 0.1922574 
quantile(spread_draws(Shanmod40, b_primerCOI)$b_primerCOI, c(0.05, 0.95))
# -0.6288685  0.1213552  

Shanmod40_df <- conditional_effects(Shanmod40, prob = 0.95) %>%
  do.call(rbind, .) %>%
  mutate(var = c("Tree type", "Tree type", "Substrate", "Substrate", "Primer set", "Primer set"),
         effect1__ = case_when(effect1__ == "Coniferous" ~ "Conifer",
                              TRUE ~ effect1__)) %>%
  rename(q97.5 = upper__,
         q2.5 = lower__) %>%
  bind_cols(select(do.call(rbind, conditional_effects(Shanmod40, prob = 0.80)), q10 = lower__, q90 = upper__))

Shanmod_ann <- data.frame(var = c("Tree type", "Substrate", "Primer set"),
                       x = 1.5,
                       y = .1,
                       label = c("**", "***", ""))

(Shanplot40 <-
Shanmod40_df %>%
  ggplot() +
  geom_point(aes(x=effect1__, y = estimate__),
             size = 3) +
  geom_errorbar(aes(x = effect1__, ymin = q2.5, ymax = q97.5), width = 0) +
  geom_errorbar(aes(x = effect1__, ymin = q10, ymax = q90), width = 0, size = 1) +
  geom_text(aes(x = x, y = y, label = label), data = Shanmod_ann, size = 6) +
  facet_wrap(~var, scales = "free_x") +
  labs(y = "Rarefied MOTU diversity (Shannon)", x = "") +
  theme_bw()
)

# plot of marginal effects
gg = gridExtra::arrangeGrob(Shanplot40, Splot40, Simpplot40)

plot(gg)

ggsave(plot = gg, filename = "figures/rev_FigS1_rarediversity_swarm1_1.png", 
       height = 9, width = 4, dpi = 400)
```
# Fig. S1 (old) - Richness, diversity, evenness - by substrate, tree type, primer
# rarefied to 40000 reads (drops 12 samples)
Plots & models of rarefied species richness, shannon & simpson indexes, relative read frequency by primer
```{r}

# MOTU rarefied species richness by sample and primer set
names(art_full) # to get column numbers of matrix
data_cols <- 9:45
coleop_rare_mat <- art_full %>%
  filter(primer == "coleop") %>%
  select(all_of(data_cols)) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("V")), na.rm = T)) %>%
  as.data.frame()
rownames(coleop_rare_mat) <- coleop_rare_mat$rowname

zeale_rare_mat <- art_full %>%
  filter(primer == "zeale") %>%
  select(all_of(data_cols)) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("V")), na.rm = T)) %>%
  as.data.frame()
rownames(zeale_rare_mat) <- zeale_rare_mat$rowname

# get list of samples with low read depth (< 40000)

lowreadsampsC40 <- coleop_rare_mat %>%
  filter(sample.sum < 40000); (lowreadsampsC40 <- lowreadsampsC40$rowname)

lowreadsampsZ40 <- zeale_rare_mat %>%
  filter(sample.sum < 40000); (lowreadsampsZ40 <- lowreadsampsZ40$rowname)

unique(c(lowreadsampsC40, lowreadsampsZ40))

# create versions of the dataset with only samples reaching threshold read depths

# 40000 threshold removes 12 samples ...
coleop_rare_mat40 <- coleop_rare_mat %>%
  filter(rowname %notin% unique(c(lowreadsampsC40, lowreadsampsZ40))) %>%
  select(-rowname)

zeale_rare_mat40 <- zeale_rare_mat %>%
  filter(rowname %notin% unique(c(lowreadsampsC40, lowreadsampsZ40))) %>%
  select(-rowname)

# get lowest number of reads, excluding the 12 samples with < 40,000
minreads40 <- min(c(coleop_rare_mat40$sample.sum, zeale_rare_mat40$sample.sum))

rarefy_coleop40 <-rarefy(select(coleop_rare_mat40,-sample.sum), 
                         minreads40, se = T)
rarefy_zeale40 <- rarefy(select(zeale_rare_mat40,-sample.sum), 
                         minreads40, se = T)

rarefy_coleop_df40 <- rarefy_coleop40 %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_coleop40),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "16S")

rarefy_zeale_df40 <- rarefy_zeale40 %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_zeale40),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "COI")

rarefy_df40 <- rarefy_coleop_df40 %>%
  bind_rows(rarefy_zeale_df40) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

rarefy_df_wide40 <- data.frame(
  sample = colnames(rarefy_coleop40),
  coleop_S = rarefy_coleop40[1,],
  zeale_S = rarefy_zeale40[1,])

# now create a rarified version of the read matrixes for diversity metrics

coleop_rare_mat_rare40 <- rrarefy(select(coleop_rare_mat40,-sample.sum), minreads40)

zeale_rare_mat_rare40 <- rrarefy(select(zeale_rare_mat40,-sample.sum), minreads40)

# Calculate diversity metrics
coleop_rare_shannon40 <- diversity(coleop_rare_mat_rare40, 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare40),.) %>%
  rename(sample = 1, shannon = 2) %>%
  mutate(primer = "16S")

zeale_rare_shannon40 <- diversity(zeale_rare_mat_rare40, 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare40),.) %>%
  rename(sample = 1, shannon = 2) %>%
  mutate(primer = "COI")

shannon_df40 <- coleop_rare_shannon40 %>%
  bind_rows(zeale_rare_shannon40) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# Calculate Simpsons index on rarefied data
coleop_rare_simpson40 <- diversity(coleop_rare_mat_rare40, 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat_rare40),.) %>%
  rename(sample = 1, simpson = 2) %>%
  mutate(primer = "16S")

zeale_rare_simpson40 <- diversity(zeale_rare_mat_rare40, 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(zeale_rare_mat_rare40),.) %>%
  rename(sample = 1, simpson = 2) %>%
  mutate(primer = "COI")

simpson_df40 <- coleop_rare_simpson40 %>%
  bind_rows(zeale_rare_simpson40) %>%
  mutate(method = stringr::str_sub(sample, -1,-1),
         tree = stringr::str_sub(sample, -2,-2)) %>%
  mutate(method = case_when(method == "R" ~ "Bark",
                            TRUE ~ "Understory"),
         tree = case_when(tree == "D" ~ "Deciduous",
                          TRUE ~ "Coniferous"))

# set font size for all plot annotations below
annfont <- 2
ytextsize = 10
xtextsize = 11

# Richness stats
# rarefy_stats <- rarefy_df_wide %>%
#   mutate(lc = log(coleop_S),
#          lz = log(zeale_S)) %>%
#   summarize(mean_lc = mean(lc),
#             sd_lc = sd(lc),
#             n_lc = length(lc),
#             se_lc = sd_lc/sqrt(n_lc),
#             lcl_lc = mean_lc - 2*se_lc,
#             ucl_lc = mean_lc + 2*se_lc,
#             mean_lz = mean(lz),
#             sd_lz = sd(lz),
#             n_lz = length(lz),
#             se_lz = sd_lz/sqrt(n_lz),
#             lcl_lz = mean_lz - 2*se_lz,
#             ucl_lz = mean_lz + 2*se_lz)
# 
# exp(rarefy_stats)

# Richness boxplots
library(wesanderson)
(rareS.plot <- 
rarefy_df40 %>%
  ggplot() + # aes(x = method, y = value)
  geom_violin(draw_quantiles = .5, aes(x = method, y = S, fill = tree),
              alpha = 0.6) +
    facet_wrap(~primer) +
  geom_point(size = 3, alpha = .75, aes(x = method, y = S, color = tree),
             position = position_dodge(width = .9)) +  
    scale_fill_manual(values = wes_palettes$Cavalcanti1[c(2,1)] ) +
  scale_color_manual(values = wes_palettes$Cavalcanti1[c(2,1)]) +
  theme_bw() +
  theme(legend.background = element_rect(fill = "transparent"),
        legend.key.size = unit(.35, 'cm'), #change legend key size
        legend.key.height = unit(.35, 'cm'), #change legend key height
        legend.key.width = unit(.35, 'cm'), #change legend key width
        legend.title = element_text(size=10), #change legend title font size
        legend.text = element_text(size=10)) +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize),
        legend.position = c(0.85, 0.81)) +
  labs(x = "",
       y = "Rarefied MOTU richness",
       fill = "",
       color = "") 
  )
  
  
summary(lm(log(S) ~ tree + method + primer, data = rarefy_df40))
  
# Evenness stats
# simpson_stats <- simpson_df40 %>%
#   mutate(lc = coleop_simpson,
#          lz = zeale_simpson) %>%
#   summarize(mean_lc = mean(lc),
#             sd_lc = sd(lc),
#             n_lc = length(lc),
#             se_lc = sd_lc/sqrt(n_lc),
#             lcl_lc = mean_lc - 2*se_lc,
#             ucl_lc = mean_lc + 2*se_lc,
#             mean_lz = mean(lz),
#             sd_lz = sd(lz),
#             n_lz = length(lz),
#             se_lz = sd_lz/sqrt(n_lz),
#             lcl_lz = mean_lz - 2*se_lz,
#             ucl_lz = mean_lz + 2*se_lz)
# 
# simpson_stats

# Evenness boxplots
(rareSimpson.plot <- 
simpson_df40 %>%
  ggplot() + # aes(x = method, y = value)
  geom_violin(draw_quantiles = .5, aes(x = method, y = simpson, fill = tree),
              alpha = 0.6) +
    facet_wrap(~primer) +
  geom_point(size = 3, alpha = .75, aes(x = method, y = simpson, color = tree),
             position = position_dodge(width = .9)) +  
    scale_fill_manual(values = wes_palettes$Cavalcanti1[c(2,1)] ) +
  scale_color_manual(values = wes_palettes$Cavalcanti1[c(2,1)]) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize),
        legend.position = c(0.85, 0.86)) +
  labs(x = "",
       y = "MOTU evenness (Simpson)",
       fill = "",
       color = "") +
    guides(fill = "none", color = "none")
  )

summary(lm(simpson ~ tree + method + primer, data = simpson_df40))

# Diversity stats
# shannon_stats <- shannon_df %>%
#   mutate(lc = coleop_shannon,
#          lz = zeale_shannon) %>%
#   summarize(mean_lc = mean(lc),
#             sd_lc = sd(lc),
#             n_lc = length(lc),
#             se_lc = sd_lc/sqrt(n_lc),
#             lcl_lc = mean_lc - 2*se_lc,
#             ucl_lc = mean_lc + 2*se_lc,
#             mean_lz = mean(lz),
#             sd_lz = sd(lz),
#             n_lz = length(lz),
#             se_lz = sd_lz/sqrt(n_lz),
#             lcl_lz = mean_lz - 2*se_lz,
#             ucl_lz = mean_lz + 2*se_lz)
# 
# shannon_stats

# Diversity boxplots
(rareShannon.plot <- 
shannon_df40 %>%
  ggplot() + # aes(x = method, y = value)
  geom_violin(draw_quantiles = .5, aes(x = method, y = shannon, fill = tree),
              alpha = 0.6) +
    facet_wrap(~primer) +
  geom_point(size = 3, alpha = .75, aes(x = method, y = shannon, color = tree),
             position = position_dodge(width = .9)) +  
    scale_fill_manual(values = wes_palettes$Cavalcanti1[c(2,1)] ) +
  scale_color_manual(values = wes_palettes$Cavalcanti1[c(2,1)]) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize),
        legend.position = c(0.85, 0.86)) +
  labs(x = "",
       y = "MOTU diversity (Shannon)",
       fill = "",
       color = "") +
    guides(fill = "none", color = "none")
  )

summary(lm(shannon ~ tree + method + primer, data = shannon_df40))

library(patchwork)

rareS.plot / rareSimpson.plot / rareShannon.plot

ggsave("figures/rev_FigS1_rarefieddiv_40k.png", width = 5, height = 7, dpi = 400)
```
# Examining reference databases
```{r}

library(data.table)

cref <- fread("data/ncbi.bold.art.hum.mus.may23.pcr.pga.tax.uni.cln.C.tsv") %>%
  mutate(len = nchar(sequence))

zref <- fread("data/ncbi.bold.art.hum.mus.may23.pcr.pga.tax.uni.cln.Z.tsv") %>%
  mutate(len = nchar(sequence))

```
# summarize read counts by primer
```{r}

orderreadsC = art_order_reads %>% rowwise() %>% mutate(sumC = sum(c_across(ends_with("coleop")), na.rm = T)) %>% select(order, sumC)

ordersC$pct = ordersC$motus / sum(ordersC$motus)
ordersZ$pct = ordersZ$motus / sum(ordersZ$motus)

```











...old below here...










# Feeding guild - fine scale
```{r}
feed_step1 <- diet %>%
  select(family, diet.cat) %>%
  left_join(lineage_lookup) %>%
  right_join(n_otus_max) %>%
  group_by(diet.cat) %>%
  # counts number of families with > 0 reads in each sample
  summarize(across(starts_with("sample"), 
                   .fns = function(x) sum(x > 0)),
            .groups = "drop")

feedcats <- feed_step1$diet.cat
feed <- feed_step1 %>%
  select(starts_with("sample")) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  mutate(sample_id = colnames(feed_step1)[2:41],
         method = str_sub(sample_id, -1,-1),
         tree = str_sub(sample_id, -2,-2)) %>%
  group_by(method, tree) %>%
  # calculates mean number of families in samples of each group
  summarize(across(contains("V"), mean),
            .groups = "drop")
colnames(feed) <- c("method", "tree", feedcats)
rm(feed_step1, feedcats)

feed_df <- feed %>% 
  pivot_longer(., cols = "detritophagous/scavenger":"zoophagous-predator") %>%
  mutate(diet.cat = case_when(name == 
                                "detritophagous/scavenger" ~ 
                                "Scavenger",
                              name == 
                                "phytophagous-galls/miners" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-multiple/other" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-phyllophagous" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-sap sucking" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-saproxylic" ~
                                "Phytophagous",
                              name == 
                                "zoophagous-parasite/parasitoid" ~
                                "Parasite/parasitoid",
                              name == 
                                "zoophagous-predator" ~
                                "Predator",
                              name == 
                                "multiple categories" ~
                                "Mulitple categories",
                              name == 
                                "mycetophagous" ~
                                "Mycetophagous",
                              TRUE ~ name),
         tree = case_when(tree == "C" ~ "Coniferous",
                          tree == "D" ~ "Deciduous"),
         method = case_when(method == "R" ~ "Roller",
                          method == "S" ~ "Spray"),
         order = case_when(diet.cat == "Predator" ~ 1,
                           diet.cat == "Parasite/parasitoid" ~ 2,
                           diet.cat == "Phytophagous" ~ 3,
                           diet.cat == "Scavenger" ~ 4,
                           diet.cat == "Mycetophagous" ~ 5,
                           diet.cat == "Multiple categories" ~ 6),
         diet.cat = fct_reorder(diet.cat, order)
         )
  
(feed_plot <- 
  ggplot(data = feed_df, 
       aes(x = method, y = value, fill = name)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Feeding Guild") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotG_feeding_guild_condensed.jpg", 
#        width = 6, height = 4, dpi = 400)

(feed_plot_condensed <- 
  ggplot(data = feed_df, 
       aes(x = method, y = value, fill = diet.cat)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Feeding Guild") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotH_feeding_guild_condensed.jpg", 
#        width = 6, height = 4, dpi = 400)

# Habitat / life mode
life_step1 <- lifemode %>%
  select(family, life.cat) %>%
  left_join(lineage_lookup) %>%
  right_join(n_otus_max) %>%
  group_by(life.cat) %>%
  summarize(across(starts_with("sample"), 
                   .fns = function(x) sum(x > 0)),
            .groups = "drop")

lifecats <- life_step1$life.cat
life <- life_step1 %>%
  select(starts_with("sample")) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  mutate(sample_id = colnames(life_step1)[2:41],
         method = str_sub(sample_id, -1,-1),
         tree = str_sub(sample_id, -2,-2)) %>%
  group_by(method, tree) %>%
  summarize(across(contains("V"), mean),
            .groups = "drop")
colnames(life) <- c("method", "tree", lifecats)
rm(life_step1, lifecats)

life_df <- life %>% 
  pivot_longer(., cols = starts_with("All"):starts_with("Some")) %>%
  mutate(tree = case_when(tree == "C" ~ "Coniferous",
                          tree == "D" ~ "Deciduous"),
         method = case_when(method == "R" ~ "Roller",
                          method == "S" ~ "Spray"),
         order = case_when(name == "Fully aquatic" ~ 4,
                           name == "Fully terrestrial" ~ 1,
                           name == "All larvae aquatic" ~ 3,
                           name == "Some larvae aquatic" ~ 2),
         name = fct_reorder(name, order, .desc = F))
  
(life_plot <- 
  ggplot(data = life_df, 
       aes(x = method, y = value, fill = name)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Life mode") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotI_life_mode.jpg",
#        width = 6, height = 4, dpi = 400)

# Body size
size_step1 <- size %>%
  select(family, size.cat = Body_Size, order = Body_Size_order) %>%
  left_join(lineage_lookup) %>%
  right_join(n_otus_max) %>%
  group_by(size.cat) %>%
  summarize(across(starts_with("sample"), 
                   .fns = function(x) sum(x > 0)),
            .groups = "drop")

sizecats <- size_step1$size.cat
size_step2 <- size_step1 %>%
  select(starts_with("sample")) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  mutate(sample_id = colnames(size_step1)[2:41],
         method = str_sub(sample_id, -1,-1),
         tree = str_sub(sample_id, -2,-2)) %>%
  group_by(method, tree) %>%
  summarize(across(contains("V"), mean),
            .groups = "drop")
colnames(size_step2) <- c("method", "tree", sizecats)
rm(size_step1, sizecats)

size_df <- size_step2 %>% 
  pivot_longer(., cols = 3:6) %>%
  mutate(tree = case_when(tree == "C" ~ "Coniferous",
                          tree == "D" ~ "Deciduous"),
         method = case_when(method == "R" ~ "Roller",
                          method == "S" ~ "Spray")) %>%
  left_join(distinct(select(size, name = Body_Size, 
                            order = Body_Size_order))) %>%
  mutate(name = fct_reorder(name, order, .desc = F))
  
(size_plot <- 
  ggplot(data = size_df, 
       aes(x = method, y = value, fill = name)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Body size") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotJ_body_size.jpg", 
#        width = 6, height = 4, dpi = 400)

#################
##### combine feeding guild, life mode, and body size plots
#################

feed_plot_condensed / life_plot / size_plot

ggsave("figures/PlotK_guilds_bar.jpg", 
       width = 6, height = 12, dpi = 400)

```
# Plots of rarefied species richness, shannon & simpson indexes, relative read frequency by primer
```{r}
# MOTU rarefied species richness by sample and primer set
names(art_order) # to get column numbers of matrix
data_cols <- 8:44
coleop_rare_mat <- art_full %>%
  filter(primer == "coleop") %>%
  select(data_cols) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame()

zeale_rare_mat <- art_order %>%
  filter(primer == "zeale") %>%
  select(data_cols) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame()

apply(coleop_rare_mat, 1, sum)
sort(apply(coleop_rare_mat, 1, sum))
apply(zeale_rare_mat, 1, sum)
sort(apply(zeale_rare_mat, 1, sum))

# get lowest number of reads, excluding the 1 sample with only 195
minreads <- 3660

rarefy_coleop <- rarefy(coleop_rare_mat[-9,], minreads, se = T)
rarefy_zeale <- rarefy(zeale_rare_mat[-9,], minreads, se = T)

rarefy_coleop_df <- rarefy_coleop %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_coleop),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "Coleop")

rarefy_zeale_df <- rarefy_zeale %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_zeale),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "Zeale")

rarefy_df <- rarefy_coleop_df %>%
  bind_rows(rarefy_zeale_df)

rarefy_df_wide <- data.frame(
  sample = colnames(rarefy_coleop),
  coleop_S = rarefy_coleop[1,],
  zeale_S = rarefy_zeale[1,],
  coleop_se = rarefy_coleop[2,],
  zeale_se = rarefy_zeale[2,])

# now create a rarified version of the read matrixes for diversity metrics

coleop_rare_mat_rare <- rrarefy(coleop_rare_mat[-9,], minreads)
zeale_rare_mat_rare <- rrarefy(zeale_rare_mat[-9,], minreads)

# Calculate diversity metrics
coleop_rare_shannon <- diversity(coleop_rare_mat[-9,], 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat[-9,]),.) %>%
  rename(sample = 1, coleop_shannon = 2)

zeale_rare_shannon <- diversity(zeale_rare_mat[-9,], 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat[-9,]),.) %>%
  rename(sample = 1, zeale_shannon = 2)

shannon_df <- coleop_rare_shannon %>%
  bind_cols(select(zeale_rare_shannon, -sample))

# Calculate Simpsons index on rarefied data
coleop_rare_simpson <- diversity(coleop_rare_mat[-9,], 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat[-9,]),.) %>%
  rename(sample = 1, coleop_simpson = 2)

zeale_rare_simpson <- diversity(zeale_rare_mat[-9,], 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(zeale_rare_mat[-9,]),.) %>%
  rename(sample = 1, zeale_simpson = 2)

simpson_df <- coleop_rare_simpson %>%
  bind_cols(select(zeale_rare_simpson, -sample))

# set font size for all plot annotations below
annfont <- 2
ytextsize = 10
xtextsize = 11

# Richness boxplots
rarefy_stats <- rarefy_df_wide %>%
  mutate(lc = log(coleop_S),
         lz = log(zeale_S)) %>%
  summarize(mean_lc = mean(lc),
            sd_lc = sd(lc),
            n_lc = length(lc),
            se_lc = sd_lc/sqrt(n_lc),
            lcl_lc = mean_lc - 2*se_lc,
            ucl_lc = mean_lc + 2*se_lc,
            mean_lz = mean(lz),
            sd_lz = sd(lz),
            n_lz = length(lz),
            se_lz = sd_lz/sqrt(n_lz),
            lcl_lz = mean_lz - 2*se_lz,
            ucl_lz = mean_lz + 2*se_lz)

exp(rarefy_stats)

(rareS.plot <- 
rarefy_df_wide %>%
  select(-ends_with("_se")) %>%
  tidyr::pivot_longer(cols = ends_with("_S")) %>%
  mutate(name = case_when(name == "coleop_S" ~ "Coleop",
                          name == "zeale_S" ~ "Zeale")) %>%
  ggplot(aes(x = name, y = value)) +
  geom_line(size = .5, alpha = .5, 
            aes(group = sample), 
            color = "darkgray") +
  geom_violin(draw_quantiles = .5, aes(fill = name),
              alpha = 0.6) +
  geom_point(size = 4, alpha = .5, aes(color = name)) +  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize)) +
  labs(x = "",
       y = "Rarefied MOTU richness") +
  guides(fill = "none", color = "none") +
  annotate("text", x = 2.5, y = 160, 
           label = paste0("Means (log-normal)",
           "\nColeop: ",
           round(exp(rarefy_stats$mean_lc),1), 
           ", 95% CI = [",
           round(exp(rarefy_stats$lcl_lc),1),
           ", ",
           round(exp(rarefy_stats$ucl_lc),1), 
           "]\n",
           "Zeale: ",
           round(exp(rarefy_stats$mean_lz),1),
           ", 95% CI = [",
           round(exp(rarefy_stats$lcl_lz),1),
           ", ",
           round(exp(rarefy_stats$ucl_lz),1),
           "]"),
           hjust = 1,
           size = annfont,
           color = "darkgray")
)
  
# Evenness boxplots
simpson_stats <- simpson_df %>%
  mutate(lc = coleop_simpson,
         lz = zeale_simpson) %>%
  summarize(mean_lc = mean(lc),
            sd_lc = sd(lc),
            n_lc = length(lc),
            se_lc = sd_lc/sqrt(n_lc),
            lcl_lc = mean_lc - 2*se_lc,
            ucl_lc = mean_lc + 2*se_lc,
            mean_lz = mean(lz),
            sd_lz = sd(lz),
            n_lz = length(lz),
            se_lz = sd_lz/sqrt(n_lz),
            lcl_lz = mean_lz - 2*se_lz,
            ucl_lz = mean_lz + 2*se_lz)

simpson_stats

(simpson.plot <- 
simpson_df %>%
  tidyr::pivot_longer(cols = ends_with("_simpson")) %>%
  mutate(name = case_when(name == "coleop_simpson" ~ "Coleop",
                          name == "zeale_simpson" ~ "Zeale")) %>%
  ggplot(aes(x = name, y = value)) +
  geom_line(size = .5, alpha = .5, 
            aes(group = sample), 
            color = "darkgray") +
  geom_violin(draw_quantiles = .5, aes(fill = name),
              alpha = 0.6) +
  geom_point(size = 4, alpha = .5, aes(color = name)) +  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize)) +
  labs(x = "",
       y = "MOTU evenness (Simpson)") +
  guides(fill = "none", color = "none") +
  annotate("text", x = 2.5, y = 1.3,
           label = paste0("Means",
           "\nColeop: ",
           round(simpson_stats$mean_lc,2),
           ", 95% CI = [",
           round(simpson_stats$lcl_lc,2),
           ", ",
           round(simpson_stats$ucl_lc,2),
           "]\n",
           "Zeale: ",
           round(simpson_stats$mean_lz,2),
           ", 95% CI = [",
           round(simpson_stats$lcl_lz,2),
           ", ",
           round(simpson_stats$ucl_lz,2),
           "]"),
           hjust = 1,
           vjust = 1,
           size = annfont,
           color = "darkgray")
)
  
# Diversity boxplots
shannon_stats <- shannon_df %>%
  mutate(lc = coleop_shannon,
         lz = zeale_shannon) %>%
  summarize(mean_lc = mean(lc),
            sd_lc = sd(lc),
            n_lc = length(lc),
            se_lc = sd_lc/sqrt(n_lc),
            lcl_lc = mean_lc - 2*se_lc,
            ucl_lc = mean_lc + 2*se_lc,
            mean_lz = mean(lz),
            sd_lz = sd(lz),
            n_lz = length(lz),
            se_lz = sd_lz/sqrt(n_lz),
            lcl_lz = mean_lz - 2*se_lz,
            ucl_lz = mean_lz + 2*se_lz)

shannon_stats

(shannon.plot <- 
shannon_df %>%
  tidyr::pivot_longer(cols = ends_with("_shannon")) %>%
  mutate(name = case_when(name == "coleop_shannon" ~ "Coleop",
                          name == "zeale_shannon" ~ "Zeale")) %>%
  ggplot(aes(x = name, y = value)) +
  geom_line(size = .5, alpha = .5, 
            aes(group = sample), 
            color = "darkgray") +
  geom_violin(draw_quantiles = .5, aes(fill = name),
              alpha = 0.6) +
  geom_point(size = 4, alpha = .5, aes(color = name)) +  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize)) +
  labs(x = "",
       y = "MOTU diversity (Shannon)") +
  guides(fill = "none", color = "none") +
  annotate("text", x = 2.5, y = 4.8,
           label = paste0("Means",
           "\nColeop: ",
           round(shannon_stats$mean_lc,2),
           ", 95% CI = [",
           round(shannon_stats$lcl_lc,2),
           ", ",
           round(shannon_stats$ucl_lc,2),
           "]\n",
           "Zeale: ",
           round(shannon_stats$mean_lz,2),
           ", 95% CI = [",
           round(shannon_stats$lcl_lz,2),
           ", ",
           round(shannon_stats$ucl_lz,2),
           "]"),
           hjust = 1,
           vjust = 1,
           size = annfont,
           color = "darkgray")
)

#################
#### Plotting relative read abundance by order (bar chart)
#################

# set up plotting order df
order_df <- data.frame(cat = unique(paste0(art_full$class, "-", 
                                      art_full$order)),
           order = c(2,4,1,3,6,15,9,7,10,12,8,23,
                     14,13,11,22,16,24,5,20,19,17,21,18),
           name = c("Spiders",
                    "Unknown Arachnids",
                    "Sarcoptiform mites",
                    "Trombidiform mites",
                    "Collembola",
                    "Cockroaches",
                    "Beetles",
                    "Flies",
                    "True bugs",
                    "Bees, wasps, & ants",
                    "Butterflies & moths",
                    "Unknown insects",
                    "Lacewings",
                    "Crickets & grasshoppers",
                    "Bark lice",
                    "Fleas",
                    "Caddisflies",
                    "Unknown arthropods",
                    "Centipedes",
                    "Mayflies",
                    "Mantids",
                    "Dragonflies & damselflies",
                    "Walking sticks",
                    "Thrips"))
                    

names(art_full)
primer_who <- art_full %>%
  mutate(cat = paste0(class, "-", order)) %>%
  select(primer, cat) %>%
  bind_cols(total.reads = rowSums(art_full[,8:44])) %>%
  group_by(primer, cat) %>%
  summarize(total.reads = sum(total.reads),
            .groups = "drop") %>%
  group_by(primer) %>%
  mutate(rel.reads = 100*total.reads / sum(total.reads)) %>%
  ungroup() %>%
  # this part adds primer-taxa categories that had no reads
  right_join(expand.grid(cat = unique(paste0(art_full$class, "-", 
                                      art_full$order)),
                         primer = c("coleop", "zeale"))) %>%
  # this part converts the NAs in the previous step to 0's
  mutate(rel.reads = tidyr::replace_na(rel.reads, 0),
         primer = case_when(primer == "coleop" ~ "Coleop",
                            primer == "zeale" ~ "Zeale")) %>%
  group_by(cat) %>%
  mutate(mean_rel.reads = mean(rel.reads)) %>%
  ungroup() %>%
  left_join(order_df) %>%
  mutate(name = forcats::fct_reorder(name, order, .desc = T))

(who_plot <- 
primer_who %>%
  ggplot() +
  geom_col(aes(y = name, x = sqrt(rel.reads),
               fill = primer),
           position = position_dodge(),
           alpha = 0.6) +
  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  scale_x_continuous(breaks = c(0,2,4,6),
                     labels = c(0,4,16,36)) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 10),
        legend.position = c(.7,.1)) +
  labs(x = "% of total reads",
       y = "",
       fill = "")# +
  # geom_text(aes(y = name, x = rel.reads + 1.5, 
  #               color = primer, 
  #               label = round(rel.reads,2)),
  #           position = position_dodge(width = 1),
  #           size = 1.5,
  #           data = filter(primer_who, rel.reads < 1))
)

# Stitch together primer plots
patchobj <- who_plot | (rareS.plot / simpson.plot / shannon.plot)

ggsave("figures/primer_compare6.png", width = 7, height = 7, dpi = 400)

```


