# Arthropod metabarcoding (roller and spray eDNA aggregation methods)
Second field effort testing roller and spray aggregation methods with n = 20 samples each.

# Load libraries. Read, clean and format data
```{r}
library(dplyr)
# library(metacoder)
library(readxl)
library(ggplot2)
library(stringr)
# library(taxa)
# library(vegan)
library(BiodiversityR)
# library(wesanderson)
# library(patchwork)
library(tidyr)
library(iNEXT)
# library(forcats)
select <- dplyr::select
# path for taxonomy and ecology lookup tables
tax_data_path <- "data/arthropod.families_20220816.xlsx"
# note: the 20220816 version is up-to-date for 2methods
#   but hadn't received new families from 4methods yet
'%notin%' <- Negate('%in%')

# read in family name look up 
    # will use to match each OTU with an NCBI family name
    # created by looking up each entry in NCBI if it didn't 
      # have a family name yet from bioinformatics work
famlookup <- read_xlsx(tax_data_path,
                       sheet = "arthropod.families") %>%
  select(final_lowest,
         final_ID = 'final species_name',
         family:kingdom) %>%
  distinct()

# read in Zeale & Coleop MOTU data with taxonomy
zs3 <- read.csv("data/pinebugs.samples.step2.onlyarth.negsub.Z.csv")
cs3 <- read.csv("data/pinebugs.samples.step2.onlyarth.negsub.C.csv")

# read in family trait info (life mode)
    # note: class and order added with taxize package using script:
    # "scripts/adding_taxonomy_to_families.R"
    # terrestrial/aquatic info from Horren et al. database & others
lifemode <- read_xlsx(tax_data_path, 
                      sheet = "arthropod.families.lifemode") %>%
  select(-person) %>%
  mutate(life.cat = case_when(larva_aquatic >= 1 & 
                                 adult_aquatic >= 1 ~ 
                                 "Fully aquatic",
                               larva_terrestrial >= 1 &
                                 adult_terrestrial >= 1 ~ 
                                 "Fully terrestrial",
                               larva_aquatic >= 1 &
                                 adult_terrestrial >= 1 ~
                                 "All larvae aquatic",
                               TRUE ~ "Some larvae aquatic"))

# list families without trait info
missing.familes <- cs3 %>% 
  select(family) %>%
  bind_rows(select(zs3, family)) %>%
  filter(!is.na(family)) %>%
  left_join(select(lifemode, family, larva_terrestrial)) %>%
  select(family, larva_terrestrial) %>%
  distinct() %>%
  filter(is.na(larva_terrestrial)) %>%
  select(family)

# read in family trait info (diet/feeding)
    # note: class and order added with taxize package using script:
    # "scripts/adding_taxonomy_to_families.R"
    # diet/feeding info from Horren et al. database & others
diet <- read_xlsx(tax_data_path, 
                      sheet = "arthropod.families.diet") %>%
  select(-person) %>%
  # create a condensed diet/feeding classification
  mutate(diet.cat = case_when(final_diet == "mixed" ~ 
                                "multiple categories",
                              final_diet %in% 
                                c("zoophagous", "phytophagous") ~
                                paste(final_diet, feed_mode, 
                                      sep = "-"),
                              final_diet %in% c("saprophagous", 
                                                "detritophagous") ~ 
                                "detritophagous/scavenger",
                              TRUE ~ final_diet),
         diet.cat = case_when(notes == ".9 det/sap/cop" ~ 
                                "detritophagous/scavenger",
                              family %in% c("Calliphoridae",
                                                  "Polleniidae",
                                                  "Rhiniidae") ~
                                "detritophagous/scavenger",
                              diet.cat == "zoophagous-necro" ~
                                "detritophagous/scavenger",
                           diet.cat %in% c("phytophagous-mixed",
                                        "phytophagous-not specified") ~
                             "phytophagous-multiple/other",
                           diet.cat == "phytophagous-flower" ~
                             "phytophagous-multiple/other",
                           diet.cat %in% c("phytophagous-miners",
                                           "phytophagous-gall") ~
                             "phytophagous-galls/miners",
                           diet.cat %in% c("zoophagous-parasite",
                                           "zoophagous-parasitoid",
                                           "zoophagous-micropredator") ~
                             "zoophagous-parasite/parasitoid",
                              TRUE ~ diet.cat),
         diet.cat2 = case_when(diet.cat == 
                                "detritophagous/scavenger" ~ 
                                "Scavenger",
                              diet.cat == 
                                "phytophagous-galls/miners" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-multiple/other" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-phyllophagous" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-sap sucking" ~
                                "Phytophagous",
                              diet.cat == 
                                "phytophagous-saproxylic" ~
                                "Phytophagous",
                              diet.cat == 
                                "zoophagous-parasite/parasitoid" ~
                                "Parasite/parasitoid",
                              diet.cat == 
                                "zoophagous-predator" ~
                                "Predator",
                              diet.cat == 
                                "multiple categories" ~
                                "Mulitple categories",
                              diet.cat == 
                                "mycetophagous" ~
                                "Mycetophagous",
                              TRUE ~ diet.cat))

table(diet$diet.cat)
# final category notes: 
# detritophagous/scavenger = detrit., copro., sapro., and zoo.-necro.

# read in family trait info (body size)
    # note: class and order added with taxize package using script:
    # "scripts/adding_taxonomy_to_families.R"
    # diet/feeding info from Horren et al. database & others
size <- read_xlsx(tax_data_path, 
                      sheet = "arthropod.families.size") %>%
  select(-person) %>%
  select(family:final_source) %>%
  mutate(Body_Size = case_when(final_mid > 20 ~ ">20 mm",
                               final_mid > 10 & 
                                 final_mid <= 20 ~ "10-20 mm",
                               final_mid > 5 &
                                 final_mid <= 10 ~ "5-10 mm",
                               final_mid <= 5 ~ "<5 mm",
                               TRUE ~ "Unknown"),
         Body_Size_order = case_when(Body_Size == "<5 mm" ~ 5,
                           Body_Size == "5-10 mm" ~ 4,
                           Body_Size == "10-20 mm" ~ 3,
                           Body_Size == ">20 mm" ~ 2,
                           Body_Size == "Unknown" ~ 1)
  )

art_full <- cs3 %>%
  mutate(primer = "coleop") %>%
  bind_rows(mutate(zs3, primer = "zeale")) %>%
  select(primer, id, phylum, class, order, family, 
         genus, sci, contains("CS")[c(2:10,1)], contains("DS")[c(2:7,1)],
         contains("CR")[c(2:10,1)], contains("DR")[c(2:10,1)]) %>%
  rowwise() %>% 
  mutate(sample.sum = sum(c_across(starts_with("sample")), na.rm = T)) %>%
  ungroup() %>%
  mutate(sample.3CS = case_when(primer == "zeale" ~ 0,
                                  TRUE ~ sample.3CS))

# rename columns
new_col_names <- art_full %>% 
  select(starts_with("sample")) %>%
  colnames() %>%
  substr(., 1, 11) %>%
  gsub(., pattern = ":", replacement = "") %>%
  gsub(., pattern = "B", replacement = "") %>%
  gsub(., pattern = "Z", replacement = "")

colnames(art_full) # check which columns are being replaced

colnames(art_full)[9:46] <- c(new_col_names)

colnames(art_full) # check that it worked

# look at read depth
sort(colSums(select(filter(art_full, primer == "coleop"), starts_with("sample")), na.rm = T))
sort(colSums(select(filter(art_full, primer == "zeale"), starts_with("sample")), na.rm = T))

mean(colSums(select(filter(art_full, primer == "coleop"), starts_with("sample")), na.rm = T))
mean(colSums(select(filter(art_full, primer == "zeale"), starts_with("sample")), na.rm = T))

# make family-lineage lookup table
lineage_lookup <- art_full %>%
  dplyr::select(phylum, class, order, family, genus, sci) %>%
  distinct()

# format MOTU database for use in sp accum 
art_motu_reads <- art_full %>%
  select(primer, id, starts_with("sample")) %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(id, ends_with("coleop"), ends_with("zeale"))

# subset databases that summarize by order, family, genera, species-level IDs
art_order_reads <- art_full %>%
  filter(order != "NA" & is.na(order)==F) %>%
  group_by(primer, order) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(order, ends_with("coleop"), ends_with("zeale"))

art_order_motus <- art_full %>%
  filter(order != "NA" & is.na(order)==F) %>%
  group_by(primer, order) %>%
  summarise(across(starts_with("sample"), function(x) sum(x>0)),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(order, ends_with("coleop"), ends_with("zeale"))

art_fam_reads <- art_full %>%
  filter(family != "NA" & is.na(family)==F) %>%
  group_by(primer, family) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(family, ends_with("coleop"), ends_with("zeale"))

art_fam_motus <- art_full %>%
  filter(family != "NA" & is.na(family)==F) %>%
  group_by(primer, family) %>%
  summarise(across(starts_with("sample"), function(x) sum(x>0)),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(family, ends_with("coleop"), ends_with("zeale"))

art_gen_reads <- art_full %>%
  filter(genus != "NA" & is.na(genus)==F) %>%
  group_by(primer, genus) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(genus, ends_with("coleop"), ends_with("zeale"))

art_gen_motus <- art_full %>%
  filter(genus != "NA" & is.na(genus)==F) %>%
  group_by(primer, genus) %>%
  summarise(across(starts_with("sample"), function(x) sum(x>0)),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(genus, ends_with("coleop"), ends_with("zeale"))

art_sp_reads <- art_full %>%
  filter(genus != "NA" & is.na(genus)==F & (genus != sci) & 
           (sci != "Paria ") & (sci != "Lucilia ")) %>% # this part fixes 2 typos in sci
  group_by(primer, sci) %>%
  summarise(across(starts_with("sample"), sum),
            .groups = "drop") %>%
  pivot_wider(names_from = primer, values_from = starts_with("sample")) %>%
  select(sci, ends_with("coleop"), ends_with("zeale"))

#### Format data for accumulation curves

# format data matrices

# spray samples - zeale - motu level
sprayZ_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "Z")) %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no sample for the Zeale primer
  t()
sprayZ_motu_mat[is.na(sprayZ_motu_mat)] <- 0 
sprayZ_motu_mat[sprayZ_motu_mat>0] <- 1 
colnames(sprayZ_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "Z"))$id

# roller samples - zeale - motu level
rollZ_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "Z")) %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_motu_mat[is.na(rollZ_motu_mat)] <- 0 
rollZ_motu_mat[rollZ_motu_mat>0] <- 1 
colnames(rollZ_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "Z"))$id

# both - zeale - sp level
zeromat.motu.Z <- matrix(0,1,421)
sprayZ_motu_mat2 <- rbind(sprayZ_motu_mat[1:2,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[3:9,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[10,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[11:15,],
                         zeromat.motu.Z,
                         sprayZ_motu_mat[16,])

bothZ_motu_mat <- sprayZ_motu_mat2 + rollZ_motu_mat
bothZ_motu_mat[bothZ_motu_mat>0] <- 1

# spray samples - coleop - motu level
sprayC_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "C")) %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_motu_mat[is.na(sprayC_motu_mat)] <- 0 
sprayC_motu_mat[sprayC_motu_mat>0] <- 1 
colnames(sprayC_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "C"))$id

# roller samples - coleop - motu level
rollC_motu_mat <- art_motu_reads %>%
  filter(grepl(id, pattern = "C")) %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_motu_mat[is.na(rollC_motu_mat)] <- 0 
rollC_motu_mat[rollC_motu_mat>0] <- 1 
colnames(rollC_motu_mat) <- filter(art_motu_reads, grepl(id, pattern = "C"))$id

# both - coleop - motu level
zeromat.motu.C <- matrix(0,1,656)
sprayC_motu_mat2 <- rbind(sprayC_motu_mat[1:10,],
                         zeromat.motu.C,
                         sprayC_motu_mat[11,],
                         zeromat.motu.C,
                         sprayC_motu_mat[12:16,],
                         zeromat.motu.C,
                         sprayC_motu_mat[17,])

bothC_motu_mat <- sprayC_motu_mat2 + rollC_motu_mat
bothC_motu_mat[bothC_motu_mat>0] <- 1

# spray samples - zeale - sp level
sprayZ_sp_mat <- art_sp_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no reads for the Zeale primer
  t()
sprayZ_sp_mat[is.na(sprayZ_sp_mat)] <- 0 
sprayZ_sp_mat[sprayZ_sp_mat>0] <- 1 
colnames(sprayZ_sp_mat) <- art_sp_reads$sci

# roller samples - zeale - sp level
rollZ_sp_mat <- art_sp_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_sp_mat[is.na(rollZ_sp_mat)] <- 0 
rollZ_sp_mat[rollZ_sp_mat>0] <- 1 
colnames(rollZ_sp_mat) <- art_sp_reads$sci

# both - zeale - sp level
zeromat.sp <- matrix(0,1,238)
sprayZ_sp_mat2 <- rbind(sprayZ_sp_mat[1:2,],
                         zeromat.sp,
                         sprayZ_sp_mat[3:9,],
                         zeromat.sp,
                         sprayZ_sp_mat[10,],
                         zeromat.sp,
                         sprayZ_sp_mat[11:15,],
                         zeromat.sp,
                         sprayZ_sp_mat[16,])

bothZ_sp_mat <- sprayZ_sp_mat2 + rollZ_sp_mat
bothZ_sp_mat[bothZ_sp_mat>0] <- 1

# spray samples - coleop - sp level
sprayC_sp_mat <- art_sp_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_sp_mat[is.na(sprayC_sp_mat)] <- 0 
sprayC_sp_mat[sprayC_sp_mat>0] <- 1 
colnames(sprayC_sp_mat) <- art_sp_reads$sci

# roller samples - coleop - sp level
rollC_sp_mat <- art_sp_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_sp_mat[is.na(rollC_sp_mat)] <- 0 
rollC_sp_mat[rollC_sp_mat>0] <- 1 
colnames(rollC_sp_mat) <- art_sp_reads$sci

# both - coleop - sp level
sprayC_sp_mat2 <- rbind(sprayC_sp_mat[1:10,],
                         zeromat.sp,
                         sprayC_sp_mat[11,],
                         zeromat.sp,
                         sprayC_sp_mat[12:16,],
                         zeromat.sp,
                         sprayC_sp_mat[17,])

bothC_sp_mat <- sprayC_sp_mat2 + rollC_sp_mat
bothC_sp_mat[bothC_sp_mat>0] <- 1

# spray samples - zeale - genus level
sprayZ_gen_mat <- art_gen_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no reads for the Zeale primer
  t()
sprayZ_gen_mat[is.na(sprayZ_gen_mat)] <- 0 
sprayZ_gen_mat[sprayZ_gen_mat>0] <- 1 
colnames(sprayZ_gen_mat) <- art_gen_reads$genus

# roller samples - zeale - genus level
rollZ_gen_mat <- art_gen_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_gen_mat[is.na(rollZ_gen_mat)] <- 0 
rollZ_gen_mat[rollZ_gen_mat>0] <- 1 
colnames(rollZ_gen_mat) <- art_gen_reads$genus

# both - zeale - genus level
zeromat.gen <- matrix(0,1,310)
sprayZ_gen_mat2 <- rbind(sprayZ_gen_mat[1:2,],
                         zeromat.gen,
                         sprayZ_gen_mat[3:9,],
                         zeromat.gen,
                         sprayZ_gen_mat[10,],
                         zeromat.gen,
                         sprayZ_gen_mat[11:15,],
                         zeromat.gen,
                         sprayZ_gen_mat[16,])

bothZ_gen_mat <- sprayZ_gen_mat2 + rollZ_gen_mat
bothZ_gen_mat[bothZ_gen_mat>0] <- 1

# spray samples - coleop - genus level
sprayC_gen_mat <- art_gen_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_gen_mat[is.na(sprayC_gen_mat)] <- 0 
sprayC_gen_mat[sprayC_gen_mat>0] <- 1 
colnames(sprayC_gen_mat) <- art_gen_reads$genus

# roller samples - coleop - genus level
rollC_gen_mat <- art_gen_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_gen_mat[is.na(rollC_gen_mat)] <- 0 
rollC_gen_mat[rollC_gen_mat>0] <- 1 
colnames(rollC_gen_mat) <- art_gen_reads$genus

# both - coleop - genus level
sprayC_gen_mat2 <- rbind(sprayC_gen_mat[1:10,],
                         zeromat.gen,
                         sprayC_gen_mat[11,],
                         zeromat.gen,
                         sprayC_gen_mat[12:16,],
                         zeromat.gen,
                         sprayC_gen_mat[17,])

bothC_gen_mat <- sprayC_gen_mat2 + rollC_gen_mat
bothC_gen_mat[bothC_gen_mat>0] <- 1

# spray samples - zeale - family level
sprayZ_fam_mat <- art_fam_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("zeale")) %>%
  select(-sample.3CS_zeale) %>% # a sample with no reads for the Zeale primer
  t()
sprayZ_fam_mat[is.na(sprayZ_fam_mat)] <- 0 
sprayZ_fam_mat[sprayZ_fam_mat>0] <- 1 
colnames(sprayZ_fam_mat) <- art_fam_reads$family

# roller samples - zeale primer - family level
rollZ_fam_mat <- art_fam_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("zeale")) %>%
  t()
rollZ_fam_mat[is.na(rollZ_fam_mat)] <- 0 
rollZ_fam_mat[rollZ_fam_mat>0] <- 1 
colnames(rollZ_fam_mat) <- art_fam_reads$family

# both - zeale - family level
zeromat <- matrix(0,1,172)
sprayZ_fam_mat2 <- rbind(sprayZ_fam_mat[1:2,],
                         zeromat,
                         sprayZ_fam_mat[3:9,],
                         zeromat,
                         sprayZ_fam_mat[10,],
                         zeromat,
                         sprayZ_fam_mat[11:15,],
                         zeromat,
                         sprayZ_fam_mat[16,])

bothZ_fam_mat <- sprayZ_fam_mat2 + rollZ_fam_mat
bothZ_fam_mat[bothZ_fam_mat>0] <- 1

# spray samples - coleop - family level
sprayC_fam_mat <- art_fam_reads %>%
  select(contains("CS"), contains("DS")) %>%
  select(contains("coleop")) %>%
  t()
sprayC_fam_mat[is.na(sprayC_fam_mat)] <- 0 
sprayC_fam_mat[sprayC_fam_mat>0] <- 1 
colnames(sprayC_fam_mat) <- art_fam_reads$family

# roller samples - coleop primer - family level
rollC_fam_mat <- art_fam_reads %>%
  select(contains("CR"), contains("DR")) %>%
  select(contains("coleop")) %>%
  t()
rollC_fam_mat[is.na(rollC_fam_mat)] <- 0 
rollC_fam_mat[rollC_fam_mat>0] <- 1 
colnames(rollC_fam_mat) <- art_fam_reads$family

# both - coleop - family level
sprayC_fam_mat2 <- rbind(sprayC_fam_mat[1:10,],
                         zeromat,
                         sprayC_fam_mat[11,],
                         zeromat,
                         sprayC_fam_mat[12:16,],
                         zeromat,
                         sprayC_fam_mat[17,])

bothC_fam_mat <- sprayC_fam_mat2 + rollC_fam_mat
bothC_fam_mat[bothC_fam_mat>0] <- 1

```
# construct accumulation curves using iNEXT
```{r}

# format data for iNEXT
accum.data.motu <- list('Leaves (COI)' = t(sprayZ_motu_mat),
                 'Bark (COI)' = t(rollZ_motu_mat),
                 'Both substrates (COI)' = t(bothZ_motu_mat),
                 'Leaves (16S)' = t(sprayC_motu_mat),
                 'Bark (16S)' = t(rollC_motu_mat),
                 'Both substrates (16S)' = t(bothC_motu_mat))

accum.data.sp <- list('Leaves (COI)' = t(sprayZ_sp_mat),
                 'Bark (COI)' = t(rollZ_sp_mat),
                 'Both substrates (COI)' = t(bothZ_sp_mat),
                 'Leaves (16S)' = t(sprayC_sp_mat),
                 'Bark (16S)' = t(rollC_sp_mat),
                 'Both substrates (16S)' = t(bothC_sp_mat))

accum.data.gen <- list('Leaves (COI)' = t(sprayZ_gen_mat),
                 'Bark (COI)' = t(rollZ_gen_mat),
                 'Both substrates (COI)' = t(bothZ_gen_mat),
                 'Leaves (16S)' = t(sprayC_gen_mat),
                 'Bark (16S)' = t(rollC_gen_mat),
                 'Both substrates (16S)' = t(bothC_gen_mat))

accum.data.fam <- list('Leaves (COI)' = t(sprayZ_fam_mat),
                 'Bark (COI)' = t(rollZ_fam_mat),
                 'Both substrates (COI)' = t(bothZ_fam_mat),
                 'Leaves (16S)' = t(sprayC_fam_mat),
                 'Bark (16S)' = t(rollC_fam_mat),
                 'Both substrates (16S)' = t(bothC_fam_mat))

# run accumulation curve analyses in iNEXT
accum.motu <- iNEXT(accum.data.motu, q=0, datatype="incidence_raw", endpoint = 60)
accum.motu$iNextEst$coverage_based

accum.sp <- iNEXT(accum.data.sp, q=0, datatype="incidence_raw", endpoint = 60)
accum.sp$iNextEst$coverage_based

accum.gen <- iNEXT(accum.data.gen, q=0, datatype="incidence_raw", endpoint = 60)
accum.gen$iNextEst$coverage_based

accum.fam <- iNEXT(accum.data.fam, q=0, datatype="incidence_raw", endpoint = 60)
accum.fam$iNextEst$coverage_based

(motuplot <-
ggiNEXT(accum.motu, type=1, se=TRUE, 
        facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. MOTUs") +
    theme_bw() +
  theme(text = element_text(size = 14)) +
  guides(color = "none", shape = "none", fill = "none", linetype = "none")
)

(spplot <-
ggiNEXT(accum.sp, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. species identified") +
    theme_bw() +
  theme(text = element_text(size = 14)) +
  guides(color = "none", shape = "none", fill = "none", linetype = "none")
)

(genplot <-
ggiNEXT(accum.gen, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. genera identified") +
    theme_bw() +
  theme(text = element_text(size = 14)) +
  guides(color = "none", shape = "none", fill = "none", linetype = "none")
)

(famplot <- 
ggiNEXT(accum.fam, type=1, se=TRUE, facet.var="None", color.var="Assemblage", grey=FALSE) +
  scale_x_continuous(limits = c(0,60)) +
  labs(y = "No. families identified") +
    theme_bw() +
  theme(text = element_text(size = 15),
        legend.position = c(.78, .19)) +
  guides(linetype = "none") +
  theme(legend.key.size = unit(.12, 'cm'), #change legend key size
        legend.key.height = unit(.12, 'cm'), #change legend key height
        legend.key.width = unit(.12, 'cm'), #change legend key width
        legend.title = element_text(size=12), #change legend title font size
        legend.text = element_text(size=12)) #change legend text font size
)

library(patchwork)
(motuplot + spplot) / (genplot + famplot)
ggsave("figures/rev1_accumcurves5.png", width = 12, height = 5, dpi = 400) # 3-panel
ggsave("figures/rev1_accumcurves10.png", width = 12, height = 10, dpi = 400) # 4-panel

```
# NMDS COI vs 16S, Spray vs Roller, conifer vs. deciduous 
```{r}
# format data matrices
mds.data.sp <- rbind(sprayZ_sp_mat,
                  rollZ_sp_mat,
                  sprayC_sp_mat,
                  rollC_sp_mat)

mds.data.gen <- rbind(sprayZ_gen_mat,
                  rollZ_gen_mat,
                  sprayC_gen_mat,
                  rollC_gen_mat)

mds.data.fam <- rbind(sprayZ_fam_mat,
                  rollZ_fam_mat,
                  sprayC_fam_mat,
                  rollC_fam_mat)

# check for blank rows
apply(mds.data.sp, 1, max)
apply(mds.data.gen, 1, max)
apply(mds.data.fam, 1, max)

# remove blank rows (no blanks)
  # these result from samples where no MOTUs were ID'd to this taxonomic level
mds.data.sp.nb <- mds.data.sp[-which(rownames(mds.data.sp) %in% 
                                     c("sample.9DR_zeale")),]

mds.data.gen.nb <- mds.data.gen[-which(rownames(mds.data.gen) %in% 
                                     c("sample.9DR_zeale")),]

mds.data.fam.nb <- mds.data.fam
  
# format covariate data - species level
env.data <- data.frame(
  tree = str_sub(gsub(gsub(rownames(mds.data.sp.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                      -2, -2),
  method = str_sub(gsub(gsub(rownames(mds.data.sp.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                        -1, -1)) %>%
  mutate(tree = case_when(tree == "C" ~ "Conifer",
                          tree == "D" ~ "Decid."),
         method = case_when(method == "R" ~ "Bark",
                            method == "S" ~ "Leaves"),
  primer = case_when(grepl(rownames(mds.data.sp.nb), pattern = "zeal") ~ "COI",
                     TRUE ~ "16S"))

# package vegan NMDS
set.seed(110)
# arthNMDS.sp = metaMDS(mds.data.sp.nb,
#                      k=2, distance = 'jaccard', trymax = 10000) 
# saveRDS(arthNMDS.sp, "output/arthNMDS.sp.rds")
arthNMDS.sp <- readRDS("output/arthNMDS.sp.rds")

plot1 <- ordiplot(arthNMDS.sp, choices = c(1, 2))

sites.long1 <-
  BiodiversityR::sites.long(plot1, env.data = env.data) %>%
  mutate(axis1 = as.numeric(axis1),
         axis2 = as.numeric(axis2),
         Method = paste0(method, " - ", tree, " (", primer, ")"))
head(sites.long1)
str(sites.long1)

(
  plot.sp <-
    ggplot(sites.long1) +
    geom_vline(
      xintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    geom_hline(
      yintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    scale_x_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_y_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_color_manual(values = c("steelblue", "steelblue", "steelblue", "steelblue", 
                                  "firebrick","firebrick", "firebrick", "firebrick")) +
    scale_fill_manual(values = c("white", "steelblue", "white", "steelblue", 
                                  "white", "firebrick", "white", "firebrick")) +
    scale_shape_manual(values = c(21, 21, 24, 24, 21, 21, 24, 24)) +
    geom_point(
      aes(
        x = axis1,
        y = axis2,
        shape = Method,
        color = Method,
        fill = Method,
      ),
      size = 3,
      alpha = 0.85
    ) +
    # coord_fixed(ratio = 1) +
    labs(
      y = "NMDS2",
      x = "NMDS1",
      color = "Substrate",
      shape = "Tree type"
    ) +
    theme_bw() +
    theme(text = element_text(size = 12)) +
    ggtitle("Species") +
    guides(shape = "none", color = "none", fill = "none")
)

# ggsave("figures/rev2_NMDS_by_method_and_tree_type2.jpg",
#        dpi = 600, width = 6, height = 6)


# Test for differences among Method and Tree type
# PERMANOVA
set.seed(12555)
perm_arth <- adonis2(mds.data.sp.nb ~ primer + method + tree, data = env.data, 
       permutations = 9999, method = "jaccard",
       strata = NULL, contr.unordered = "contr.sum",
       contr.ordered = "contr.poly")

perm_arth

### genus level ###

# format covariate data - species level
 # (can use same data as for the species level)

# package vegan NMDS
set.seed(110)
# arthNMDS.gen = metaMDS(mds.data.gen.nb,
#                      k=2, distance = 'jaccard', trymax = 10000) 
# saveRDS(arthNMDS.gen, "output/arthNMDS.gen.rds")
arthNMDS.gen <- readRDS("output/arthNMDS.gen.rds")

plot1.gen <- ordiplot(arthNMDS.gen, choices = c(1, 2))

sites.long1.gen <-
  BiodiversityR::sites.long(plot1.gen, env.data = env.data) %>%
  mutate(axis1 = as.numeric(axis1),
         axis2 = as.numeric(axis2),
         Method = paste0(method, " - ", tree, " (", primer, ")"))
head(sites.long1.gen)
str(sites.long1.gen)

(
  plot.gen <-
    ggplot(sites.long1.gen) +
    geom_vline(
      xintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    geom_hline(
      yintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    scale_x_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_y_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_color_manual(values = c("steelblue", "steelblue", "steelblue", "steelblue", 
                                  "firebrick","firebrick", "firebrick", "firebrick")) +
    scale_fill_manual(values = c("white", "steelblue", "white", "steelblue", 
                                  "white", "firebrick", "white", "firebrick")) +
    scale_shape_manual(values = c(21, 21, 24, 24, 21, 21, 24, 24)) +
    geom_point(
      aes(
        x = axis1,
        y = axis2,
        shape = Method,
        color = Method,
        fill = Method
      ),
      size = 3,
      alpha = 0.85
    ) +
    # coord_fixed(ratio = 1) +
    labs(
      y = "NMDS2",
      x = "NMDS1",
      color = "Substrate",
      shape = "Tree type"
    ) +
    theme_bw() +
    theme(text = element_text(size = 12)) +
    ggtitle("Genus") +
    guides(shape = "none", color = "none", fill = "none")
)

# ggsave("figures/rev2_NMDS_by_method_and_tree_type2.jpg",
#        dpi = 600, width = 6, height = 6)


# Test for differences among Method and Tree type (genus level)
# PERMANOVA
set.seed(12555)
perm_arth.gen <- adonis2(mds.data.gen.nb ~ primer + method + tree, data = env.data, 
       permutations = 9999, method = "jaccard",
       strata = NULL, contr.unordered = "contr.sum",
       contr.ordered = "contr.poly")

perm_arth.gen

### family level ###

# format covariate data - family level
env.data.fam <- data.frame(
  tree = str_sub(gsub(gsub(rownames(mds.data.fam.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                      -2, -2),
  method = str_sub(gsub(gsub(rownames(mds.data.fam.nb), 
                           pattern="_zeale", replacement = ""),
                      pattern="_coleop", replacement = ""), 
                                        -1, -1)) %>%
  mutate(tree = case_when(tree == "C" ~ "Conifer",
                          tree == "D" ~ "Decid."),
         method = case_when(method == "R" ~ "Bark",
                            method == "S" ~ "Leaves"),
  primer = case_when(grepl(rownames(mds.data.fam.nb), pattern = "zeal") ~ "COI",
                     TRUE ~ "16S"))

# package vegan NMDS
set.seed(110)
# arthNMDS.fam = metaMDS(mds.data.fam.nb,
#                      k=2, distance = 'jaccard', trymax = 10000) 
# saveRDS(arthNMDS.fam, "output/arthNMDS.fam.rds")
arthNMDS.fam <- readRDS("output/arthNMDS.fam.rds")

plot1.fam <- ordiplot(arthNMDS.fam, choices = c(1, 2))

sites.long1.fam <-
  BiodiversityR::sites.long(plot1.fam, env.data = env.data.fam) %>%
  mutate(axis1 = as.numeric(axis1),
         axis2 = as.numeric(axis2),
         Method = paste0(method, " - ", tree, " (", primer, ")"))
head(sites.long1.fam)
str(sites.long1.fam)

(
  plot.fam <-
    ggplot(sites.long1.fam) +
    geom_vline(
      xintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    geom_hline(
      yintercept = c(0),
      color = "grey70",
      linetype = 2
    ) +
    scale_x_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_y_continuous(sec.axis = dup_axis(labels = NULL, name = NULL)) +
    scale_color_manual(values = c("steelblue", "steelblue", "steelblue", "steelblue", 
                                  "firebrick","firebrick", "firebrick", "firebrick")) +
    scale_fill_manual(values = c("white", "steelblue", "white", "steelblue", 
                                  "white", "firebrick", "white", "firebrick")) +
    scale_shape_manual(values = c(21, 21, 24, 24, 21, 21, 24, 24)) +
    geom_point(
      aes(
        x = axis1,
        y = axis2,
        shape = Method,
        color = Method,
        fill = Method
      ),
      size = 3,
      alpha = 0.85
    ) +
    # coord_fixed(ratio = 1) +
    labs(
      y = "NMDS2",
      x = "NMDS1",
      color = "Substrate",
      shape = "Tree type"
    ) +
    theme_bw() +
    theme(text = element_text(size = 12)) +
    ggtitle("Family") +
    guides(shape = "none", color = "none", fill = "none")
)

# ggsave("figures/rev2_NMDS_by_method_and_tree_type2.jpg",
#        dpi = 600, width = 6, height = 6)


# Test for differences among Method and Tree type
# PERMANOVA
set.seed(12555)
perm_arth.fam <- adonis2(mds.data.fam.nb ~ primer + method + tree, data = env.data.fam, 
       permutations = 9999, method = "jaccard",
       strata = NULL, contr.unordered = "contr.sum",
       contr.ordered = "contr.poly")

perm_arth.fam

library(patchwork)
plot.sp / plot.gen / plot.fam #+ 
  # plot_layout(widths = c(3, 3, 3),
              # heights = c(3,3,3))

ggsave("figures/rev2_nmds_all7.png", height = 9, width = 3, dpi = 400)
```
# Count up taxa by method and make Venn diagram
```{r}

### family

# subdivide 'art' into columns by primer / substrate / tree type for counting
Z_spray <- art_fam_reads %>%
  select(family, contains(c("CS_zeale", "DS_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() # 76 for zeale/spray (68 each for zeale/spray/conifer & zeale/spray/decid)

Z_roll <- art_fam_reads %>%
  select(family, contains(c("CR_zeale", "DR_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() # 82 for zeale/roll (58 for zeale/roll/conifer; 57 for zeale/roll/decid

C_spray <- art_fam_reads %>%
  select(family, contains(c("CS_coleop", "DS_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() # 107 for coleop/spray (80 for coleop/spray/conifer; 91 for coleop/spray/decid)

C_roll <- art_fam_reads %>%
  select(family, contains(c("CR_coleop", "DR_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(family) %>%
  distinct() # 88 for coleop/roll (69 for coleop/roll/conifer; 68 for coleop/roll/decid)

# get unique families within each method (change names to get different combos)
# zeale has 110 families (50 of which are not found in coleop)
# coleop has 122 families (62 of which are not found in zeale)
(C_uni <- C_spray %>%
    bind_rows(C_roll) %>%
    distinct() %>%
  filter(family %notin% unique(c(
                           Z_spray$family,
                           Z_roll$family)))) 

# Venn diagram of the 8 primer/substrates
# code from: https://www.datanovia.com/en/blog/venn-diagram-with-r-or-rstudio-a-million-ways/
# devtools::install_github("yanlinlin82/ggvenn")

venn_methods_fam <- list(
  'Leaves (COI)' = Z_spray$family,
  'Leaves (16S)' = C_spray$family,
  'Bark (COI)' = Z_roll$family,
  'Bark (16S)' = C_roll$family
)

library(ggvenn)
(vennfam <- ggvenn(
  venn_methods_fam, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF", "#0073C2FF"),
  # blue, yellow, gray, red
  stroke_size = 0.6, set_name_size = 3,
  text_size = 2.25
  ) + scale_x_continuous(expand = expansion(mult = .2)) #+
  #ggtitle("Family")
)

# ggsave("figures/venn3.jpg", height = 6, width = 6, dpi = 400)

# pairwise overlap (manually counted from Venn diagram)
# leaves-bark: 96 overlapping of 173 total (55% overlap)
# COI-16S: 60 overlapping of 173 total (35% overlap)
# conif-decid: 112 overlapping of 173 total (65% overlap)

### genus

# subdivide 'art' into columns by primer / substrate / tree type for counting
Z_spray.gen <- art_gen_reads %>%
  select(genus, contains(c("CS_zeale", "DS_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 147 for zeale/spray

Z_roll.gen <- art_gen_reads %>%
  select(genus, contains(c("CR_zeale", "DR_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 98 for zeale/roll

C_spray.gen <- art_gen_reads %>%
  select(genus, contains(c("CS_coleop", "DS_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 126 for coleop/spray

C_roll.gen <- art_gen_reads %>%
  select(genus, contains(c("CR_coleop", "DR_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(genus) %>%
  distinct() # 81 for coleop/roll

# get unique genera within each method (change names to get different combos)
# zeale has 191 genera (156 of which are not found in coleop)
# coleop has 152 genera (117 of which are not found in zeale)
(C_uni.gen <- C_spray.gen %>%
    bind_rows(C_roll.gen) %>%
    distinct() %>%
  filter(genus %notin% unique(c(
                           Z_spray.gen$genus, 
                           Z_roll.gen$genus)))) 

(Z_uni.gen <- Z_spray.gen %>%
    bind_rows(Z_roll.gen) %>%
    distinct() %>%
  filter(genus %notin% unique(c(
                           C_spray.gen$genus, 
                           C_roll.gen$genus)))) 

# Venn diagram of the 8 primer/substrates
# code from: https://www.datanovia.com/en/blog/venn-diagram-with-r-or-rstudio-a-million-ways/
# devtools::install_github("yanlinlin82/ggvenn")

venn_methods_gen <- list(
  'Leaves (COI)' = Z_spray.gen$genus,
  'Leaves (16S)' = C_spray.gen$genus,
  'Bark (COI)' = Z_roll.gen$genus,
  'Bark (16S)' = C_roll.gen$genus
)

library(ggvenn)
(venngen <- ggvenn(
  venn_methods_gen, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF", "#0073C2FF"),
  # blue, yellow, gray, red
  stroke_size = 0.6, set_name_size = 3,
  text_size = 2.25
  ) + scale_x_continuous(expand = expansion(mult = .2))#+
  #ggtitle("Genus")
)

# ggsave("figures/venn3.jpg", height = 6, width = 6, dpi = 400)

# pairwise overlap (manually counted from Venn diagram)
# leaves-bark: XX overlapping of XX total (XX% overlap)
# COI-16S: XX overlapping of XX total (XX% overlap)
# conif-decid: XX overlapping of XX total (65% overlap)

### species

# subdivide 'art' into columns by primer / substrate / tree type for counting
Z_spray.sp <- art_sp_reads %>%
  select(sci, contains(c("CS_zeale", "DS_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(sci) %>%
  distinct() # 119 for zeale/spray

Z_roll.sp <- art_sp_reads %>%
  select(sci, contains(c("CR_zeale", "DR_zeale"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(sci) %>%
  distinct() # 73 for zeale/roll

C_spray.sp <- art_sp_reads %>%
  select(sci, contains(c("CS_coleop", "DS_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(sci) %>%
  distinct() # 80 for coleop/spray

C_roll.sp <- art_sp_reads %>%
  select(sci, contains(c("CR_coleop", "DR_coleop"))) %>%
   mutate(sum = rowSums(across(where(is.numeric)))) %>%
  filter(sum > 0) %>%
  select(sci) %>%
  distinct() # 45 for coleop/roll

# get unique genera within each method (change names to get different combos)
# zeale has 157 sp (146 of which are not found in coleop)
# coleop has 92 sp (81 of which are not found in zeale)
(C_uni.sp <- C_spray.sp %>%
    bind_rows(C_roll.sp) %>%
    distinct() %>%
  filter(sci %notin% unique(c(
                           Z_spray.sp$sci, 
                           Z_roll.sp$sci)))) 

(Z_uni.sp <- Z_spray.sp %>%
    bind_rows(Z_roll.sp) %>%
    distinct() %>%
  filter(sci %notin% unique(c(
                           C_spray.sp$sci, 
                           C_roll.sp$sci)))) 

# Venn diagram of the 8 primer/substrates
# code from: https://www.datanovia.com/en/blog/venn-diagram-with-r-or-rstudio-a-million-ways/
# devtools::install_github("yanlinlin82/ggvenn")

venn_methods_sp <- list(
  'Leaves (COI)' = Z_spray.sp$sci,
  'Leaves (16S)' = C_spray.sp$sci,
  'Bark (COI)' = Z_roll.sp$sci,
  'Bark (16S)' = C_roll.sp$sci
)

library(ggvenn)
(vennsp <- ggvenn(
  venn_methods_sp, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF", "#0073C2FF"),
  # blue, yellow, gray, red
  stroke_size = 0.6, set_name_size = 3,
  text_size = 2.25
  ) + scale_x_continuous(expand = expansion(mult = .2))#+
  #ggtitle("Species")
)

# ggsave("figures/venn3.jpg", height = 6, width = 6, dpi = 400)

# pairwise overlap (manually counted from Venn diagram)
# leaves-bark: XX overlapping of XX total (XX% overlap)
# COI-16S: XX overlapping of XX total (XX% overlap)
# conif-decid: XX overlapping of XX total (65% overlap)

library(patchwork)

( plot.sp + vennsp ) / ( plot.gen + venngen ) / (plot.fam + vennfam ) # +

#   plot_layout(widths = c(3, 3, 3, 3, 3, 3),
#                heights = c(3,3,3,3,3,3))

test = gridExtra::arrangeGrob(plot.sp, vennsp, plot.gen, venngen, plot.fam, vennfam)

plot(test)

# ggsave("figures/rev3_venn_all10.png", height = 10, width = 6, dpi = 400)

ggsave(plot = test, filename = "figures/rev4_nmdsvenn_all13.png", height = 10, width = 7, dpi = 400)

# rm(art_spray, art_water, art_roll, art_soil,
#    soil_unique, spray_unique, roll_unique, water_unique,
#    venn_methods)
```

# Plot relative abundance of orders by method (for each primer set)
% of samples detected
```{r}
#################
#### Plotting abundance of each order by method (bar charts)
#### (no. MOTUs)
#################

### Coleop ###

ordersC <- art_full %>%
  filter(!is.na(order), primer == "coleop", sample.sum > 0) %>%
  mutate(idsp = case_when(!is.na(genus) & (genus != sci) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & (genus == sci) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0))%>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "16S")

ordersZ <- art_full %>%
  filter(!is.na(order), primer == "zeale", sample.sum > 0) %>%
  mutate(idsp = case_when(!is.na(genus) & (genus != sci) & sample.sum > 0 ~ 1,
                          TRUE ~ 0),
         idgen = case_when(!is.na(genus) & (genus == sci) & sample.sum > 0 ~ 1,
                           TRUE ~ 0),
         idfam = case_when((!is.na(family) & is.na(genus) & sample.sum > 0) ~ 1,
                           TRUE ~ 0),
         idord = case_when((!is.na(order) & is.na(family) & sample.sum > 0) ~ 1,
                           TRUE ~ 0)) %>%
  select(order, idsp, idgen, idfam, idord, sample.sum) %>%
  group_by(order) %>%
  summarize(motus = sum(sample.sum > 0),
            ord = sum(idord),
            fam = sum(idfam),
            gen = sum(idgen),
            sp = sum(idsp),
            .groups = "drop") %>%
  right_join(select(art_order_reads, order)) %>%
  tidyr::replace_na(list(motus = 0, ord = 0, fam = 0, gen = 0, sp = 0)) %>%
  left_join(distinct(select(lineage_lookup, class, order))) %>%
  arrange(class, order) %>%
  mutate(primer = "COI") 

orders_df <- ordersC %>%
  bind_rows(ordersZ) %>%
  mutate(engname = case_when(order == "Mesostigmata" ~ "Mites",
                          order == "Sarcoptiformes" ~ "Mites",
                          order == "Trombidiformes" ~ "Mites",
                          class == "Collembola" ~ "Springtails",
                          class == "Chilopoda" ~ "Centipedes",
                          order == "Araneae" ~ "Spiders",
                          order == "Opiliones" ~ "Harvestmen",
                          order == "Blattodea" ~ "Cockroaches",
                          order == "Coleoptera" ~ "Beetles",
                          order == "Dermaptera" ~ "Earwigs",
                          order == "Diptera" ~ "Flies",
                          order == "Ephemeroptera" ~ "Mayflies",
                          order == "Hemiptera" ~ "True bugs",
                          order == "Hymenoptera" ~ "Bees, wasps, & ants",
                          order == "Lepidoptera" ~ "Butterflies & moths",
                          order == "Mantodea" ~ "Mantids",
                          order == "Megaloptera" ~ "Dobsonflies",
                          order == "Cyclopoida" ~ "Copepods",
                          order == "Neuroptera" ~ "Lacewings",
                          order == "Odonata" ~ "Dragonflies & damselflies",
                          order == "Orthoptera" ~ "Crickets & grasshoppers",
                          order == "Phasmatodea" ~ "Walking sticks",
                          order == "Phthiraptera" ~ "Thrips",
                          order == "Psocoptera" ~ "Bark lice",
                          order == "Siphonaptera" ~ "Fleas",
                          order == "Trichoptera" ~ "Caddisflies"
                          )) %>%
  group_by(class, primer, engname) %>%
  summarize(Order = sum(ord, na.rm = T),
            Family = sum(fam, na.rm = T),
            Genus = sum(gen, na.rm = T),
            Species = sum(sp, na.rm = T),
            .groups = "drop") %>%
  pivot_longer(Order:Species, names_to = "idlevel", values_to = "motus") %>%
  group_by(class, engname) %>%
  mutate(summotu = sum(motus)) %>%
  ungroup() %>%
  arrange(primer, class, engname, idlevel) %>%
  mutate(nameorder = case_when(engname == "Flies" ~ 1,
                             engname == "Beetles" ~ 2,
                             engname == "Butterflies & moths" ~ 3,
                             engname == "True bugs" ~ 4,
                             engname == "Crickets & grasshoppers" ~ 5,
                             engname == "Bees, wasps, & ants" ~ 6,
                             engname == "Bark lice" ~ 7,
                             engname == "Cockroaches" ~ 8,
                             engname == "Lacewings" ~ 9,
                             engname == "Dragonflies & damselflies" ~ 10,
                             engname == "Caddisflies" ~ 11,
                             engname == "Mantids" ~ 12,
                             engname == "Mayflies" ~ 13,
                             engname == "Fleas" ~ 14,
                             engname == "Dobsonflies" ~ 15,
                             engname == "Earwigs" ~ 16,
                             engname == "Thrips" ~ 17,
                             engname == "Walking sticks" ~ 18,
                             engname == "Springtails" ~ 19,
                             engname == "Mites" ~ 20,
                             engname == "Spiders" ~ 21,
                             engname == "Harvestmen" ~ 22,
                             engname == "Centipedes" ~ 23,
                             engname == "Copepods" ~ 24
                             )) %>%
  mutate(taxorder = case_when(idlevel == "Species" ~ 1,
                              idlevel == "Genus" ~ 2,
                              idlevel == "Family" ~ 3,
                              idlevel == "Order" ~ 4)) %>%
  mutate(engnam2 = fct_reorder(.f = engname, .x = nameorder, .desc = T),
         engname = fct_reorder(.f = engname, .x = summotu, .desc = F),
         idlevel = fct_reorder(.f = idlevel, .x = taxorder, .desc = T))

# make plot for less-diverse taxa
library(forcats)
ibrary(wesanderson)
(
ldiv_plot <- 
orders_df %>%
    mutate(group = case_when(summotu >= 31 ~ "More",
                             TRUE ~ "Less")) %>%
    filter(group == "Less") %>%
  ggplot() +
  geom_col(aes(y = engname, x = motus,
               fill = idlevel)) +
  facet_wrap(~ primer, ncol = 2) +
  # scale_fill_manual(values = wes_palettes$Zissou1) +
  # scale_color_manual(values = wes_palettes$Zissou1) +
  scale_fill_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  scale_color_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 11), 
        legend.position = c(.89,.22),
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = "black")
        ) +
  theme(legend.key.size = unit(.45, 'cm'), #change legend key size
        legend.key.height = unit(.45, 'cm'), #change legend key height
        legend.key.width = unit(.45, 'cm'), #change legend key width
        legend.title = element_text(size=12), #change legend title font size
        legend.text = element_text(size=12)) +
  labs(x = "Number of MOTUs",
       y = "",
       fill = "Lowest \ntaxonomic \nID") #+
  #  guides(fill = "none")
)

# make plot for more-diverse taxa
(
hdiv_plot <- 
orders_df %>%
    mutate(group = case_when(summotu >= 31 ~ "More",
                             TRUE ~ "Less")) %>%
    filter(group == "More") %>%
  ggplot() +
  geom_col(aes(y = engname, x = motus,
               fill = idlevel)) +
  facet_wrap(~ primer, ncol = 2) +
  # scale_fill_manual(values = wes_palettes$Darjeeling1) +
  # scale_color_manual(values = wes_palettes$Darjeeling1) +
  scale_fill_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  scale_color_manual(values = c("gray10", "gray30", "gray50", "gray70")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 11)#, legend.position = c(.89,.24)
        ) +
  labs(x = "",
       y = "",
       fill = "Lowest \ntaxonomic \nID") +
    guides(fill = "none")
)

library(patchwork)
hdiv_plot / ldiv_plot

ggsave("figures/rev5_ordermotus12.png", height = 9, width = 7.5, dpi = 400)

```


# Feeding guild - fine scale
```{r}
feed_step1 <- diet %>%
  select(family, diet.cat) %>%
  left_join(lineage_lookup) %>%
  right_join(n_otus_max) %>%
  group_by(diet.cat) %>%
  # counts number of families with > 0 reads in each sample
  summarize(across(starts_with("sample"), 
                   .fns = function(x) sum(x > 0)),
            .groups = "drop")

feedcats <- feed_step1$diet.cat
feed <- feed_step1 %>%
  select(starts_with("sample")) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  mutate(sample_id = colnames(feed_step1)[2:41],
         method = str_sub(sample_id, -1,-1),
         tree = str_sub(sample_id, -2,-2)) %>%
  group_by(method, tree) %>%
  # calculates mean number of families in samples of each group
  summarize(across(contains("V"), mean),
            .groups = "drop")
colnames(feed) <- c("method", "tree", feedcats)
rm(feed_step1, feedcats)

feed_df <- feed %>% 
  pivot_longer(., cols = "detritophagous/scavenger":"zoophagous-predator") %>%
  mutate(diet.cat = case_when(name == 
                                "detritophagous/scavenger" ~ 
                                "Scavenger",
                              name == 
                                "phytophagous-galls/miners" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-multiple/other" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-phyllophagous" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-sap sucking" ~
                                "Phytophagous",
                              name == 
                                "phytophagous-saproxylic" ~
                                "Phytophagous",
                              name == 
                                "zoophagous-parasite/parasitoid" ~
                                "Parasite/parasitoid",
                              name == 
                                "zoophagous-predator" ~
                                "Predator",
                              name == 
                                "multiple categories" ~
                                "Mulitple categories",
                              name == 
                                "mycetophagous" ~
                                "Mycetophagous",
                              TRUE ~ name),
         tree = case_when(tree == "C" ~ "Coniferous",
                          tree == "D" ~ "Deciduous"),
         method = case_when(method == "R" ~ "Roller",
                          method == "S" ~ "Spray"),
         order = case_when(diet.cat == "Predator" ~ 1,
                           diet.cat == "Parasite/parasitoid" ~ 2,
                           diet.cat == "Phytophagous" ~ 3,
                           diet.cat == "Scavenger" ~ 4,
                           diet.cat == "Mycetophagous" ~ 5,
                           diet.cat == "Multiple categories" ~ 6),
         diet.cat = fct_reorder(diet.cat, order)
         )
  
(feed_plot <- 
  ggplot(data = feed_df, 
       aes(x = method, y = value, fill = name)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Feeding Guild") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotG_feeding_guild_condensed.jpg", 
#        width = 6, height = 4, dpi = 400)

(feed_plot_condensed <- 
  ggplot(data = feed_df, 
       aes(x = method, y = value, fill = diet.cat)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Feeding Guild") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotH_feeding_guild_condensed.jpg", 
#        width = 6, height = 4, dpi = 400)

# Habitat / life mode
life_step1 <- lifemode %>%
  select(family, life.cat) %>%
  left_join(lineage_lookup) %>%
  right_join(n_otus_max) %>%
  group_by(life.cat) %>%
  summarize(across(starts_with("sample"), 
                   .fns = function(x) sum(x > 0)),
            .groups = "drop")

lifecats <- life_step1$life.cat
life <- life_step1 %>%
  select(starts_with("sample")) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  mutate(sample_id = colnames(life_step1)[2:41],
         method = str_sub(sample_id, -1,-1),
         tree = str_sub(sample_id, -2,-2)) %>%
  group_by(method, tree) %>%
  summarize(across(contains("V"), mean),
            .groups = "drop")
colnames(life) <- c("method", "tree", lifecats)
rm(life_step1, lifecats)

life_df <- life %>% 
  pivot_longer(., cols = starts_with("All"):starts_with("Some")) %>%
  mutate(tree = case_when(tree == "C" ~ "Coniferous",
                          tree == "D" ~ "Deciduous"),
         method = case_when(method == "R" ~ "Roller",
                          method == "S" ~ "Spray"),
         order = case_when(name == "Fully aquatic" ~ 4,
                           name == "Fully terrestrial" ~ 1,
                           name == "All larvae aquatic" ~ 3,
                           name == "Some larvae aquatic" ~ 2),
         name = fct_reorder(name, order, .desc = F))
  
(life_plot <- 
  ggplot(data = life_df, 
       aes(x = method, y = value, fill = name)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Life mode") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotI_life_mode.jpg",
#        width = 6, height = 4, dpi = 400)

# Body size
size_step1 <- size %>%
  select(family, size.cat = Body_Size, order = Body_Size_order) %>%
  left_join(lineage_lookup) %>%
  right_join(n_otus_max) %>%
  group_by(size.cat) %>%
  summarize(across(starts_with("sample"), 
                   .fns = function(x) sum(x > 0)),
            .groups = "drop")

sizecats <- size_step1$size.cat
size_step2 <- size_step1 %>%
  select(starts_with("sample")) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame() %>%
  mutate(sample_id = colnames(size_step1)[2:41],
         method = str_sub(sample_id, -1,-1),
         tree = str_sub(sample_id, -2,-2)) %>%
  group_by(method, tree) %>%
  summarize(across(contains("V"), mean),
            .groups = "drop")
colnames(size_step2) <- c("method", "tree", sizecats)
rm(size_step1, sizecats)

size_df <- size_step2 %>% 
  pivot_longer(., cols = 3:6) %>%
  mutate(tree = case_when(tree == "C" ~ "Coniferous",
                          tree == "D" ~ "Deciduous"),
         method = case_when(method == "R" ~ "Roller",
                          method == "S" ~ "Spray")) %>%
  left_join(distinct(select(size, name = Body_Size, 
                            order = Body_Size_order))) %>%
  mutate(name = fct_reorder(name, order, .desc = F))
  
(size_plot <- 
  ggplot(data = size_df, 
       aes(x = method, y = value, fill = name)) +
  theme_bw() +
  viridis::scale_fill_viridis(discrete=TRUE) +
  geom_bar(stat="identity", position = "fill") +
  scale_y_continuous(labels = scales::percent) +
  facet_wrap(~tree) +
  labs(y = "% of families", x = "", 
       fill = "") +
  ggtitle("Body size") +
  theme(axis.text.x = element_text(size = 15),
        strip.background = element_rect(fill = "white"))
)

# ggsave("figures/PlotJ_body_size.jpg", 
#        width = 6, height = 4, dpi = 400)

#################
##### combine feeding guild, life mode, and body size plots
#################

feed_plot_condensed / life_plot / size_plot

ggsave("figures/PlotK_guilds_bar.jpg", 
       width = 6, height = 12, dpi = 400)

```
# multinomial models for feeding guild data
https://discourse.mc-stan.org/t/example-with-family-multinomial/8707
https://discourse.mc-stan.org/t/interpret-parameters-from-multinomial-categorical-logit-regression/26374
```{r}
n_otus_pa <- n_otus_max[,-1]
n_otus_pa[n_otus_pa > 0] <- 1
n_otus_pa <- cbind(n_otus_max[,1], n_otus_pa)

feed_freq <- n_otus_pa %>%
  left_join(lineage_lookup) %>%
  left_join(select(diet, family, diet.cat2)) %>%
  select(-family, -lineage)

feed_freq_sum1 <- aggregate(feed_freq[,1:40], 
                           by = list(feed_freq$diet.cat2), sum)

feed_freq_sum2 <- feed_freq_sum1 %>%
  select(2:41) %>%
  t()
colnames(feed_freq_sum2) <- feed_freq_sum1$Group.1

feed_freq_sum <- feed_freq_sum2 %>%
  as.data.frame() %>%
  mutate(sample = rownames(feed_freq_sum2),
         tree = stringr::str_sub(sample, -2,-2),
         method = stringr::str_sub(sample, -1, -1)) %>%
  rename(mult = 1, myce = 2, para = 3, phyt = 4, pred = 5, scav = 6) %>%
  select(-sample)

library(brms)

dat <- feed_freq_sum
dat$a1 <- 1
dat$size <- with(dat, mult + myce + para + phyt + pred + scav)
dat$y <- with(dat, cbind(mult, myce, para, phyt, pred, scav))
# remove 3 blank samples
dat <- dat %>%
  filter(size > 0)

# model with both method and tree type
set.seed(102)
fit.method.tree <- brm(bf(y | trials(size) ~ method + tree), 
            data = dat, family = multinomial(refcat = "mult"),
            save_pars = save_pars(all = TRUE))

# model with only method
set.seed(102)
fit.method <- brm(bf(y | trials(size) ~ method), 
            data = dat, family = multinomial(refcat = "mult"),
            save_pars = save_pars(all = TRUE))

# model with only tree type
set.seed(102)
fit.tree <- brm(bf(y | trials(size) ~ tree), 
            data = dat, family = multinomial(refcat = "mult"),
            save_pars = save_pars(all = TRUE))

# adding LOOIC criterion to 3 models
fit.method.tree <- add_criterion(fit.method.tree, "loo", 
                                 moment_match = T,
                          reloo = TRUE)
fit.method <- add_criterion(fit.method, "loo", 
                                 moment_match = T,
                          reloo = TRUE)
fit.tree <- add_criterion(fit.tree, "loo", 
                                 moment_match = T,
                          reloo = TRUE)

# save model output so we don't have to refit each time
saveRDS(fit.method.tree, "output/feed_multinomial_method.tree_nb.rds")
saveRDS(fit.method, "output/feed_multinomial_method_nb.rds")
saveRDS(fit.tree, "output/feed_multinomial_tree_nb.rds")

# load the models so we don't have to refit them each time
fit.method.tree <- readRDS("output/feed_multinomial_method.tree_nb.rds")
fit.method <- readRDS("output/feed_multinomial_method_nb.rds")
fit.tree <- readRDS("output/feed_multinomial_tree_nb.rds")

# evaluating models based on LOOIC
brms::loo_compare(fit.method.tree, fit.method, fit.tree, 
            criterion = "loo") %>% 
  print(simplify = F)

# method is lowest, delta looic = 2.6
```
# multinomial models for lifemode data
https://discourse.mc-stan.org/t/example-with-family-multinomial/8707
https://discourse.mc-stan.org/t/interpret-parameters-from-multinomial-categorical-logit-regression/26374
```{r}
n_otus_pa <- n_otus_max[,-1]
n_otus_pa[n_otus_pa > 0] <- 1
n_otus_pa <- cbind(n_otus_max[,1], n_otus_pa)

life_freq <- n_otus_pa %>%
  left_join(lineage_lookup) %>%
  left_join(select(lifemode, family, life.cat)) %>%
  select(-family, -lineage)

life_freq_sum1 <- aggregate(life_freq[,1:40], 
                           by = list(life_freq$life.cat), sum)

life_freq_sum2 <- life_freq_sum1 %>%
  select(2:41) %>%
  t()
colnames(life_freq_sum2) <- life_freq_sum1$Group.1

life_freq_sum <- life_freq_sum2 %>%
  as.data.frame() %>%
  mutate(sample = rownames(life_freq_sum2),
         tree = stringr::str_sub(sample, -2,-2),
         method = stringr::str_sub(sample, -1, -1)) %>%
  rename(aqlar = 1, fullaq = 2, fullter = 3, 
         someaq = 4) %>%
  select(-sample)

library(brms)

dat <- life_freq_sum
dat$size <- with(dat, aqlar + fullaq + fullter + someaq)
dat$y <- with(dat, cbind(aqlar, fullaq, fullter, someaq))
# remove 3 blank samples
dat <- dat %>%
  filter(size > 0)

# model with both method and tree type
set.seed(102)
life.method.tree <- brms::brm(bf(y | trials(size) ~ method + tree), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 3, cores = 3)

# model with only method
set.seed(102)
life.method <- brms::brm(bf(y | trials(size) ~ method), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 3, cores = 3)

# model with only tree type
set.seed(102)
life.tree <- brms::brm(bf(y | trials(size) ~ tree), 
            data = dat, family = multinomial(refcat = "aqlar"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 3, cores = 3)

# adding LOOIC criterion to 3 models
life.method.tree <- brms::add_criterion(life.method.tree, "loo", 
                                 moment_match = T,
                          reloo = TRUE)
life.method <- brms::add_criterion(life.method, "loo", 
                                 moment_match = T,
                          reloo = TRUE)
life.tree <- brms::add_criterion(life.tree, "loo", 
                                 moment_match = T,
                          reloo = TRUE)

# save model output so we don't have to refit each time
saveRDS(life.method.tree, "output/life_multinomial_method.tree_nb.rds")
saveRDS(life.method, "output/life_multinomial_method_nb.rds")
saveRDS(life.tree, "output/life_multinomial_tree_nb.rds")

# load the models so we don't have to refit them each time
life.method.tree <- readRDS("output/life_multinomial_method.tree_nb.rds")
life.method <- readRDS("output/life_multinomial_method_nb.rds")
life.tree <- readRDS("output/life_multinomial_tree_nb.rds")

# evaluating models based on LOOIC
brms::loo_compare(life.method.tree, life.method, life.tree, 
            criterion = "loo") %>% 
  print(simplify = F)

# method is lowest, delta looic = 3.2
```
# multinomial models for size data
https://discourse.mc-stan.org/t/example-with-family-multinomial/8707
https://discourse.mc-stan.org/t/interpret-parameters-from-multinomial-categorical-logit-regression/26374
```{r}
n_otus_pa <- n_otus_max[,-1]
n_otus_pa[n_otus_pa > 0] <- 1
n_otus_pa <- cbind(n_otus_max[,1], n_otus_pa)

size_freq <- n_otus_pa %>%
  left_join(lineage_lookup) %>%
  left_join(select(size, family, size.cat = Body_Size)) %>%
  select(-family, -lineage)

size_freq_sum1 <- aggregate(size_freq[,1:40], 
                           by = list(size_freq$size.cat), sum)

size_freq_sum2 <- size_freq_sum1 %>%
  select(2:41) %>%
  t()
colnames(size_freq_sum2) <- size_freq_sum1$Group.1

size_freq_sum <- size_freq_sum2 %>%
  as.data.frame() %>%
  mutate(sample = rownames(size_freq_sum2),
         tree = stringr::str_sub(sample, -2,-2),
         method = stringr::str_sub(sample, -1, -1)) %>%
  rename(vsmal = 1, vbig = 2, big = 3, 
         smal = 4) %>%
  select(-sample)

# library(brms)

dat <- size_freq_sum
dat$size <- with(dat, vsmal + vbig + big + smal)
dat$y <- with(dat, cbind(vsmal, vbig, big, smal))
# remove 3 blank samples
dat <- dat %>%
  filter(size > 0)

# model with both method and tree type
set.seed(102)
size.method.tree <- brms::brm(bf(y | trials(size) ~ method + tree), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 3, cores = 3)

# model with only method
set.seed(102)
size.method <- brms::brm(bf(y | trials(size) ~ method), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 3, cores = 3)

# model with only tree type
set.seed(102)
size.tree <- brms::brm(bf(y | trials(size) ~ tree), 
            data = dat, family = multinomial(refcat = "vsmal"),
            save_pars = save_pars(all = TRUE), 
                          iter = 4000, warmup = 2000,
                          chains = 3, cores = 3)

# adding LOOIC criterion to 3 models
size.method.tree <- brms::add_criterion(size.method.tree, "loo", 
                                 moment_match = T,
                          reloo = TRUE)
size.method <- brms::add_criterion(size.method, "loo", 
                                 moment_match = T,
                          reloo = TRUE)
size.tree <- brms::add_criterion(size.tree, "loo", 
                                 moment_match = T,
                          reloo = TRUE)

# save model output so we don't have to refit each time
saveRDS(size.method.tree, "output/size_multinomial_method.tree_nb.rds")
saveRDS(size.method, "output/size_multinomial_method_nb.rds")
saveRDS(size.tree, "output/size_multinomial_tree_nb.rds")

# load the models so we don't have to refit them each time
size.method.tree <- readRDS("output/size_multinomial_method.tree_nb.rds")
size.method <- readRDS("output/size_multinomial_method_nb.rds")
size.tree <- readRDS("output/size_multinomial_tree_nb.rds")

# evaluating models based on LOOIC
brms::loo_compare(size.method.tree, size.method, size.tree, 
            criterion = "loo") %>% 
  print(simplify = F)

# method is lowest; delta looic = 4.7
```
# Multinomial model plots (feeding, life mode, body size)
```{r}
### Feeding guild

# load the models so we don't have to refit them each time
fit.method.tree <- readRDS("output/feed_multinomial_method.tree_nb.rds")
fit.method <- readRDS("output/feed_multinomial_method_nb.rds")
fit.tree <- readRDS("output/feed_multinomial_tree_nb.rds")

# plot top model predictions
fit.newdata = expand.grid(method = c("S", "R"),
                                 size = 1)

fitsies <- fitted(fit.method, 
                  newdata = fit.newdata,probs = c(0.025, 0.1,
                                                  0.9, 0.975))

fitsies

fitsies.list <- list()
for(i in 1:dim(fitsies)[3]){
  fitsies.list[[i]] <- fitsies[,,i]
}

fits_df <- do.call(rbind, fitsies.list) %>%
  as.data.frame() %>%
  mutate(method = rep(fit.newdata$method, 6),
         feed = c(rep("Mulitple categories", 2),
                  rep("Mycetophagous", 2),
                  rep("Parasite/Parasitoid", 2),
                  rep("Phytophagous", 2),
                  rep("Predator", 2),
                  rep("Scavenger", 2))) %>%
  group_by(feed) %>%
  mutate(mean_p = mean(Estimate)) %>%
  ungroup() %>%
  mutate(feed = fct_reorder(feed, mean_p),
         method = case_when(method == "S" ~ "Spray",
                            method == "R" ~ "Roller"))
         
(
  feed_plot2 <- 
fits_df %>%
  ggplot() +
  geom_point(aes(y = feed, x = 100*Estimate, color = method, 
                 group = method),
             position = position_dodge(width = .75),
             size = 2) +
  geom_errorbar(aes(y = feed, xmin = 100*Q10, 
                    xmax = 100*Q90, color = method,
                    group = method),
                size = 1,
                position = position_dodge(width = .75),
                width = 0) +
  geom_errorbar(aes(y = feed, xmin = 100*Q2.5, 
                    xmax = 100*Q97.5, color = method,
                    group = method),
                position = position_dodge(width = .75),
                width = 0) +
  scale_color_manual(values = c("steelblue", "firebrick")) +
  labs(color = "",
       y = "",
       x = "% of families",
       title = "Feeding guild") +
  theme_bw() +
  theme(text = element_text(size = 13),
        legend.position = c(.8,.2))
)

### Life mode

# load the models so we don't have to refit them each time
life.method.tree <- readRDS("output/life_multinomial_method.tree_nb.rds")
life.method <- readRDS("output/life_multinomial_method_nb.rds")
life.tree <- readRDS("output/life_multinomial_tree_nb.rds")

# plot top model predictions
fit.newdata = expand.grid(method = c("S", "R"),
                                 size = 1)

fitsies <- fitted(life.method, 
                  newdata = fit.newdata,probs = c(0.025, 0.1,
                                                  0.9, 0.975))

fitsies

fitsies.list <- list()
for(i in 1:dim(fitsies)[3]){
  fitsies.list[[i]] <- fitsies[,,i]
}

life_fits_df <- do.call(rbind, fitsies.list) %>%
  as.data.frame() %>%
  mutate(method = rep(fit.newdata$method, 4),
         life = c(rep("All larvae aquatic", 2),
                  rep("Fully aquatic", 2),
                  rep("Fully terrestrial", 2),
                  rep("Some larvae aquatic", 2))) %>%
  group_by(life) %>%
  mutate(mean_p = mean(Estimate)) %>%
  ungroup() %>%
  mutate(life = fct_reorder(life, mean_p),
         method = case_when(method == "S" ~ "Spray",
                            method == "R" ~ "Roller"))
         
(life_plot2 <- 
life_fits_df %>%
  ggplot() +
  geom_point(aes(y = life, x = 100*Estimate, color = method, 
                 group = method),
             position = position_dodge(width = .75),
             size = 2) +
  geom_errorbar(aes(y = life, xmin = 100*Q10, 
                    xmax = 100*Q90, color = method,
                    group = method),
                size = 1,
                position = position_dodge(width = .75),
                width = 0) +
  geom_errorbar(aes(y = life, xmin = 100*Q2.5, 
                    xmax = 100*Q97.5, color = method,
                    group = method),
                position = position_dodge(width = .75),
                width = 0) +
  scale_color_manual(values = c("steelblue", "firebrick")) +
  labs(color = "",
       y = "",
       x = "% of families",
       title = "Life mode") +
  theme_bw() +
  theme(text = element_text(size = 13),
        legend.position = c(.8,.2))
)

### Body size

# load the models so we don't have to refit them each time
size.method.tree <- readRDS("output/size_multinomial_method.tree.rds")
size.method <- readRDS("output/size_multinomial_method.rds")
size.tree <- readRDS("output/size_multinomial_tree.rds")

# plot top model predictions
fit.newdata = expand.grid(method = c("S", "R"),
                                 size = 1)

fitsies <- fitted(size.method, 
                  newdata = fit.newdata,probs = c(0.025, 0.1,
                                                  0.9, 0.975))

fitsies

fitsies.list <- list()
for(i in 1:dim(fitsies)[3]){
  fitsies.list[[i]] <- fitsies[,,i]
}

size_fits_df <- do.call(rbind, fitsies.list) %>%
  as.data.frame() %>%
  mutate(method = rep(fit.newdata$method, 4),
         size = c(rep("< 5 mm", 2),
                  rep("> 20 mm", 2),
                  rep("10-20 mm", 2),
                  rep("5-10 mm", 2))) %>%
  group_by(size) %>%
  mutate(mean_p = mean(Estimate)) %>%
  ungroup() %>%
  mutate(size = fct_reorder(size, mean_p),
         method = case_when(method == "S" ~ "Spray",
                            method == "R" ~ "Roller"))
         
(size_plot2 <- 
size_fits_df %>%
  ggplot() +
  geom_point(aes(y = size, x = 100*Estimate, color = method, 
                 group = method),
             position = position_dodge(width = .5),
             size = 2) +
  geom_errorbar(aes(y = size, xmin = 100*Q10, 
                    xmax = 100*Q90, color = method,
                    group = method),
                size = 1,
                position = position_dodge(width = .5),
                width = 0) +
  geom_errorbar(aes(y = size, xmin = 100*Q2.5, 
                    xmax = 100*Q97.5, color = method,
                    group = method),
                position = position_dodge(width = .5),
                width = 0) +
  scale_color_manual(values = c("steelblue", "firebrick")) +
  labs(color = "",
       y = "",
       x = "% of families",
       title = "Body size") +
  theme_bw() +
  theme(text = element_text(size = 13),
        legend.position = c(.8,.2))
)

### Combine all 3 plots with patchwork
feed_plot2 / life_plot2 / size_plot2

ggsave("figures/PlotL_guilds_multinom3.jpg", 
       width = 5, height = 12, dpi = 400)
```






# Plots of rarefied species richness, shannon & simpson indexes, relative read frequency by primer
```{r}
# MOTU rarefied species richness by sample and primer set
names(art_order) # to get column numbers of matrix
data_cols <- 8:44
coleop_rare_mat <- art_full %>%
  filter(primer == "coleop") %>%
  select(data_cols) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame()

zeale_rare_mat <- art_order %>%
  filter(primer == "zeale") %>%
  select(data_cols) %>%
  as.matrix() %>%
  t() %>%
  as.data.frame()

apply(coleop_rare_mat, 1, sum)
sort(apply(coleop_rare_mat, 1, sum))
apply(zeale_rare_mat, 1, sum)
sort(apply(zeale_rare_mat, 1, sum))

# get lowest number of reads, excluding the 1 sample with only 195
minreads <- 3660

rarefy_coleop <- rarefy(coleop_rare_mat[-9,], minreads, se = T)
rarefy_zeale <- rarefy(zeale_rare_mat[-9,], minreads, se = T)

rarefy_coleop_df <- rarefy_coleop %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_coleop),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "Coleop")

rarefy_zeale_df <- rarefy_zeale %>%
  t() %>%
  as.data.frame() %>%
  bind_cols(colnames(rarefy_zeale),.) %>%
  rename(sample = 1) %>%
  mutate(primer = "Zeale")

rarefy_df <- rarefy_coleop_df %>%
  bind_rows(rarefy_zeale_df)

rarefy_df_wide <- data.frame(
  sample = colnames(rarefy_coleop),
  coleop_S = rarefy_coleop[1,],
  zeale_S = rarefy_zeale[1,],
  coleop_se = rarefy_coleop[2,],
  zeale_se = rarefy_zeale[2,])

# now create a rarified version of the read matrixes for diversity metrics

coleop_rare_mat_rare <- rrarefy(coleop_rare_mat[-9,], minreads)
zeale_rare_mat_rare <- rrarefy(zeale_rare_mat[-9,], minreads)

# Calculate diversity metrics
coleop_rare_shannon <- diversity(coleop_rare_mat[-9,], 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat[-9,]),.) %>%
  rename(sample = 1, coleop_shannon = 2)

zeale_rare_shannon <- diversity(zeale_rare_mat[-9,], 
                               index = "shannon", 
                               MARGIN = 1, base = exp(1)) %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat[-9,]),.) %>%
  rename(sample = 1, zeale_shannon = 2)

shannon_df <- coleop_rare_shannon %>%
  bind_cols(select(zeale_rare_shannon, -sample))

# Calculate Simpsons index on rarefied data
coleop_rare_simpson <- diversity(coleop_rare_mat[-9,], 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(coleop_rare_mat[-9,]),.) %>%
  rename(sample = 1, coleop_simpson = 2)

zeale_rare_simpson <- diversity(zeale_rare_mat[-9,], 
                               index = "simpson") %>%
  as.data.frame() %>%
  bind_cols(rownames(zeale_rare_mat[-9,]),.) %>%
  rename(sample = 1, zeale_simpson = 2)

simpson_df <- coleop_rare_simpson %>%
  bind_cols(select(zeale_rare_simpson, -sample))

# set font size for all plot annotations below
annfont <- 2
ytextsize = 10
xtextsize = 11

# Richness boxplots
rarefy_stats <- rarefy_df_wide %>%
  mutate(lc = log(coleop_S),
         lz = log(zeale_S)) %>%
  summarize(mean_lc = mean(lc),
            sd_lc = sd(lc),
            n_lc = length(lc),
            se_lc = sd_lc/sqrt(n_lc),
            lcl_lc = mean_lc - 2*se_lc,
            ucl_lc = mean_lc + 2*se_lc,
            mean_lz = mean(lz),
            sd_lz = sd(lz),
            n_lz = length(lz),
            se_lz = sd_lz/sqrt(n_lz),
            lcl_lz = mean_lz - 2*se_lz,
            ucl_lz = mean_lz + 2*se_lz)

exp(rarefy_stats)

(rareS.plot <- 
rarefy_df_wide %>%
  select(-ends_with("_se")) %>%
  tidyr::pivot_longer(cols = ends_with("_S")) %>%
  mutate(name = case_when(name == "coleop_S" ~ "Coleop",
                          name == "zeale_S" ~ "Zeale")) %>%
  ggplot(aes(x = name, y = value)) +
  geom_line(size = .5, alpha = .5, 
            aes(group = sample), 
            color = "darkgray") +
  geom_violin(draw_quantiles = .5, aes(fill = name),
              alpha = 0.6) +
  geom_point(size = 4, alpha = .5, aes(color = name)) +  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize)) +
  labs(x = "",
       y = "Rarefied MOTU richness") +
  guides(fill = "none", color = "none") +
  annotate("text", x = 2.5, y = 160, 
           label = paste0("Means (log-normal)",
           "\nColeop: ",
           round(exp(rarefy_stats$mean_lc),1), 
           ", 95% CI = [",
           round(exp(rarefy_stats$lcl_lc),1),
           ", ",
           round(exp(rarefy_stats$ucl_lc),1), 
           "]\n",
           "Zeale: ",
           round(exp(rarefy_stats$mean_lz),1),
           ", 95% CI = [",
           round(exp(rarefy_stats$lcl_lz),1),
           ", ",
           round(exp(rarefy_stats$ucl_lz),1),
           "]"),
           hjust = 1,
           size = annfont,
           color = "darkgray")
)
  
# Evenness boxplots
simpson_stats <- simpson_df %>%
  mutate(lc = coleop_simpson,
         lz = zeale_simpson) %>%
  summarize(mean_lc = mean(lc),
            sd_lc = sd(lc),
            n_lc = length(lc),
            se_lc = sd_lc/sqrt(n_lc),
            lcl_lc = mean_lc - 2*se_lc,
            ucl_lc = mean_lc + 2*se_lc,
            mean_lz = mean(lz),
            sd_lz = sd(lz),
            n_lz = length(lz),
            se_lz = sd_lz/sqrt(n_lz),
            lcl_lz = mean_lz - 2*se_lz,
            ucl_lz = mean_lz + 2*se_lz)

simpson_stats

(simpson.plot <- 
simpson_df %>%
  tidyr::pivot_longer(cols = ends_with("_simpson")) %>%
  mutate(name = case_when(name == "coleop_simpson" ~ "Coleop",
                          name == "zeale_simpson" ~ "Zeale")) %>%
  ggplot(aes(x = name, y = value)) +
  geom_line(size = .5, alpha = .5, 
            aes(group = sample), 
            color = "darkgray") +
  geom_violin(draw_quantiles = .5, aes(fill = name),
              alpha = 0.6) +
  geom_point(size = 4, alpha = .5, aes(color = name)) +  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize)) +
  labs(x = "",
       y = "MOTU evenness (Simpson)") +
  guides(fill = "none", color = "none") +
  annotate("text", x = 2.5, y = 1.3,
           label = paste0("Means",
           "\nColeop: ",
           round(simpson_stats$mean_lc,2),
           ", 95% CI = [",
           round(simpson_stats$lcl_lc,2),
           ", ",
           round(simpson_stats$ucl_lc,2),
           "]\n",
           "Zeale: ",
           round(simpson_stats$mean_lz,2),
           ", 95% CI = [",
           round(simpson_stats$lcl_lz,2),
           ", ",
           round(simpson_stats$ucl_lz,2),
           "]"),
           hjust = 1,
           vjust = 1,
           size = annfont,
           color = "darkgray")
)
  
# Diversity boxplots
shannon_stats <- shannon_df %>%
  mutate(lc = coleop_shannon,
         lz = zeale_shannon) %>%
  summarize(mean_lc = mean(lc),
            sd_lc = sd(lc),
            n_lc = length(lc),
            se_lc = sd_lc/sqrt(n_lc),
            lcl_lc = mean_lc - 2*se_lc,
            ucl_lc = mean_lc + 2*se_lc,
            mean_lz = mean(lz),
            sd_lz = sd(lz),
            n_lz = length(lz),
            se_lz = sd_lz/sqrt(n_lz),
            lcl_lz = mean_lz - 2*se_lz,
            ucl_lz = mean_lz + 2*se_lz)

shannon_stats

(shannon.plot <- 
shannon_df %>%
  tidyr::pivot_longer(cols = ends_with("_shannon")) %>%
  mutate(name = case_when(name == "coleop_shannon" ~ "Coleop",
                          name == "zeale_shannon" ~ "Zeale")) %>%
  ggplot(aes(x = name, y = value)) +
  geom_line(size = .5, alpha = .5, 
            aes(group = sample), 
            color = "darkgray") +
  geom_violin(draw_quantiles = .5, aes(fill = name),
              alpha = 0.6) +
  geom_point(size = 4, alpha = .5, aes(color = name)) +  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.title.y = element_text(size = ytextsize),
        axis.text.x = element_text(size = xtextsize)) +
  labs(x = "",
       y = "MOTU diversity (Shannon)") +
  guides(fill = "none", color = "none") +
  annotate("text", x = 2.5, y = 4.8,
           label = paste0("Means",
           "\nColeop: ",
           round(shannon_stats$mean_lc,2),
           ", 95% CI = [",
           round(shannon_stats$lcl_lc,2),
           ", ",
           round(shannon_stats$ucl_lc,2),
           "]\n",
           "Zeale: ",
           round(shannon_stats$mean_lz,2),
           ", 95% CI = [",
           round(shannon_stats$lcl_lz,2),
           ", ",
           round(shannon_stats$ucl_lz,2),
           "]"),
           hjust = 1,
           vjust = 1,
           size = annfont,
           color = "darkgray")
)

#################
#### Plotting relative read abundance by order (bar chart)
#################

# set up plotting order df
order_df <- data.frame(cat = unique(paste0(art_full$class, "-", 
                                      art_full$order)),
           order = c(2,4,1,3,6,15,9,7,10,12,8,23,
                     14,13,11,22,16,24,5,20,19,17,21,18),
           name = c("Spiders",
                    "Unknown Arachnids",
                    "Sarcoptiform mites",
                    "Trombidiform mites",
                    "Collembola",
                    "Cockroaches",
                    "Beetles",
                    "Flies",
                    "True bugs",
                    "Bees, wasps, & ants",
                    "Butterflies & moths",
                    "Unknown insects",
                    "Lacewings",
                    "Crickets & grasshoppers",
                    "Bark lice",
                    "Fleas",
                    "Caddisflies",
                    "Unknown arthropods",
                    "Centipedes",
                    "Mayflies",
                    "Mantids",
                    "Dragonflies & damselflies",
                    "Walking sticks",
                    "Thrips"))
                    

names(art_full)
primer_who <- art_full %>%
  mutate(cat = paste0(class, "-", order)) %>%
  select(primer, cat) %>%
  bind_cols(total.reads = rowSums(art_full[,8:44])) %>%
  group_by(primer, cat) %>%
  summarize(total.reads = sum(total.reads),
            .groups = "drop") %>%
  group_by(primer) %>%
  mutate(rel.reads = 100*total.reads / sum(total.reads)) %>%
  ungroup() %>%
  # this part adds primer-taxa categories that had no reads
  right_join(expand.grid(cat = unique(paste0(art_full$class, "-", 
                                      art_full$order)),
                         primer = c("coleop", "zeale"))) %>%
  # this part converts the NAs in the previous step to 0's
  mutate(rel.reads = tidyr::replace_na(rel.reads, 0),
         primer = case_when(primer == "coleop" ~ "Coleop",
                            primer == "zeale" ~ "Zeale")) %>%
  group_by(cat) %>%
  mutate(mean_rel.reads = mean(rel.reads)) %>%
  ungroup() %>%
  left_join(order_df) %>%
  mutate(name = forcats::fct_reorder(name, order, .desc = T))

(who_plot <- 
primer_who %>%
  ggplot() +
  geom_col(aes(y = name, x = sqrt(rel.reads),
               fill = primer),
           position = position_dodge(),
           alpha = 0.6) +
  scale_fill_manual(values = wes_palette("Cavalcanti1")) +
  scale_color_manual(values = wes_palette("Cavalcanti1")) +
  scale_x_continuous(breaks = c(0,2,4,6),
                     labels = c(0,4,16,36)) +
  theme_bw() +
  theme(text = element_text(size = 14),
        axis.text.y = element_text(size = 10),
        legend.position = c(.7,.1)) +
  labs(x = "% of total reads",
       y = "",
       fill = "")# +
  # geom_text(aes(y = name, x = rel.reads + 1.5, 
  #               color = primer, 
  #               label = round(rel.reads,2)),
  #           position = position_dodge(width = 1),
  #           size = 1.5,
  #           data = filter(primer_who, rel.reads < 1))
)

# Stitch together primer plots
patchobj <- who_plot | (rareS.plot / simpson.plot / shannon.plot)

ggsave("figures/primer_compare6.png", width = 7, height = 7, dpi = 400)

```

