# read in data and load libraries
```{r}
library(dplyr)
library(data.table)
'%notin%' <- Negate('%in%')

### Zeale ###

# Read in Zeale (COI) primer results (all raw data = 560 MOTUs)
z0 <- fread("data/merged.uni.c10.l130.L185.sht.srt.nochi.1line.swarm1.fix.tag.ann.srt.Z.tsv") %>%
  rename(match = 3) 

z <- z0 %>%
  filter(match >= 0.80)
# 649 of 819 left

# 138 are well matched to species already
test <- z %>% filter(match >= 0.99,
             !is.na(species_name))
nrow(test)

# 123 unique species names (before BLAST and manual curation)
sort(unique(test$scientific_name))
rm(test)

# that leaves 511 to blast if desired (to format for this, see make_blast.Rmd)
zblast <- z %>% 
  filter(!(match >= 0.99 & !is.na(species_name)))

# Read in Zeale (COI) primer read counts by sample
zc <- fread("data/merged.uni.c10.l130.L185.sht.srt.nochi.1line.swarm1.Z_output.counts.csv")

zz <- z %>%
  left_join(zc, by = join_by(id, definition, sequence))

### Coleop ###

# Read in Coleop (16S) primer results
c0 <- fread("data/merged.uni.c10.l75.L125.sht.srt.nochi.1line.swarm1.fix.tag.ann.srt.C.tsv") %>%
  rename(match = 3)

c <- c0 %>%
  filter(match >= 0.80)
# 1084 of 1179 left

# 116 are well matched to species already
test2 <- c %>% filter(match >= 0.99,
             !is.na(species_name))
nrow(test2)

# 116 unique species names (before BLAST and manual curation)
sort(unique(test2$scientific_name))
rm(test2)

# that leaves 968 to blast
cblast <- c %>% 
  filter(!(match >= 0.99 & !is.na(species_name)))

# Read in Zeale (COI) primer read counts by sample
c2 <- fread("data/merged.uni.c10.l75.L125.sht.srt.nochi.1line.swarm1.C_output.counts.csv")

cc <- c %>%
  left_join(c2, by = join_by(id, definition, sequence))

```
# make fasta files to blast (if applicable)
See make_blast.Rmd for code to do this. Otherwise, skip to "separate the samples and negatives". The output from performing blastn on these sequences is a series of text files that are combined in the next step.
```{r}

# Zeale

# data.frame of the sequences to be blasted
input_db <- zblast
# Define names for the output files
output_db <- "data/merged.uni.c10.l30.L185.sht.srt.nochi.1line.swarm1.fix.tag.ann.srt.bla.Z.fasta"

# Delete previous files
db_file <- file(output_db,open = "wt")

closeAllConnections()

# Loop to write reference fasta file that will work with ecotag
showlines <- 10

for (i in 1:nrow(input_db)) {

        # write it in db_file
        output_file <- file(output_db, open = "at")
        writeLines(text = paste0(">",as.character(input_db$id[i])),
                   con = output_file)
        writeLines(text = gsub("-","",as.character(input_db$sequence[i])), con = output_file)
        close(output_file)
        
  if (i %% showlines == 0) message(i,"/",nrow(input_db)," sequences processed.","\r",appendLF = FALSE)
}

# Coleop

# data.frame of the sequences to be blasted
input_db <- cblast
# Define names for the output files
output_db <- "data/merged.uni.c10.l75.L125.sht.srt.nochi.1line.swarm1.fix.tag.ann.srt.bla.C.fasta"

# Delete previous files
db_file <- file(output_db,open = "wt")

closeAllConnections()

# Loop to write reference fasta file that will work with ecotag
showlines <- 10

for (i in 1:nrow(input_db)) {

        # write it in db_file
        output_file <- file(output_db, open = "at")
        writeLines(text = paste0(">",as.character(input_db$id[i])),
                   con = output_file)
        writeLines(text = gsub("-","",as.character(input_db$sequence[i])), con = output_file)
        close(output_file)
        
  if (i %% showlines == 0) message(i,"/",nrow(input_db)," sequences processed.","\r",appendLF = FALSE)
}


```
# stitch together the output of the blast searches (if applicable)
If not using blast, then skip to "separate the samples and negatives".
Headers of the tables: https://www.metagenomics.wiki/tools/blast/blastn-output-format-6
```{r}

### Zeale ###

# make a list of the file names for each sequence
bzf <- list.files("data/blast.Zswarm1")[!grepl(x = list.files("data/blast.Zswarm1"), pattern = ">")]

# make a blank dataframe to collect the info on each blast
blast.Z.output <- data.frame(sequence = rep("", length(bzf)),
                             bestblst = rep(0, length(bzf)),
                             blast100 = rep("", length(bzf)),
                             blast99 = rep("", length(bzf)),
                             blast98 = rep("", length(bzf)),
                             blast95 = rep("", length(bzf)),
                             blast90 = rep("", length(bzf)),
                             blast80 = rep("", length(bzf)))

# loop through the blast tables from each sequence and extract info
for(i in 1:length(bzf)){

# insert sequence
blast.Z.output[i,1] <- 
  gsub(gsub(bzf[i], pattern = "results_", replacement = ""), pattern = ".txt", replacement = "")  

# only proceed if the file isn't empty
if (nrow(fread(paste0("data/blast.Zswarm1/", bzf[i]))) > 0) { 
bzt <- fread(paste0("data/blast.Zswarm1/", bzf[i])) %>%
  select(sseqid = 2, match = 3, length = 4, evalue = 11, qlen = 13,
         taxid = 15, sci = 16) %>%
  mutate(qcov = length/qlen) %>%
  filter(qcov >= 0.95)

# populate the data fields
blast.Z.output[i,2] <- if(nrow(bzt)>0){max(bzt$match)} else{NA}

bl100 <- bzt %>% 
  filter(match >= 100) %>%
  select(sci) %>%
  distinct()

bl99 <- bzt %>% 
  filter(match >= 99) %>%
  select(sci) %>%
  distinct()

bl98 <- bzt %>% 
  filter(match >= 98) %>%
  select(sci) %>%
  distinct()

bl95 <- bzt %>% 
  filter(match >= 95) %>%
  select(sci) %>%
  distinct()

bl90 <- bzt %>% 
  filter(match >= 90) %>%
  select(sci) %>%
  distinct()

bl80 <- bzt %>% 
  filter(match >= 80) %>%
  select(sci) %>%
  distinct()

blast.Z.output[i,3] <- paste(bl100$sci, collapse = "; ")
blast.Z.output[i,4] <- paste(bl99$sci, collapse = "; ")
blast.Z.output[i,5] <- paste(bl98$sci, collapse = "; ")
blast.Z.output[i,6] <- paste(bl95$sci, collapse = "; ")
blast.Z.output[i,7] <- paste(bl90$sci, collapse = "; ")
blast.Z.output[i,8] <- paste(bl80$sci, collapse = "; ")
} # END IF (data table is empty)
} # END LOOP 

### Coleop ###

# make a list of the file names for each sequence
bcf <- list.files("data/blast.Cswarm1")[!grepl(x = list.files("data/blast.Cswarm1"), pattern = ">")]

# make a blank dataframe to collect the info on each blast
blast.C.output <- data.frame(sequence = rep("", length(bcf)),
                             bestblst = rep(0, length(bcf)),
                             blast100 = rep("", length(bcf)),
                             blast99 = rep("", length(bcf)),
                             blast98 = rep("", length(bcf)),
                             blast95 = rep("", length(bcf)),
                             blast90 = rep("", length(bcf)),
                             blast80 = rep("", length(bcf)))

# loop through the blast tables from each sequence and extract info
for(i in 1:length(bcf)){

# insert sequence
blast.C.output[i,1] <- 
  gsub(gsub(bcf[i], pattern = "results_", replacement = ""), pattern = ".txt", replacement = "")  

# only proceed if the file isn't empty
if (nrow(fread(paste0("data/blast.Cswarm1/", bcf[i]))) > 0) { 
bct <- fread(paste0("data/blast.Cswarm1/", bcf[i])) %>%
  select(sseqid = 2, match = 3, length = 4, evalue = 11, qlen = 13,
         taxid = 15, sci = 16) %>%
  mutate(qcov = length/qlen) %>%
  filter(qcov >= 0.95)

# populate the data fields
blast.C.output[i,2] <- if(nrow(bct)>0){max(bct$match)} else{NA}

bl100 <- bct %>% 
  filter(match >= 100) %>%
  select(sci) %>%
  distinct()

bl99 <- bct %>% 
  filter(match >= 99) %>%
  select(sci) %>%
  distinct()

bl98 <- bct %>% 
  filter(match >= 98) %>%
  select(sci) %>%
  distinct()

bl95 <- bct %>% 
  filter(match >= 95) %>%
  select(sci) %>%
  distinct()

bl90 <- bct %>% 
  filter(match >= 90) %>%
  select(sci) %>%
  distinct()

bl80 <- bct %>% 
  filter(match >= 80) %>%
  select(sci) %>%
  distinct()

blast.C.output[i,3] <- paste(bl100$sci, collapse = "; ")
blast.C.output[i,4] <- paste(bl99$sci, collapse = "; ")
blast.C.output[i,5] <- paste(bl98$sci, collapse = "; ")
blast.C.output[i,6] <- paste(bl95$sci, collapse = "; ")
blast.C.output[i,7] <- paste(bl90$sci, collapse = "; ")
blast.C.output[i,8] <- paste(bl80$sci, collapse = "; ")
} # END IF (data table is empty)
} # END LOOP 

```
# combine ecotag and blast info into a csv for manual final species determination
If not using blast, then skip to "separate the samples and negatives".
```{r}

### Zeale ###

# Write csv file for manually overriding ecotag identifications with BLAST where appropriate
zzblastID <- zz %>%
  left_join(blast.Z.output, by = join_by(sequence)) %>%
  select(id, reads = total_reads, wt = cluster_weight, match, bestblst, 
         ecotagID = scientific_name, blast100:blast80, splist = starts_with("species_list"), 
         order_name, family_name, genus_name, species_name, sequence) %>%
  mutate(bestblst = bestblst / 100,
         final_sci = "",
         blastnotes = "",
         gbif = "") %>%
  arrange(desc(bestblst)) %>%
  arrange(desc(match)) %>%
  select(id, reads, wt, ecotagmatch = match, bestblastmatch = bestblst, ecotagID, blast100:blast80, 
         splist, ecotag_order = order_name, ecotag_family = family_name, 
         ecotag_genus = genus_name, overrideID = final_sci, blastnotes, gbif, sequence) 
# write.csv(zzblastID, "data/zzblastID_swarm1b.csv", row.names = F)

# Now you can curate the final species list MOTU by MOTU 
  # (choosing between blast and ecotag hits at XX level of match etc.)
  # I recommend renaming the file something like blastID_finaltax.csv
  # That way you don't accidentally overwrite it if you ran the code again.

### Coleop ###

# Write csv file for manually overriding ecotag identifications with BLAST where appropriate
ccblastID <- cc %>%
  left_join(blast.C.output, by = join_by(sequence)) %>%
  select(id, reads = total_reads, wt = cluster_weight, match, bestblst, 
         ecotagID = scientific_name, blast100:blast80, splist = starts_with("species_list"),
         order_name, family_name, genus_name, species_name, sequence) %>%
  mutate(bestblst = bestblst / 100,
         overrideID = "",
         blastnotes = "",
         gbif = "") %>%
  arrange(desc(bestblst)) %>%
  arrange(desc(match)) %>%
  select(id, reads, wt, ecotagmatch = match, bestblastmatch = bestblst, ecotagID, blast100:blast80, splist, 
         ecotag_order = order_name, ecotag_family = family_name, ecotag_genus = genus_name,
         overrideID, blastnotes, gbif, sequence)
# write.csv(ccblastID, "data/ccblastID_swarm1b.csv", row.names = F)

# Now you can curate the final species list MOTU by MOTU 
  # (choosing between blast and ecotag hits at XX level of match etc.)
  # I recommend renaming the file something like blastID_finaltax.csv
  # That way you don't accidentally overwrite it if you ran the code again.
```
# check geography of species & genus level IDs
In this step, I opened the files just created (e.g., "zzblastID_swarm1_finaltax.csv"). Where bestblastmatch > ecotagmatch, I reviewed the BLAST hits to see if they conflicted with the ecotagID. If so, and the match was much better (e.g., a clear, 100% BLAST match to a species but a different ecotagID based on a 93% match), I entered the BLAST match in the overrideID column. If BLAST and ecotag disagreed and they were both at a similar match level, or if BLAST returned multiple similar matches, I entered the nearest common ancestor taxon of the conflicting IDs in the overrideID column. I used only recognized NCBI taxon names in the overrideID column.

# check geography of species & genus level IDs
As an additional step to 'reality check' the data, I examined range maps for species- and genus-level IDs in GBIF (gbif.org). Some closely related species may share the same amplicon sequence with a 99 or 100% match. If they are from different parts of the world, and the local species isn't in the database, a species-level ID that is out of range may occur. It isn't always clear what species the real match is in those cases. So, in these cases, I demoted the ID in the overrideID column to the nearest common ancestor taxa that did occur in North America according to GBIF (e.g., change species-level to genus-level). To get maps to review, I used the make_list_to_gbif.R script to create a list of species and/or genera to look up. A python script (see code at end of the R script) was then used to scrape the GBIF website for the maps (python code was run in spyder). That sped up the process of looking up and reviewing the maps.

# read in the final manually curated MOTU-level identifications, create 'finaltax' column, & join to MOTU read count data
At this point, the identifications were manually curated (reivewed against BLAST matches and GBIF range maps) and the csv file (e.g., "zzblastID_swarm1_finaltax.csv") had a complete column named overrideID. That column contained the appropriate NCBI taxon name in cases where BLAST or geography caused a manual override of the ecotag ID. The script below selects the appropriate final ID (ecotagID or overrideID) and puts it in a new column called finaltax. That column was then used to look up the higher-level taxonomy using the taxize package. Note that 'finaltax' was not necessarily at an appropriate taxonomic level given the strength of the match (e.g., it may have been ID'd to species by ecotag with only a 95% match). Later, the taxonomic resolution will be manually 'coarsened' where appropriate based on defined match thresholds.
```{r}

### Zeale ###

zzblastID.cur <- read.csv("data/zzblastID_swarm1_finaltax.csv")

zz_fin <- zz %>%
  left_join(select(zzblastID.cur, id, bestblastmatch, overrideID, blastnotes)) %>%
  select(id, ecotagmatch = match, bestblastmatch, ecotagID = scientific_name, overrideID, 
         count = total_reads, wt = cluster_weight, starts_with("sample"), sequence, blastnotes) %>%
  mutate(finaltax = case_when(overrideID == "" ~ ecotagID,
                              is.na(overrideID) ~ ecotagID,
                              TRUE ~ overrideID),
         bestmatch = case_when(ecotagmatch >= bestblastmatch ~ ecotagmatch,
                               bestblastmatch > ecotagmatch ~ bestblastmatch,
                               is.na(bestblastmatch) ~ ecotagmatch))

### Coleop ###

ccblastID.cur <- read.csv("data/ccblastID_swarm1_finaltax.csv")

cc_fin <- cc %>%
  left_join(select(ccblastID.cur, id, bestblastmatch, overrideID, blastnotes)) %>%
  select(id, ecotagmatch = match, bestblastmatch, ecotagID = scientific_name, overrideID, 
         count = total_reads, wt = cluster_weight, starts_with("sample"), sequence, blastnotes) %>%
  mutate(finaltax = case_when(overrideID == "" ~ ecotagID,
                              is.na(overrideID) ~ ecotagID,
                              TRUE ~ overrideID),
         bestmatch = case_when(ecotagmatch >= bestblastmatch ~ ecotagmatch,
                               bestblastmatch > ecotagmatch ~ bestblastmatch,
                               is.na(bestblastmatch) ~ ecotagmatch))

```
# Look up higher taxonomy with taxize package & 'coarsen' final ID as appropriate
Note: there is surely a better way to do this (e.g., using a for loop), but the commented section below was used to get taxonomy in batches and save to csv files. It is broken up into batches as it sometimes throws an error midway and requires starting over. Note: you may need API key to run this; follow taxize instructions to do so.
```{r}

# library(taxize)
# 
# #### adding higher taxa
taxlistfull <- sort(unique(c(cc_fin$finaltax, zz_fin$finaltax)))
# taxlist1 <- taxlistfull[1:50]
# # add taxonomy to the lowest taxa rank based on NCBI database
# full_tax_lookup1 <- taxize::tax_name(sci = taxlist1,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup1, "data/full_tax_lookup1.csv", row.names = F)
# 
# taxlist2 <- taxlistfull[51:100]
# # add taxonomy to the lowest taxa rank based on NCBI database
# full_tax_lookup2 <- taxize::tax_name(sci = taxlist2,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup2, "data/full_tax_lookup2.csv", row.names = F)
# 
# taxlist3 <- taxlistfull[101:150]
# # add taxonomy to the lowest taxa rank based on NCBI database
# full_tax_lookup3 <- taxize::tax_name(sci = taxlist3,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup3, "data/full_tax_lookup3.csv", row.names = F)
# 
# taxlist4 <- taxlistfull[151:200]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup4 <- taxize::tax_name(sci = taxlist4,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup4, "data/full_tax_lookup4.csv", row.names = F)
# 
# taxlist5 <- taxlistfull[201:250]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax5_lookup5 <- taxize::tax_name(sci = taxlist5,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax5_lookup5, "data/full_tax_lookup5.csv", row.names = F)
# 
# taxlist6 <- taxlistfull[251:300]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup6 <- taxize::tax_name(sci = taxlist6,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup6, "data/full_tax_lookup6.csv", row.names = F)
# 
# taxlist7 <- taxlistfull[301:350]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup7 <- taxize::tax_name(sci = taxlist7,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup7, "data/full_tax_lookup7.csv", row.names = F)
# 
# taxlist8 <- taxlistfull[351:400]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup8 <- taxize::tax_name(sci = taxlist8,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup8, "data/full_tax_lookup8.csv", row.names = F)
# 
# taxlist9 <- taxlistfull[401:450]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup9 <- taxize::tax_name(sci = taxlist9,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup9, "data/full_tax_lookup9.csv", row.names = F)
# 
# taxlist10 <- taxlistfull[451:500]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup10 <- taxize::tax_name(sci = taxlist10,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup10, "data/full_tax_lookup10.csv", row.names = F)
# 
# taxlist11 <- taxlistfull[501:550]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup11 <- taxize::tax_name(sci = taxlist11,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup11, "data/full_tax_lookup11.csv", row.names = F)
# 
# taxlist12 <- taxlistfull[551:600]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup12 <- taxize::tax_name(sci = taxlist12,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup12, "data/full_tax_lookup12.csv", row.names = F)
# 
# taxlist13 <- taxlistfull[601:650]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup13 <- taxize::tax_name(sci = taxlist13,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup13, "data/full_tax_lookup13.csv", row.names = F)
# 
# taxlist14 <- taxlistfull[651:700]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup14 <- taxize::tax_name(sci = taxlist14,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup14, "data/full_tax_lookup14.csv", row.names = F)
# 
# taxlist15 <- taxlistfull[701:779]
# # add orders to the lowest taxa rank based on NCBI database
# full_tax_lookup15 <- taxize::tax_name(sci = taxlist15,
#                    get = c("species", "genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_tax_lookup15, "data/full_tax_lookup15.csv", row.names = F)

# full_taxa_lookup <- read.csv("data/full_tax_lookup1.csv") %>%
#   bind_rows(read.csv("data/full_tax_lookup2.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup3.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup4.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup5.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup6.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup7.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup8.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup9.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup10.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup11.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup12.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup13.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup14.csv")) %>%
#   bind_rows(read.csv("data/full_tax_lookup15.csv")) %>%
#   filter(!is.na(kingdom)) %>%
#   rename(finaltax = query)

# write.csv(full_taxa_lookup, "data/full_tax_lookup.csv", row.names = F)
full_taxa_lookup <- read.csv("data/full_tax_lookup.csv") %>%
  distinct()

# check that some change to the taxonomy in the manual taxonomy curation files did not result in a species not in the lookup table
full_taxa_lookup$finaltax[full_taxa_lookup$finaltax %notin% taxlistfull]

```
# separate the samples and negatives and adjust taxonomy based on match score 
```{r}

#### Zeale ####

# add final taxonomic IDs

zz_finID <- zz_fin  %>%
  left_join(full_taxa_lookup, by = join_by(finaltax)) %>%
  select(id, finaltax, ecotagID, ecotagmatch, bestblastmatch, bestmatch, phylum, class, order, family, genus, species,
         count, wt, starts_with("sample"), sequence, blastnotes) %>%
  mutate(genus = case_when(bestmatch < 0.98 ~ NA,
                           TRUE ~ genus),
         species = case_when(bestmatch < 0.99 ~ NA,
                             TRUE ~ species),
         family = case_when(bestmatch < 0.95 ~ NA,
                            TRUE ~ family),
         order = case_when(bestmatch < 0.9 ~ NA,
                           TRUE ~ order),
         finalID = case_when(is.na(species) & is.na(genus) & is.na(family) & is.na(order) & is.na(class) ~ phylum,
                                    is.na(species) & is.na(genus) & is.na(family) & is.na(order) & !is.na(class) ~ class,
                                    is.na(species) & is.na(genus) & is.na(family) & !is.na(order) ~ order,
                                    is.na(species) & is.na(genus) & !is.na(family) ~ family,
                                    is.na(species) & !is.na(genus) ~ genus,
                                    !is.na(species) ~ species)) %>%
  select(id, finalID, bestmatch, phylum, class, order, family, genus, species,
         count, wt, starts_with("sample"), sequence) 

# check which columns are negatives
colnames(zz_finID) # input those column numbers in the next line
negatives <- c(12,13,21,51:53)
colnames(as.data.frame(zz_finID)[, negatives])

# select out just the samples
zs <- zz_finID %>%
  select(-all_of(negatives))

# sum the reads across all samples
sum(select(zs, starts_with("sample"))) # 4119060

# select out just the negatives
zn <- zz_finID %>%
  select(id, finalID, bestmatch, phylum, class, order, family, genus, species,
         count, wt, all_of(negatives), sequence)

# sum the reads across all negatives
sum(select(zn, starts_with("sample"))) # 997

# get the max read count of each seqid present in the negatives
colnames(zn) # check which columns contain negative samples
zn$seqmax <- apply(zn[,12:17],1, max)

# write.csv(zs, file = "data/pinebugs.samples.step0.nofilters.swarm1.Z.csv", row.names = F)
# write.csv(zn, file = "data/pinebugs.negatives.step0.nofilters.swarm1.Z.csv", row.names = F)

#### Coleop ####

# add final taxonomic IDs

cc_finID <- cc_fin  %>%
  left_join(full_taxa_lookup, by = join_by(finaltax)) %>%
  select(id, finaltax, ecotagmatch, bestblastmatch, bestmatch, phylum, class, order, family, genus, species,
         count, wt, starts_with("sample"), sequence, blastnotes) %>%
  mutate(genus = case_when(bestmatch < 0.98 ~ NA,
                           TRUE ~ genus),
         species = case_when(bestmatch < 0.99 ~ NA,
                             TRUE ~ species),
         family = case_when(bestmatch < 0.95 ~ NA,
                            TRUE ~ family),
         order = case_when(bestmatch < 0.9 ~ NA,
                           TRUE ~ order),
         finalID = case_when(is.na(species) & is.na(genus) & is.na(family) & is.na(order) & is.na(class) ~ phylum,
                                    is.na(species) & is.na(genus) & is.na(family) & is.na(order) & !is.na(class) ~ class,
                                    is.na(species) & is.na(genus) & is.na(family) & !is.na(order) ~ order,
                                    is.na(species) & is.na(genus) & !is.na(family) ~ family,
                                    is.na(species) & !is.na(genus) ~ genus,
                                    !is.na(species) ~ species)) %>%
  select(id, finalID, bestmatch, phylum, class, order, family, genus, species,
         count, wt, starts_with("sample"), sequence) 

# check which columns are negatives
colnames(cc_finID) # input those column numbers in the next line
negatives.C <- c(12,13,21,22,53:65)
colnames(as.data.frame(cc_finID)[,negatives.C])

# select out just the samples
cs <- cc_finID %>%
  select(-all_of(negatives.C))

# sum the reads across all samples
sum(select(cs, starts_with("sample"))) # 2218638

# select out just the negatives
cn <- cc_finID %>%
  select(id, finalID, bestmatch, phylum, class, order, family, genus, species,
         count, wt, all_of(negatives.C), sequence)

# sum the reads across all negatives
sum(select(cn, starts_with("sample"))) # 58965

# get the max read count of each seqid present in the negatives
colnames(cn) # check which columns contain negative samples
cn$seqmax <- apply(cn[,12:17],1, max)

# write.csv(cs, file = "data/pinebugs.samples.step0.nofilters.swarm1.C.csv", row.names = F)
# write.csv(cn, file = "data/pinebugs.negatives.step0.nofilters.swarm1.C.csv", row.names = F)
```
# remove humans & mice
```{r}

### Coleop

cs1 <- cs %>%
  filter(genus %notin% c("Mus", "Homo"))

# sum the reads across all samples
sum(select(cs1, starts_with("sample"))) # 2161052

cn1 <- cn %>%
  filter(genus %notin% c("Mus", "Homo"))

# sum the reads across all samples
sum(select(cn1, starts_with("sample"))) # 5306

### Zeale

zs1 <- zs %>%
  filter(genus %notin% c("Mus", "Homo"))

# sum the reads across all samples
sum(select(zs1, starts_with("sample"))) # 4119060

zn1 <- zn %>%
  filter(genus %notin% c("Mus", "Homo"))

# sum the reads across all samples
sum(select(zn1, starts_with("sample"))) # 997

```

# subtract contamination found in negative controls
```{r}

#### Zeale ####

# add the max reads by sequence in all the negatives
zs2a <- zs1 %>%
  left_join(select(zn1, id, seqmax), by = "id") %>%
  data.frame()
zs2 <- zs2a

colnames(zs2) # 12:47 are the samples

# zero out samples with less reads of a sequence than the max contamination among negative of that sequence
for(i in 12:47){
  tmp <- zs2a[,i] - zs2a$seqmax
  tmp2 <- as.numeric(ifelse(tmp<0,0,tmp))
  zs2[,i] <- tmp2
}

# count remaining reads in samples
sum(select(zs2, starts_with("sample"))) # 4116683
rm(zs2a)

# look at sequence lengths
hist(nchar(zs2$sequence))
# zs2$len <- nchar(zs2$sequence)

#### Coleop ####

# add the max reads by sequence in all the negatives
cs2a <- cs1 %>%
  left_join(select(cn1, id, seqmax), by = "id") %>%
  data.frame()
cs2 <- cs2a

colnames(cs2) # 12:48 are the samples

# zero out samples with less reads of a sequence than the max contamination among negative of that sequence
for(i in 12:48){
  tmp <- cs2a[,i] - cs2a$seqmax
  tmp2 <- as.numeric(ifelse(tmp<0,0,tmp))
  cs2[,i] <- tmp2
}

# count remaining reads in samples
sum(select(cs2, starts_with("sample"))) # 2158954
rm(cs2a)

# look at sequence lengths
hist(nchar(cs2$sequence))
# cs2$len <- nchar(cs2$sequence)

```
# remove non-arthropods
```{r}

### Zeale ###

# check on phylum list and look at MOTUs with no phylum
unique(zs2$phylum)
zs2[is.na(zs2$phylum), c("id", "finalID")] # should be only MOTUs that were best matched to some taxon above phylum

# filter out non-arthropod - samples
zs3 <- zs2 %>%
  filter(phylum == "Arthropoda")

# add a column tallying reads within just the samples
names(zs3)
zs3$sampsum <- apply(zs3[,12:47], 1, sum)

# sum the reads across all samples
sum(select(zs3, starts_with("sample"))) # 4116475

# filter out non-arthropod - negatives
zn3 <- zn1 %>%
  filter(phylum == "Arthropoda")

# add a column tallying reads within just the negatives
names(zn3)
zn3$negsum <- apply(zn3[,12:17], 1, sum)

# sum the reads across all negatives
sum(select(zn3, starts_with("sample"))) # 997

# write.csv(zs3, file = "data/pinebugs.samples.onlyarth.negsub.Z.csv", row.names = F)
# write.csv(zn3, file = "data/pinebugs.negatives.onlyarth.Z.csv", row.names = F)

### Coleop ###

# check on phylum list and look at MOTUs with no phylum
unique(cs2$phylum)
cs2[is.na(cs2$phylum), c("id", "finalID")] # should be only MOTUs that were best matched to some taxon above phylum

# filter out non-arthropod - samples
cs3 <- cs2 %>%
  filter(phylum == "Arthropoda")

# add a column tallying reads within just the samples
names(cs3)
cs3$sampsum <- apply(cs3[,12:48], 1, sum)

# sum the reads across all samples
sum(select(cs3, starts_with("sample"))) # 2068121

# filter out non-fish - negatives
cn3 <- cn1 %>%
  filter(phylum == "Arthropoda")

# add a column tallying reads within just the negatives
names(cn3)
cn3$negsum <- apply(cn3[,12:28], 1, sum)

# sum the reads across all negatives
sum(select(cn3, starts_with("sample"))) # 3402

# write.csv(cs3, file = "data/pinebugs.samples.onlyarth.negsub.C.csv", row.names = F)
# write.csv(cn3, file = "data/pinebugs.negatives.onlyarth.C.csv", row.names = F)

```

# tallying orders, families, species
```{r}

### Zeale ###

# what species; n = 196 (note: was 158 with swarm d = 7)
zs3 %>% filter(!is.na(species),
               sampsum > 0) %>% 
  select(species) %>%
  arrange(species) %>%
  distinct()

# what genus; n = 225 (note: was 192 with swarm d = 7)
zs3 %>% filter(!is.na(genus),
               sampsum > 0) %>% 
  select(genus) %>%
  arrange(genus) %>%
  distinct()

# what families; n = 115 (note: was 110 with swarm d = 7)
zs3 %>% filter(!is.na(family),
               sampsum > 0) %>% 
  select(family) %>%
  arrange(family) %>%
  distinct()

# what orders; n = 16 (note: same result with swarm d = 7)
zs3 %>% filter(!is.na(order)) %>% 
  select(order) %>%
  arrange(order) %>%
  distinct()

### Coleop ###

# what species; n = 110 (was 94 with swarm d = 5)
cs3 %>% filter(!is.na(species),
               sampsum > 0) %>% 
  select(species) %>%
  arrange(species) %>%
  distinct()

# what genus; n = 177 (was 154)
cs3 %>% filter(!is.na(genus),
               sampsum > 0) %>% 
  select(genus) %>%
  arrange(genus) %>%
  distinct()

# what families; n = 127 (was 123)
cs3 %>% filter(!is.na(family),
               sampsum > 0) %>% 
  select(family) %>%
  arrange(family) %>%
  distinct()

# what orders; n = 23 (was 23)
cs3 %>% filter(!is.na(order),
               sampsum > 0) %>% 
  select(order) %>%
  arrange(order) %>%
  distinct()

```