# read in data and load libraries
```{r}
library(dplyr)
library(data.table)
'%notin%' <- Negate('%in%')

### Zeale ###

# Read in Zeale (COI) primer results (all raw data = 560 MOTUs)
z0 <- fread("data/merged.uni.c10.l130.L185.sht.srt.nochi.1line.swarm7.fix3.tag.ann.srt.Z.tab.tsv") %>%
  rename(match = 3) 

z <- z0 %>%
  filter(match >= 0.80)
# 427 of 560 left

# 96 are well matched to species already
z %>% filter(match >= 0.98,
             !is.na(species_name)) %>% 
  tally()

# that leaves 331 to blast if desired (to format for this, see make_blast.Rmd)
zblast <- z %>% 
  filter(!(match >= 0.98 & !is.na(species_name)))

# Read in Zeale (COI) primer read counts by sample
zc <- fread("data/merged.uni.c10.l130.L185.sht.srt.nochi.1line.swarm7.Z_output.counts.csv")

zz <- z %>%
  left_join(zc, by = join_by(id, definition, sequence))

### Coleop ###

# Read in Coleop (16S) primer results
c <- fread("data/merged.uni.c10.l75.L125.sht.srt.nochi.1line.swarm5.fix3.tag.ann.srt.C.tab.tsv") %>%
  rename(match = 3) %>%
  filter(match >= 0.80)
# 687 of 773 left

# 107 are well matched to species already
c %>% filter(match >= 0.98,
             !is.na(species_name)) %>% 
  tally()

# that leaves 580 to blast
cblast <- c %>% 
  filter(!(match >= 0.98 & !is.na(species_name)))

# Read in Zeale (COI) primer read counts by sample
c2 <- fread("data/merged.uni.c10.l75.L125.sht.srt.nochi.1line.swarm5.C_output.counts.csv")

cc <- c %>%
  left_join(c2, by = join_by(id, definition, sequence))

```
# make fasta files to blast (if applicable)
See make_blast.Rmd for code to do this. Otherwise, skip to "separate the samples and negatives". The output from performing blastn on these sequences is a series of text files that are combined in the next step.
```{r}

# Zeale

# data.frame of the sequences to be blasted
input_db <- zblast
# Define names for the output files
output_db <- "data/merged.uni.c10.l30.L185.sht.srt.nochi.1line.swarm7.fix3.tag.ann.srt.bla.Z.fasta"

# Delete previous files
db_file <- file(output_db,open = "wt")

closeAllConnections()

# Loop to write reference fasta file that will work with ecotag
showlines <- 10

for (i in 1:nrow(input_db)) {

        # write it in db_file
        output_file <- file(output_db, open = "at")
        writeLines(text = paste0(">",as.character(input_db$id[i])),
                   con = output_file)
        writeLines(text = gsub("-","",as.character(input_db$sequence[i])), con = output_file)
        close(output_file)
        
  if (i %% showlines == 0) message(i,"/",nrow(input_db)," sequences processed.","\r",appendLF = FALSE)
}

# Coleop

# data.frame of the sequences to be blasted
input_db <- cblast
# Define names for the output files
output_db <- "data/merged.uni.c10.l75.L125.sht.srt.nochi.1line.swarm5.fix3.tag.ann.srt.bla.C.fasta"

# Delete previous files
db_file <- file(output_db,open = "wt")

closeAllConnections()

# Loop to write reference fasta file that will work with ecotag
showlines <- 10

for (i in 1:nrow(input_db)) {

        # write it in db_file
        output_file <- file(output_db, open = "at")
        writeLines(text = paste0(">",as.character(input_db$id[i])),
                   con = output_file)
        writeLines(text = gsub("-","",as.character(input_db$sequence[i])), con = output_file)
        close(output_file)
        
  if (i %% showlines == 0) message(i,"/",nrow(input_db)," sequences processed.","\r",appendLF = FALSE)
}


```
# stitch together the output of the blast searches (if applicable)
If not using blast, then skip to "separate the samples and negatives".
Headers of the tables: https://www.metagenomics.wiki/tools/blast/blastn-output-format-6
```{r}

### Zeale ###

# make a list of the file names for each sequence
bzf <- list.files("data/blast.Z")[!grepl(x = list.files("data/blast.Z"), pattern = ">")]

# make a blank dataframe to collect the info on each blast
blast.Z.output <- data.frame(sequence = rep("", length(bzf)),
                             bestblst = rep(0, length(bzf)),
                             blast100 = rep("", length(bzf)),
                             blast99 = rep("", length(bzf)),
                             blast98 = rep("", length(bzf)),
                             blast95 = rep("", length(bzf)),
                             blast90 = rep("", length(bzf)),
                             blast80 = rep("", length(bzf)))

# loop through the blast tables from each sequence and extract info
for(i in 1:length(bzf)){

# insert sequence
blast.Z.output[i,1] <- 
  gsub(gsub(bzf[i], pattern = "results_", replacement = ""), pattern = ".txt", replacement = "")  

# only proceed if the file isn't empty
if (nrow(fread(paste0("data/blast.Z/", bzf[i]))) > 0) { 
bzt <- fread(paste0("data/blast.Z/", bzf[i])) %>%
  select(sseqid = 2, match = 3, length = 4, evalue = 11, qlen = 13,
         taxid = 15, sci = 16) %>%
  mutate(qcov = length/qlen) %>%
  filter(qcov >= 0.95)

# populate the data fields
blast.Z.output[i,2] <- if(nrow(bzt)>0){max(bzt$match)} else{NA}

bl100 <- bzt %>% 
  filter(match >= 100) %>%
  select(sci) %>%
  distinct()

bl99 <- bzt %>% 
  filter(match >= 99) %>%
  select(sci) %>%
  distinct()

bl98 <- bzt %>% 
  filter(match >= 98) %>%
  select(sci) %>%
  distinct()

bl95 <- bzt %>% 
  filter(match >= 95) %>%
  select(sci) %>%
  distinct()

bl90 <- bzt %>% 
  filter(match >= 90) %>%
  select(sci) %>%
  distinct()

bl80 <- bzt %>% 
  filter(match >= 80) %>%
  select(sci) %>%
  distinct()

blast.Z.output[i,3] <- paste(bl100$sci, collapse = "; ")
blast.Z.output[i,4] <- paste(bl99$sci, collapse = "; ")
blast.Z.output[i,5] <- paste(bl98$sci, collapse = "; ")
blast.Z.output[i,6] <- paste(bl95$sci, collapse = "; ")
blast.Z.output[i,7] <- paste(bl90$sci, collapse = "; ")
blast.Z.output[i,8] <- paste(bl80$sci, collapse = "; ")
} # END IF (data table is empty)
} # END LOOP 

### Coleop ###

# make a list of the file names for each sequence
bcf <- list.files("data/blast.C")[!grepl(x = list.files("data/blast.C"), pattern = ">")]

# make a blank dataframe to collect the info on each blast
blast.C.output <- data.frame(sequence = rep("", length(bcf)),
                             bestblst = rep(0, length(bcf)),
                             blast100 = rep("", length(bcf)),
                             blast99 = rep("", length(bcf)),
                             blast98 = rep("", length(bcf)),
                             blast95 = rep("", length(bcf)),
                             blast90 = rep("", length(bcf)),
                             blast80 = rep("", length(bcf)))

# loop through the blast tables from each sequence and extract info
for(i in 1:length(bcf)){

# insert sequence
blast.C.output[i,1] <- 
  gsub(gsub(bcf[i], pattern = "results_", replacement = ""), pattern = ".txt", replacement = "")  

# only proceed if the file isn't empty
if (nrow(fread(paste0("data/blast.C/", bcf[i]))) > 0) { 
bct <- fread(paste0("data/blast.C/", bcf[i])) %>%
  select(sseqid = 2, match = 3, length = 4, evalue = 11, qlen = 13,
         taxid = 15, sci = 16) %>%
  mutate(qcov = length/qlen) %>%
  filter(qcov >= 0.95)

# populate the data fields
blast.C.output[i,2] <- if(nrow(bct)>0){max(bct$match)} else{NA}

bl100 <- bct %>% 
  filter(match >= 100) %>%
  select(sci) %>%
  distinct()

bl99 <- bct %>% 
  filter(match >= 99) %>%
  select(sci) %>%
  distinct()

bl98 <- bct %>% 
  filter(match >= 98) %>%
  select(sci) %>%
  distinct()

bl95 <- bct %>% 
  filter(match >= 95) %>%
  select(sci) %>%
  distinct()

bl90 <- bct %>% 
  filter(match >= 90) %>%
  select(sci) %>%
  distinct()

bl80 <- bct %>% 
  filter(match >= 80) %>%
  select(sci) %>%
  distinct()

blast.C.output[i,3] <- paste(bl100$sci, collapse = "; ")
blast.C.output[i,4] <- paste(bl99$sci, collapse = "; ")
blast.C.output[i,5] <- paste(bl98$sci, collapse = "; ")
blast.C.output[i,6] <- paste(bl95$sci, collapse = "; ")
blast.C.output[i,7] <- paste(bl90$sci, collapse = "; ")
blast.C.output[i,8] <- paste(bl80$sci, collapse = "; ")
} # END IF (data table is empty)
} # END LOOP 

```
# combine ecotag and blast info into a csv for manual final species determination
If not using blast, then skip to "separate the samples and negatives".
```{r}

### Zeale ###

# Write csv file for manually entering BLAST identifications
zzblastID <- zz %>%
  left_join(blast.Z.output, by = join_by(sequence)) %>%
  select(id, match, bestblst, scientific_name, blast100:blast80, order_name, family_name, genus_name, sequence) %>%
  mutate(bestblst = bestblst / 100,
         final_sci = "",
         blastnotes = "") %>%
  arrange(desc(bestblst)) %>%
  arrange(desc(match)) %>%
  select(id, match, bestblst, scientific_name, blast100:blast80, order_name, family_name, genus_name, final_sci, blastnotes, sequence) 
# write.csv(zzblastID, "data/zzblastID.csv", row.names = F)

# Now you can curate the final species list MOTU by MOTU 
  # (choosing between blast and ecotag hits at XX level of match etc.)
  # I recommend renaming the file something like blastID_finaltax.csv
  # That way you don't accidentally overwrite it if you ran the code again.

### Coleop ###

# Write csv file for manually entering BLAST identifications
ccblastID <- cc %>%
  left_join(blast.C.output, by = join_by(sequence)) %>%
  select(id, match, bestblst, scientific_name, blast100:blast80, order_name, family_name, genus_name, sequence) %>%
  mutate(bestblst = bestblst / 100,
         final_sci = "",
         blastnotes = "") %>%
  arrange(desc(bestblst)) %>%
  arrange(desc(match)) %>%
  select(id, match, bestblst, scientific_name, blast100:blast80, order_name, family_name, genus_name, final_sci, blastnotes, sequence) 
# write.csv(ccblastID, "data/ccblastID.csv", row.names = F)

# Now you can curate the final species list MOTU by MOTU 
  # (choosing between blast and ecotag hits at XX level of match etc.)
  # I recommend renaming the file something like blastID_finaltax.csv
  # That way you don't accidentally overwrite it if you ran the code again.
```
# read in the curated MOTU taxonomy data & join to MOTU data
This should have a column named final_sci which has the lowest appropriate taxon name (in a style that matches the corresponding names in the NCBI taxonomy)
```{r}

### Zeale ###

zzblastID.cur <- read.csv("data/zzblastID_finaltax.csv")

zz_orig <- zz
zz <- zz %>%
  left_join(select(zzblastID.cur, id, bestblst, final_sci, blastnotes, gbif)) %>%
  select(id, ecotagmatch = match, bestblst, ecotagsci = scientific_name, final_sci, 
         count = total_reads, wt = cluster_weight, starts_with("sample"), sequence, 
         gbif, blastnotes)

### Coleop ###

ccblastID.cur <- read.csv("data/ccblastID_finaltax.csv")

cc_orig <- cc
cc <- cc %>%
  left_join(select(ccblastID.cur, id, bestblst, final_sci, blastnotes, gbif)) %>%
  select(id, ecotagmatch = match, bestblst, ecotagsci = scientific_name, final_sci, 
         count = total_reads, wt = cluster_weight, starts_with("sample"), sequence, 
         gbif, blastnotes)

```
# separate the samples and negatives
```{r}

#### Zeale ####

# check which columns are negatives
colnames(zz) # input those column numbers in the next line
negatives <- c(8,9,17,47,48,49)

# select out just the samples
zs <- zz %>%
  select(-all_of(negatives)) %>%
  select(id, ecotagmatch, bestblst, sci = final_sci,
         count, wt, starts_with("sample"), sequence)

# sum the reads across all samples
sum(select(zs, starts_with("sample"))) # 4115667

# select out just the negatives
zn <- zz %>%
  select(id, ecotagmatch, bestblst, sci = final_sci, 
         count, all_of(negatives), sequence)

# sum the reads across all negatives
sum(select(zn, starts_with("sample"))) # 997

# write.csv(zs, file = "data/pinebugs.samples.step0.nofilters.Z.csv", row.names = F)
# write.csv(zn, file = "data/pinebugs.negatives.step0.nofilters.Z.csv", row.names = F)

#### Coleop ####

# check which columns are negatives
colnames(cc) # input those column numbers in the next line
negatives.C <- c(8,9,17,18,49:61)

# select out just the samples
cs <- cc %>%
  select(-all_of(negatives.C)) %>%
  select(id, ecotagmatch, bestblst, sci = final_sci,
         count, wt, starts_with("sample"), sequence)

# sum the reads across all samples
sum(select(cs, starts_with("sample"))) # 2218624

# select out just the negatives
cn <- cc %>%
  select(id, ecotagmatch, bestblst, sci = final_sci, 
         count, all_of(negatives.C), sequence)

# sum the reads across all negatives
sum(select(cn, starts_with("sample"))) # 58965

# write.csv(cs, file = "data/pinebugs.samples.step0.nofilters.C.csv", row.names = F)
# write.csv(cn, file = "data/pinebugs.negatives.step0.nofilters.C.csv", row.names = F)
```
# make lookup table for adding full taxonomy to samples and negatives
Broken up into 5 parts as it sometimes throws an error midway.
```{r}
library(taxize)

#### adding higher taxa
# taxlistfull <- unique(c(cs$sci, zs$sci))
# taxlist1 <- taxlistfull[1:100]
# # add taxonomy to the lowest taxa rank based on NCBI database
# full_taxa_lookup1 <- taxize::tax_name(sci = taxlist1,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup1, "full_taxa_lookup1.csv", row.names = F)
# 
# taxlist2a <- taxlistfull[101:150]
# # add taxonomy to the lowest taxa rank based on NCBI database
# full_taxa_lookup2a <- taxize::tax_name(sci = taxlist2a,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup2a, "full_taxa_lookup2a.csv", row.names = F)
# 
# taxlist2b <- taxlistfull[151:200]
# # add taxonomy to the lowest taxa rank based on NCBI database
# full_taxa_lookup2b <- taxize::tax_name(sci = taxlist2b,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup2b, "full_taxa_lookup2b.csv", row.names = F)
# 
# taxlist3a <- taxlistfull[201:250]
# # add orders to the lowest taxa rank based on NCBI database
# full_taxa_lookup3a <- taxize::tax_name(sci = taxlist3a,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup3a, "full_taxa_lookup3a.csv", row.names = F)
# 
# taxlist3b <- taxlistfull[251:300]
# # add orders to the lowest taxa rank based on NCBI database
# full_taxa_lookup3b <- taxize::tax_name(sci = taxlist3b,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup3b, "full_taxa_lookup3b.csv", row.names = F)
# 
# taxlist4 <- taxlistfull[301:400]
# # add orders to the lowest taxa rank based on NCBI database
# full_taxa_lookup4 <- taxize::tax_name(sci = taxlist4,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup4, "full_taxa_lookup4.csv", row.names = F)
# 
# taxlist5 <- taxlistfull[401:490]
# # add orders to the lowest taxa rank based on NCBI database
# full_taxa_lookup5 <- taxize::tax_name(sci = taxlist5,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup5, "full_taxa_lookup5.csv", row.names = F)
# 
# # this last one is to fix 2 typos 
# taxlist6 <- c("Amara", "Cerodontha")
# # add orders to the lowest taxa rank based on NCBI database
# full_taxa_lookup6 <- taxize::tax_name(sci = taxlist6,
#                    get = c("genus", "family", "order", "class",
#                            "phylum", "kingdom"),
#                    db = "ncbi")
# write.csv(full_taxa_lookup6, "full_taxa_lookup6.csv", row.names = F)

full_taxa_lookup <- read.csv("full_taxa_lookup1.csv") %>%
  bind_rows(read.csv("full_taxa_lookup2a.csv")) %>%
  bind_rows(read.csv("full_taxa_lookup2b.csv")) %>%
  bind_rows(read.csv("full_taxa_lookup3a.csv")) %>%
  bind_rows(read.csv("full_taxa_lookup3b.csv")) %>%
  bind_rows(read.csv("full_taxa_lookup4.csv")) %>%
  bind_rows(read.csv("full_taxa_lookup5.csv")) %>%
  bind_rows(read.csv("full_taxa_lookup6.csv")) %>%
  filter(!is.na(kingdom))

```
# remove humans, mice, and non-arthropods
```{r}

### Zeale ###

# add higher taxa to sample data
zs1 <- zs %>%
  left_join(full_taxa_lookup, by = c("sci" = "query"))

# check on phylum list and look at MOTUs with no phylum
unique(zs1$phylum)
zs1[is.na(zs1$phylum), c("id", "sci")]

# filter out non-arthropod - samples
zs2 <- zs1 %>%
  filter(phylum == "Arthropoda")

# sum the reads across all samples
sum(select(zs2, starts_with("sample"))) # 4113232

# add higher taxa to negative data
zn1 <- zn %>%
  left_join(full_taxa_lookup, by = c("sci" = "query"))

# check on phylum list and look at MOTUs with no phylum
unique(zn1$phylum)
zn1[is.na(zn1$phylum), c("id", "sci")]

# filter out non-arthropod - negatives
zn2 <- zn1 %>%
  filter(phylum == "Arthropoda")

# sum the reads across all negatives
sum(select(zn2, starts_with("sample"))) # 997

# get the max read count of each seqid present in the negatives
colnames(zn2) # check which columns contain negative samples
zn2$seqmax <- apply(zn2[,6:11],1, max)

# write.csv(zs2, file = "data/pinebugs.samples.step1.onlyarth.Z.csv", row.names = F)
# write.csv(zn2, file = "data/pinebugs.negatives.step1.onlyarth.Z.csv", row.names = F)

### Coleop ###

# add higher taxa to sample data
cs1 <- cs %>%
  left_join(full_taxa_lookup, by = c("sci" = "query"))

# check on phylum list and look at MOTUs with no phylum
unique(cs1$phylum)
cs1[is.na(cs1$phylum), c("id", "sci")]

# filter out non-arthropod - samples
cs2 <- cs1 %>%
  filter(phylum == "Arthropoda")

# sum the reads across all samples
sum(select(cs2, starts_with("sample"))) # 2068994

# add higher taxa to negative data
cn1 <- cn %>%
  left_join(full_taxa_lookup, by = c("sci" = "query"))

# check on phylum list and look at MOTUs with no phylum
unique(cn1$phylum)
cn1[is.na(cn1$phylum), c("id", "sci")]

# filter out non-fish - negatives
cn2 <- cn1 %>%
  filter(phylum == "Arthropoda")

# sum the reads across all negatives
sum(select(cn2, starts_with("sample"))) # 3402

# get the max read count of each seqid present in the negatives
colnames(cn2) # check which columns contain negative samples
cn2$seqmax <- apply(cn2[,6:22],1, max)

# write.csv(cs2, file = "data/pinebugs.samples.step1.onlyarth.C.csv", row.names = F)
# write.csv(cn2, file = "data/pinebugs.negatives.step1.onlyarth.C.csv", row.names = F)

```
# subtract contamination found in negative controls
```{r}

#### Zeale ####

# add the max reads by sequence in all the negatives
zs3a <- zs2 %>%
  left_join(select(zn2, id, seqmax), by = "id") %>%
  data.frame()
zs3 <- zs3a

colnames(zs3) # 7:42 are the samples

# zero out samples with less reads of a sequence than the max contamination among negative of that sequence
for(i in 7:42){
  tmp <- zs3a[,i] - zs3a$seqmax
  tmp2 <- as.numeric(ifelse(tmp<0,0,tmp))
  zs3[,i] <- tmp2
}

# count remaining reads in samples
sum(select(zs3, starts_with("sample"))) # 4098639
rm(zs3a)

# look at sequence lengths
hist(nchar(zs3$sequence))
zs3$len <- nchar(zs3$sequence)

# write.csv(zs3, file = "data/pinebugs.samples.step2.onlyarth.negsub.Z.csv", row.names = F)

#### Coleop ####

# add the max reads by sequence in all the negatives
cs3a <- cs2 %>%
  left_join(select(cn2, id, seqmax), by = "id") %>%
  data.frame()
cs3 <- cs3a

colnames(cs3) # 7:43 are the samples

# zero out samples with less reads of a sequence than the max contamination among negative of that sequence
for(i in 7:43){
  tmp <- cs3a[,i] - cs3a$seqmax
  tmp2 <- as.numeric(ifelse(tmp<0,0,tmp))
  cs3[,i] <- tmp2
}

# count remaining reads in samples
sum(select(cs3, starts_with("sample"))) # 2065112
rm(cs3a)

# look at sequence lengths
hist(nchar(cs3$sequence))
cs3$len <- nchar(cs3$sequence)

# write.csv(cs3, file = "data/pinebugs.samples.step2.onlyarth.negsub.C.csv", row.names = F)

```
# tallying orders, families, species
```{r}

### Zeale ###

# what species; n = 158
zs3 %>% filter(!is.na(genus) & (sci != genus)) %>% 
  select(sci) %>%
  arrange(sci) %>%
  distinct()

# what genus; n = 192
zs3 %>% filter(!is.na(genus)) %>% 
  select(genus) %>%
  arrange(genus) %>%
  distinct()

# what families; n = 110
zs3 %>% filter(!is.na(family)) %>% 
  select(family) %>%
  arrange(family) %>%
  distinct()

# what orders; n = 16
zs3 %>% filter(!is.na(order)) %>% 
  select(order) %>%
  arrange(order) %>%
  distinct()

### Coleop ###

# what species; n = 94
cs3 %>% filter(!is.na(genus) & (genus != sci)) %>% 
  select(sci) %>%
  arrange(sci) %>%
  distinct()

# what genus; n = 154
cs3 %>% filter(!is.na(genus)) %>% 
  select(genus) %>%
  arrange(genus) %>%
  distinct()

# what families; n = 123
cs3 %>% filter(!is.na(family)) %>% 
  select(family) %>%
  arrange(family) %>%
  distinct()

# what orders; n = 23
cs3 %>% filter(!is.na(order)) %>% 
  select(order) %>%
  arrange(order) %>%
  distinct()

```


